{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "sys.path.insert(0, \"..\")\n",
    "from Bio import SeqIO\n",
    "import pandas as pd\n",
    "import igmap"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Run vidjil in ``detect`` mode - pre-select reads that potentially contain V-J junctions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p15 - \n",
      "# 2023-10-02 09:55:43\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p15.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 31 reads (0.0254%)\n",
      "  ==> found 12 windows in 31 reads (0.0254% of 122057 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        1    74.0          1   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       26    75.6          7   0.269\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        1    76.0          1   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        3    75.0          3   1.000\n",
      "\n",
      "  SEG               ->       31    75.5\n",
      "  SEG_+             ->       14    75.3\n",
      "  SEG_-             ->       17    75.6\n",
      "  SEG changed w     ->       11    75.4\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    22928    75.6\n",
      "  UNSEG too few V/J ->    90197    75.5\n",
      "  UNSEG only V/5'   ->     7891    75.5\n",
      "  UNSEG only J/3'   ->     1010    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p15.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p3 - \n",
      "# 2023-10-02 09:55:43\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p3.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.02%\n",
      "  ==> junction detected in 27 reads (0.0221%)\n",
      "  ==> found 15 windows in 27 reads (0.0221% of 122065 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        1    76.0          1   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        0       -          0       -\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       20    75.8          8   0.400\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        1    76.0          1   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        2    75.5          2   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        3    75.7          3   1.000\n",
      "\n",
      "  SEG               ->       27    75.7\n",
      "  SEG_+             ->       10    75.6\n",
      "  SEG_-             ->       17    75.8\n",
      "  SEG changed w     ->       12    75.8\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23101    75.6\n",
      "  UNSEG too few V/J ->    89724    75.5\n",
      "  UNSEG only V/5'   ->     8151    75.5\n",
      "  UNSEG only J/3'   ->     1062    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p3.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p5 - \n",
      "# 2023-10-02 09:55:43\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p5.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 38 reads (0.0311%)\n",
      "  ==> found 19 windows in 38 reads (0.0311% of 122062 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        1    75.0          1   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       33    75.2         14   0.424\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        1    76.0          1   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    75.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        2    76.0          2   1.000\n",
      "\n",
      "  SEG               ->       38    75.2\n",
      "  SEG_+             ->       17    75.4\n",
      "  SEG_-             ->       21    75.0\n",
      "  SEG changed w     ->       10    75.0\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23045    75.6\n",
      "  UNSEG too few V/J ->    90128    75.5\n",
      "  UNSEG only V/5'   ->     7810    75.6\n",
      "  UNSEG only J/3'   ->     1041    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p5.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p7 - \n",
      "# 2023-10-02 09:55:43\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p7.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.02%\n",
      "  ==> junction detected in 29 reads (0.0238%)\n",
      "  ==> found 14 windows in 29 reads (0.0238% of 122062 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        1    76.0          1   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        1    76.0          1   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       27    75.6         12   0.444\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        0       -          0       -\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        0       -          0       -\n",
      "\n",
      "  SEG               ->       29    75.6\n",
      "  SEG_+             ->       16    75.4\n",
      "  SEG_-             ->       13    75.8\n",
      "  SEG changed w     ->        5    75.8\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    22995    75.5\n",
      "  UNSEG too few V/J ->    90002    75.5\n",
      "  UNSEG only V/5'   ->     7943    75.6\n",
      "  UNSEG only J/3'   ->     1093    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p7.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p9 - \n",
      "# 2023-10-02 09:55:43\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p9.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 36 reads (0.0295%)\n",
      "  ==> found 17 windows in 36 reads (0.0295% of 122067 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        3    75.7          3   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       26    75.7          8   0.308\n",
      "  TRA               ->        2    74.0          1   0.500\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        0       -          0       -\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    76.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        4    76.0          4   1.000\n",
      "\n",
      "  SEG               ->       36    75.7\n",
      "  SEG_+             ->       18    75.7\n",
      "  SEG_-             ->       18    75.6\n",
      "  SEG changed w     ->       13    75.4\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23206    75.6\n",
      "  UNSEG too few V/J ->    89888    75.5\n",
      "  UNSEG only V/5'   ->     7895    75.5\n",
      "  UNSEG only J/3'   ->     1042    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p9.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p4 - \n",
      "# 2023-10-02 09:55:43\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p4.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.04%\n",
      "  ==> junction detected in 45 reads (0.0369%)\n",
      "  ==> found 19 windows in 45 reads (0.0369% of 122063 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        5    76.0          4   0.800\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       34    75.7         10   0.294\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        0       -          0       -\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    76.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        5    75.8          4   0.800\n",
      "\n",
      "  SEG               ->       45    75.7\n",
      "  SEG_+             ->       11    75.5\n",
      "  SEG_-             ->       34    75.8\n",
      "  SEG changed w     ->       17    75.8\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23223    75.6\n",
      "  UNSEG too few V/J ->    89864    75.5\n",
      "  UNSEG only V/5'   ->     7906    75.6\n",
      "  UNSEG only J/3'   ->     1025    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p4.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p11 - \n",
      "# 2023-10-02 09:55:43\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p11.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 41 reads (0.0333%)\n",
      "  ==> found 17 windows in 41 reads (0.0333% of 123279 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        0       -          0       -\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       35    75.7         13   0.371\n",
      "  TRA               ->        1    76.0          1   1.000\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        3    75.3          1   0.333\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    75.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        1    76.0          1   1.000\n",
      "\n",
      "  SEG               ->       41    75.7\n",
      "  SEG_+             ->       17    75.6\n",
      "  SEG_-             ->       24    75.7\n",
      "  SEG changed w     ->       16    75.9\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23534    75.5\n",
      "  UNSEG too few V/J ->    90670    75.5\n",
      "  UNSEG only V/5'   ->     7995    75.5\n",
      "  UNSEG only J/3'   ->     1039    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p11.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p10 - \n",
      "# 2023-10-02 09:55:43\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p10.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.04%\n",
      "  ==> junction detected in 49 reads (0.0385%)\n",
      "  ==> found 17 windows in 49 reads (0.0385% of 127144 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        1    74.0          1   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        1    76.0          1   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       38    75.3          9   0.237\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        1    74.0          1   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    76.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        7    75.6          4   0.571\n",
      "\n",
      "  SEG               ->       49    75.3\n",
      "  SEG_+             ->       26    75.3\n",
      "  SEG_-             ->       23    75.4\n",
      "  SEG changed w     ->       16    75.6\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24066    75.6\n",
      "  UNSEG too few V/J ->    93780    75.5\n",
      "  UNSEG only V/5'   ->     8204    75.6\n",
      "  UNSEG only J/3'   ->     1045    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p10.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p6 - \n",
      "# 2023-10-02 09:55:43\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p6.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.04%\n",
      "  ==> junction detected in 47 reads (0.037%)\n",
      "  ==> found 25 windows in 47 reads (0.037% of 127149 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        2    75.0          2   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        4    74.8          4   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       36    75.6         14   0.389\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        0       -          0       -\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        3    75.7          3   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        2    74.0          2   1.000\n",
      "\n",
      "  SEG               ->       47    75.4\n",
      "  SEG_+             ->       30    75.4\n",
      "  SEG_-             ->       17    75.4\n",
      "  SEG changed w     ->       12    75.4\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24336    75.6\n",
      "  UNSEG too few V/J ->    93416    75.5\n",
      "  UNSEG only V/5'   ->     8260    75.6\n",
      "  UNSEG only J/3'   ->     1090    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p6.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p12 - \n",
      "# 2023-10-02 09:55:43\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p12.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 33 reads (0.026%)\n",
      "  ==> found 17 windows in 33 reads (0.026% of 127154 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        2    76.0          2   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       27    75.5         11   0.407\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        1    76.0          1   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        3    75.7          3   1.000\n",
      "\n",
      "  SEG               ->       33    75.5\n",
      "  SEG_+             ->       14    75.1\n",
      "  SEG_-             ->       19    75.8\n",
      "  SEG changed w     ->       10    75.8\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24256    75.6\n",
      "  UNSEG too few V/J ->    93410    75.5\n",
      "  UNSEG only V/5'   ->     8330    75.6\n",
      "  UNSEG only J/3'   ->     1124    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        1    76.0\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p12.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p14 - \n",
      "# 2023-10-02 09:55:43\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p14.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 32 reads (0.0252%)\n",
      "  ==> found 18 windows in 32 reads (0.0252% of 127149 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        2    75.0          2   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        1    76.0          1   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       26    75.2         12   0.462\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        1    75.0          1   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        2    75.5          2   1.000\n",
      "\n",
      "  SEG               ->       32    75.2\n",
      "  SEG_+             ->       14    75.0\n",
      "  SEG_-             ->       18    75.3\n",
      "  SEG changed w     ->       13    75.3\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24228    75.6\n",
      "  UNSEG too few V/J ->    93745    75.5\n",
      "  UNSEG only V/5'   ->     8122    75.6\n",
      "  UNSEG only J/3'   ->     1022    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p14.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p13 - \n",
      "# 2023-10-02 09:55:43\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p13.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 44 reads (0.0346%)\n",
      "  ==> found 19 windows in 44 reads (0.0346% of 127147 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        1    76.0          1   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        3    75.7          2   0.667\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       35    75.6         13   0.371\n",
      "  TRA               ->        2    74.0          1   0.500\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        2    76.0          1   0.500\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        1    75.0          1   1.000\n",
      "\n",
      "  SEG               ->       44    75.5\n",
      "  SEG_+             ->       18    75.4\n",
      "  SEG_-             ->       26    75.6\n",
      "  SEG changed w     ->       14    75.4\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24177    75.6\n",
      "  UNSEG too few V/J ->    93615    75.5\n",
      "  UNSEG only V/5'   ->     8254    75.6\n",
      "  UNSEG only J/3'   ->     1057    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p13.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p8 - \n",
      "# 2023-10-02 09:55:43\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p8.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.05%\n",
      "  ==> junction detected in 61 reads (0.048%)\n",
      "  ==> found 25 windows in 61 reads (0.048% of 127144 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        4    76.0          4   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       53    75.5         18   0.340\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        3    75.0          2   0.667\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        1    76.0          1   1.000\n",
      "\n",
      "  SEG               ->       61    75.5\n",
      "  SEG_+             ->       26    75.5\n",
      "  SEG_-             ->       35    75.5\n",
      "  SEG changed w     ->       21    75.6\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24221    75.6\n",
      "  UNSEG too few V/J ->    93653    75.5\n",
      "  UNSEG only V/5'   ->     8150    75.6\n",
      "  UNSEG only J/3'   ->     1059    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p8.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p2 - \n",
      "# 2023-10-02 09:55:43\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p2.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.02%\n",
      "  ==> junction detected in 31 reads (0.0244%)\n",
      "  ==> found 17 windows in 31 reads (0.0244% of 127153 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        1    75.0          1   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       25    75.5         11   0.440\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        2    76.0          2   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        2    75.5          2   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        1    75.0          1   1.000\n",
      "\n",
      "  SEG               ->       31    75.5\n",
      "  SEG_+             ->       14    75.2\n",
      "  SEG_-             ->       17    75.7\n",
      "  SEG changed w     ->       12    75.8\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24070    75.5\n",
      "  UNSEG too few V/J ->    93746    75.5\n",
      "  UNSEG only V/5'   ->     8257    75.6\n",
      "  UNSEG only J/3'   ->     1049    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p2.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p1 - \n",
      "# 2023-10-02 09:55:43\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p1.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 37 reads (0.0291%)\n",
      "  ==> found 15 windows in 37 reads (0.0291% of 127150 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        3    75.3          3   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       31    75.7          9   0.290\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        1    76.0          1   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    74.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        1    76.0          1   1.000\n",
      "\n",
      "  SEG               ->       37    75.6\n",
      "  SEG_+             ->       19    75.8\n",
      "  SEG_-             ->       18    75.5\n",
      "  SEG changed w     ->       13    75.4\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24290    75.5\n",
      "  UNSEG too few V/J ->    93402    75.5\n",
      "  UNSEG only V/5'   ->     8318    75.6\n",
      "  UNSEG only J/3'   ->     1103    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p1.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p16 - \n",
      "# 2023-10-02 09:55:43\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p16.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 38 reads (0.0299%)\n",
      "  ==> found 15 windows in 38 reads (0.0299% of 127155 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        1    76.0          1   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        2    76.0          2   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       30    75.6          8   0.267\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        3    76.0          2   0.667\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    76.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        1    76.0          1   1.000\n",
      "\n",
      "  SEG               ->       38    75.7\n",
      "  SEG_+             ->       18    75.7\n",
      "  SEG_-             ->       20    75.6\n",
      "  SEG changed w     ->       11    75.5\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23640    75.6\n",
      "  UNSEG too few V/J ->    94066    75.5\n",
      "  UNSEG only V/5'   ->     8373    75.5\n",
      "  UNSEG only J/3'   ->     1038    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p16.vidjil\t(only metadata, no clone output)\n"
     ]
    }
   ],
   "source": [
    "!{igmap.VidjilWrapper().detect_cmd('assets/sample_UHRR_R*.fastq.gz', 'tmp/vidjil')}\n",
    "!cat tmp/vidjil/_p*.fa > tmp/vidjil/first_pass.fa\n",
    "!rm -r tmp/vidjil/_p*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Have a glimpse on fasta output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: ST-E00107:437:h2gtvbbxx:8:1127:2776:1301\n",
      "Name: ST-E00107:437:h2gtvbbxx:8:1127:2776:1301\n",
      "Description: ST-E00107:437:h2gtvbbxx:8:1127:2776:1301 - VJ \t1 53 60 76\t w50/-5 seed IGL SEG_- 6.83e-14 6.83e-14/4.64e-32\n",
      "Number of features: 0\n",
      "Seq('AGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTT...AGG')\n",
      "ID: ST-E00107:437:h2gtvbbxx:8:1128:8099:1018\n",
      "Name: ST-E00107:437:h2gtvbbxx:8:1128:8099:1018\n",
      "Description: ST-E00107:437:h2gtvbbxx:8:1128:8099:1018 + VJ \t1 13 15 74\t w45/10 seed TRD SEG_+ 3.25e-01 4.43e-02/2.80e-01\n",
      "Number of features: 0\n",
      "Seq('CTGCCACTTACTAGCTGGGACACCCACATTTCATCTCTCATCTTCATTTGTAAA...CGT')\n",
      "ID: ST-E00107:437:h2gtvbbxx:8:1128:2052:1022\n",
      "Name: ST-E00107:437:h2gtvbbxx:8:1128:2052:1022\n",
      "Description: ST-E00107:437:h2gtvbbxx:8:1128:2052:1022 - VJ \t1 18 55 74\t seed IGK SEG_- 3.00e-01 2.47e-01/5.28e-02\n",
      "Number of features: 0\n",
      "Seq('TCCTCCAGGGCAAGCCCCGGCTGAGCCTCAAGGGAATACCCACTCTGGTGTACA...GAA')\n"
     ]
    }
   ],
   "source": [
    "for i, x in enumerate(SeqIO.parse(\"tmp/vidjil/first_pass.fa\", \"fasta\")):\n",
    "    if i < 3:\n",
    "        print(x)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Run vidjil in ``clones`` mode - assemble reads with junctions having overlapping \"windows\" into contigs, extract CDR3 regions for some simple cases."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations and analyzing clones\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c clones --all -o tmp/vidjil -b result - \n",
      "# 2023-10-02 09:56:29\n",
      "\n",
      "* WARNING: vidjil-algo was run with '--all' option or with a large '--max-designations/-z' option\n",
      "* vidjil-algo efficiently extracts windows overlapping the CDR3\n",
      "* to cluster reads into clones ('-c clones').\n",
      "* Computing accurate V(D)J designations for many sequences ('-c designations' or large '-z' values)\n",
      "* is not as efficient as the default '-c clones' command.\n",
      "* More information is provided in doc/vidjil-algo.md.\n",
      "\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//result.detected.vdj.fa\n",
      "\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "  ==> junction detected in 612 reads (98.9%)\n",
      "  ==> found 192 windows in 612 reads (98.9% of 619 reads)\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        8    75.2          5   0.625\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->       31    75.6         28   0.903\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->      497    75.5        113   0.227\n",
      "  TRA               ->        5    74.4          1   0.200\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->       20    75.6          6   0.300\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->       14    75.5         13   0.929\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->       37    75.6         26   0.703\n",
      "\n",
      "  SEG               ->      612    75.5\n",
      "  SEG_+             ->      612    75.5\n",
      "  SEG_-             ->        0       -\n",
      "  SEG changed w     ->      202    75.5\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->        0       -\n",
      "  UNSEG too few V/J ->        1    76.0\n",
      "  UNSEG only V/5'   ->        6    75.0\n",
      "  UNSEG only J/3'   ->        0       -\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "Sort windows by number of occurrences\n",
      "  ==> tmp/vidjil//result.windows.fa\n",
      "\n",
      "Diversity measures      IGH   IGK   IGL   TRA   TRB   TRD   TRG   all   \n",
      "  index_H_entropy       1.560 3.300 2.423 0.000 1.228 2.540 3.086 3.224 \n",
      "  index_E_equitability  0.750 0.961 0.390 0.000 0.410 0.962 0.855 0.502 \n",
      "  index_Ds_diversity    0.893 0.994 0.728 0.000 0.621 0.989 0.970 0.820 \n",
      "Considering labeled windows and windows with >= 1 reads and with a ratio >= 0.000 (0)\n",
      "  ==> keep 192 windows in 612 reads (98.869%)  \n",
      "  ==> 192 clones\n",
      "No clustering\n",
      "\n",
      "Detailed analysis of all clones\n",
      "  ==> suggested edges in tmp/vidjil//result.edges\n",
      "#### Clone #001 –     240 reads – 39.2%  \n",
      ">clone-001--IGL--0000240--39.2%--window \n",
      "GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGGGACC\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ3*02 IGLJ3*01 IGLJ2*01\n",
      ">clone-001--IGL--0000240--39.2%--ST-E00107:437:h2gtvbbxx:8:1202:10776:1279-[0,75]-#9 - 76 bp (100% of 75.5 bp) + VJ\t 1 43 47 76\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 5.21e-12 7.32e-14/5.14e-12\n",
      "GGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAGGGACCAAGGTGACC\n",
      "\n",
      "\n",
      "#### Clone #002 –      95 reads – 15.5%  \n",
      ">clone-002--IGL--0000095--15.5%--window \n",
      "CTGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAG\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ3*02 IGLJ3*01 IGLJ2*01\n",
      ">clone-002--IGL--0000095--15.5%--ST-E00107:437:h2gtvbbxx:8:1202:5674:1191-[0,75]-#11 - 76 bp (100% of 75.7 bp) + VJ\t 1 56 60 76\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 1.08e-05 1.09e-21/1.08e-05\n",
      "AGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAGG\n",
      "\n",
      "\n",
      "#### Clone #003 –      28 reads – 4.58%  \n",
      ">clone-003--IGL--0000028--4.58%--window \n",
      "TATTACTGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGG\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ3*02 IGLJ3*01 IGLJ2*01\n",
      ">clone-003--IGL--0000028--4.58%--ST-E00107:437:h2gtvbbxx:8:1128:8448:1226-[0,74]-#1 - 75 bp (100% of 75.0 bp) + VJ\t 1 58 62 75\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 6.81e-04 6.73e-23/6.81e-04\n",
      "CCAGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGG\n",
      "\n",
      "\n",
      "#### Clone #004 –      12 reads – 1.96%  \n",
      ">clone-004--TRB--0000012--1.96%--window \n",
      "TCTGCAGTGCTAGAGAGTCGACTAGCGATCCAAAAAATGAGCAGTTCTTC\n",
      "[warning] (W69) Several genes with equal probability: TRBV20/OR9-2*01 TRBV20-1*01 TRBV20/OR9-2*03\n",
      ">clone-004--TRB--0000012--1.96%--ST-E00107:437:h2gtvbbxx:8:1201:10955:1318-[0,75]-#1 - 76 bp (100% of 75.6 bp) + VDJ\t 1 33 37 44 53 76\tTRBV20/OR9-2*01 0/GTC/2 TRBD2*01 6/ATCCAAAA/7 TRBJ2-1*01  TRB SEG_+ 2.68e-10 2.43e-15/2.68e-10 {20(48)67 p CSARESTSDPKNEQFF}\n",
      "AGACAGCAGCTTCTACATCTGCAGTGCTAGAGA\n",
      "GTCGACTAGCGATCCAAAA\n",
      "AATGAGCAGTTCTTCGGGCCAGGG\n",
      "\n",
      "\n",
      "#### Clone #005 –       6 reads – 0.98%  \n",
      ">clone-005--IGL--0000006--0.98%--window \n",
      "ATGGTAGGCAGCCCTCCCTCTCCATCCCCTCACCCCACCCCTTAGCCACA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-005--IGL--0000006--0.98%--ST-E00107:437:h2gtvbbxx:8:1202:3799:1263-[2,74]-#2 - 73 bp (96% of 75.3 bp) \t ! Possibly IGL @34  @8  IGL UNSEG only J/3' 1.42e+01 1.40e+01/1.70e-01\n",
      "CCCAGACCATGGTAGGCAGCCCTCCCTCTCCATCCCCTCACCCCACCCCTTAGCCACAGTGAAGGGAATGGAA\n",
      "\n",
      "\n",
      "#### Clone #006 –       5 reads – 0.817%  \n",
      ">clone-006--TRA--0000005--0.817%--window \n",
      "TGCGAAGAAGCCACTTGCAGTGACACCAGTGAGAGCATTTTGGAAGAGGA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-006--TRA--0000005--0.817%--ST-E00107:437:h2gtvbbxx:8:1202:9441:1163-[0,73]-#2 - 74 bp (99% of 74.4 bp) \t ! Possibly TRA @55  @37  TRA UNSEG only J/3' 3.07e+00 3.03e+00/4.29e-02\n",
      "GGAGTTTTGAGGAGTATCAGCTGCGAAGAAGCCACTTGCAGTGACACCAGTGAGAGCATTTTGGAAGAGGAACC\n",
      "\n",
      "\n",
      "#### Clone #007 –       4 reads – 0.654%  \n",
      ">clone-007--TRG--0000004--0.654%--window \n",
      "TATCAGAAAGAAGACATTACTATTTATTACGGCAATTCAAGATATTTTCT\n",
      ">clone-007--TRG--0000004--0.654%--ST-E00107:437:h2gtvbbxx:8:1127:9863:1321-[0,75]-#1 - 76 bp (100% of 76.0 bp) + VJ\t 13 38 44 60\tTRGV10*02 18/GGCAA/14 TRGJP1*01  TRG SEG_+ 1.06e-01 1.05e-01/6.27e-04\n",
      "GGAATGGCTATCAGAAAGAAGACATTACTATTTATTAC\n",
      "GGCAA\n",
      "TTCAAGATATTTTCTGATTTCCGGAAGTCAGGG\n",
      "\n",
      "\n",
      "#### Clone #008 –       4 reads – 0.654%  \n",
      ">clone-008--TRB--0000004--0.654%--window \n",
      "CTTCTGTGCCATCAGTGAGCCGACAGGGATCAGAAGAGACCCAGTACTTC\n",
      "[warning] (W69) Several genes with equal probability: TRBV10-3*01 TRBV10-3*02\n",
      ">clone-008--TRB--0000004--0.654%--ST-E00107:437:h2gtvbbxx:8:1202:7822:1115-[0,75]-#3 - 76 bp (100% of 75.5 bp) + VDJ\t 1 24 27 33 39 76\tTRBV10-3*01 2/CC/2 TRBD1*01 3/ATCAG/3 TRBJ2-5*01  TRB SEG_+ 1.28e-09 1.28e-09/2.50e-19 {10(46)55 u CAISEPTGIR#ETQYF}\n",
      "GTGTACTTCTGTGCCATCAGTGAG\n",
      "CCGACAGGGATCAG\n",
      "AAGAGACCCAGTACTTCGGGCCAGGCACGCGGCTCCTG\n",
      "\n",
      "\n",
      "#### Clone #009 –       4 reads – 0.654%  \n",
      ">clone-009--TRG--0000004--0.654%--window \n",
      "CAGATTGTGACTCCTGGAGATTTGGATATATGTGAATATAATTTTAGAAG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-009--TRG--0000004--0.654%--ST-E00107:437:h2gtvbbxx:8:1202:5015:1261-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly TRG @30  @49  TRG UNSEG only J/3' 3.29e+00 2.38e+00/9.08e-01\n",
      "ATAGAGCAACAGATTGTGACTCCTGGAGATTTGGATATATGTGAATATAATTTTAGAAGCTTATCATTTACCTATG\n",
      "\n",
      "\n",
      "#### Clone #010 –       4 reads – 0.654%  \n",
      ">clone-010--TRG--0000004--0.654%--window \n",
      "AGAAAAGCAATAATTTTACAGGCATTTTGAGGTGTCTCTTTCTTTGGGTT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-010--TRG--0000004--0.654%--ST-E00107:437:h2gtvbbxx:8:1128:6463:1279-[0,74]-#1 - 75 bp (99% of 75.2 bp) \t ! Possibly TRG @43  @18  TRG UNSEG too few V/J 1.02e+01 6.64e+00/3.58e+00\n",
      "CTTAGAGAAAGAAAAGCAATAATTTTACAGGCATTTTGAGGTGTCTCTTTCTTTGGGTTCTTTCTGTTTGAAAGG\n",
      "\n",
      "\n",
      "#### Clone #011 –       4 reads – 0.654%  \n",
      ">clone-011--IGL--0000004--0.654%--window \n",
      "AAGCCCTCAGCACCACTTTGAACAGAATTCCATCATCCCTGGCCCTCCCT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-011--IGL--0000004--0.654%--ST-E00107:437:h2gtvbbxx:8:1128:6541:1010-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @34  @23  IGL UNSEG only V/5' 4.04e+00 4.03e-02/4.00e+00\n",
      "CAGGAACAGACTTGGGAACAAACACAAGCCCTCAGCACCACTTTGAACAGAATTCCATCATCCCTGGCCCTCCCTT\n",
      "\n",
      "\n",
      "#### Clone #012 –       3 reads – 0.49%  \n",
      ">clone-012--IGL--0000003--0.49%--window \n",
      "GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAAGGACC\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ2*01 IGLJ3*01 IGLJ3*02\n",
      ">clone-012--IGL--0000003--0.49%--ST-E00107:437:h2gtvbbxx:8:1128:7193:1162-[0,74]-#1 - 75 bp (100% of 75.0 bp) + VJ\t 1 51 55 75\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 1.33e-06 1.10e-18/1.33e-06\n",
      "GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAAGGACC\n",
      "\n",
      "\n",
      "#### Clone #013 –       3 reads – 0.49%  \n",
      ">clone-013--IGL--0000003--0.49%--window \n",
      "CCCGTGCTGAAGAGCAGCCTGGCCTTGGGCCTGGAGCTGCGGGCCGCAGC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-013--IGL--0000003--0.49%--ST-E00107:437:h2gtvbbxx:8:1201:1772:1317-[0,75]-#2 - 76 bp (100% of 75.3 bp) \t ! Possibly IGL @25  @29  IGL UNSEG too few V/J 9.15e+00 5.15e+00/4.00e+00\n",
      "GATGCCCGTGCTGAAGAGCAGCCTGGCCTTGGGCCTGGAGCTGCGGGCCGCAGCCGGGAGCCACTTTGATGCTGCG\n",
      "\n",
      "\n",
      "#### Clone #014 –       3 reads – 0.49%  \n",
      ">clone-014--IGL--0000003--0.49%--window \n",
      "ACCCTCAGTGCCAGCCAGCCTCTGCCAGACCTCTCTTTCCCTCTTCTCCC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-014--IGL--0000003--0.49%--ST-E00107:437:h2gtvbbxx:8:1201:8130:1079-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @35  @1  IGL UNSEG only V/5' 4.00e+00 9.83e-06/4.00e+00\n",
      "GCACAGCTCCAGCTCGGACAGCACCCTCAGTGCCAGCCAGCCTCTGCCAGACCTCTCTTTCCCTCTTCTCCCCAAC\n",
      "\n",
      "\n",
      "#### Clone #015 –       2 reads – 0.327%  \n",
      ">clone-015--IGK--0000002--0.327%--window \n",
      "TTCCCTGTGACCCAGCCTGTTCCATAGCCAAAGGGACCAAAGTGAACGAT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-015--IGK--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:10087:1165-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGK @68  @34  IGK UNSEG only J/3' 7.03e+00 7.02e+00/9.07e-03\n",
      "CTGGCAGTTCCCTGTGACCCAGCCTGTTCCATAGCCAAAGGGACCAAAGTGAACGATCAGAGTGCCCCGCAACTAT\n",
      "\n",
      "\n",
      "#### Clone #016 –       2 reads – 0.327%  \n",
      ">clone-016--IGK--0000002--0.327%--window \n",
      "TTACTTATCAAATGGATTATCTCCCCCTCTTTAAAAGTAAGTTCATCTTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-016--IGK--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:3099:1077-[0,75]-#1 - 76 bp (100% of 75.5 bp) \t ! Possibly IGK @51  @67  IGK UNSEG too few V/J 5.04e+00 1.75e+00/3.28e+00\n",
      "CACCAGCCAGCTTCTCCAGTCTCCTTACTTATCAAATGGATTATCTCCCCCTCTTTAAAAGTAAGTTCATCTTCAT\n",
      "\n",
      "\n",
      "#### Clone #017 –       2 reads – 0.327%  \n",
      ">clone-017--IGL--0000002--0.327%--window \n",
      "TGTGCCTGGACGGGAAGATGGCAAGGAGGCACCGGAGGAGGCACAGCTCG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-017--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:2863:1247-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @50  @59  IGL UNSEG only J/3' 1.46e+01 1.46e+01/5.52e-03\n",
      "CAGGCCCAGATCAAGGGGCCCGTGGATGTGCCTGGACGGGAAGATGGCAAGGAGGCACCGGAGGAGGCACAGCTCG\n",
      "\n",
      "\n",
      "#### Clone #018 –       2 reads – 0.327%  \n",
      ">clone-018--IGL--0000002--0.327%--window \n",
      "TGGGGGACACAGGCACGCTGCTTTCCCTCCCAAGCCAGTGGGCCCCCACC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-018--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:10264:1397-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @45  @4  IGL UNSEG too few V/J 5.28e+00 1.29e+00/4.00e+00\n",
      "CTCCTGGGGGACACAGGCACGCTGCTTTCCCTCCCAAGCCAGTGGGCCCCCACCCGACTCCCCACTCTGCAGCTTC\n",
      "\n",
      "\n",
      "#### Clone #019 –       2 reads – 0.327%  \n",
      ">clone-019--TRG--0000002--0.327%--window \n",
      "TGAAAATATCTATTGAAAATGAGCAACTTGTGATTGGATCATATAGTCAG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-019--TRG--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:4510:1071-[0,75]-#1 - 76 bp (102% of 74.0 bp) \t ! Possibly TRG @36  @44  TRG UNSEG only J/3' 2.38e+00 2.38e+00/1.25e-03\n",
      "AAAGTACAGACTTCTGAAAATATCTATTGAAAATGAGCAACTTGTGATTGGATCATATAGTCAGCCTTCAGATTCC\n",
      "\n",
      "\n",
      "#### Clone #020 –       2 reads – 0.327%  \n",
      ">clone-020--TRD--0000002--0.327%--window \n",
      "TCTCAGAAAGCCTTTCTGACTGATCTCAGCATTACCTCTTTGGAAAAGGA\n",
      ">clone-020--TRD--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:1495:1101-[0,74]-#1 - 75 bp (100% of 75.0 bp) + VJ\t 46 60 61 72\tTRAV14/DV4*04 0//18 TRDJ1*01  TRD SEG_+ 8.04e-01 8.04e-01/2.86e-04\n",
      "CCCAGAAATCCAAGCCAGTTTTCTCAGAAAGCCTTTCTGACTGATCTCAGCATTACCTCT\n",
      "\n",
      "TTGGAAAAGGAAGGT\n",
      "\n",
      "\n",
      "#### Clone #021 –       2 reads – 0.327%  \n",
      ">clone-021--IGL--0000002--0.327%--window \n",
      "TACTGCAGTTCATATAGAGGCACCGCCACTTTCGAGGTGGTGTTCGGCGG\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ3*02 IGLJ3*01 IGLJ2*01\n",
      ">clone-021--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1201:7769:1266-[0,75]-#1 - 76 bp (100% of 76.0 bp) + VJ\t 1 56 60 76\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 1.08e-05 3.49e-20/1.08e-05\n",
      "AGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCACCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAGG\n",
      "\n",
      "\n",
      "#### Clone #022 –       2 reads – 0.327%  \n",
      ">clone-022--IGL--0000002--0.327%--window \n",
      "GTTTTTCTGCCTCAGCCTCCTGAGTAGCTAAGGGAGGTGTCTTGAGATTA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-022--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:6128:1156-[0,75]-#2 - 76 bp (100% of 75.5 bp) \t ! Possibly IGL @31  @59  IGL UNSEG too few V/J 1.14e+00 6.44e-01/5.00e-01\n",
      "GCGTGGTTTTTCTGCCTCAGCCTCCTGAGTAGCTAAGGGAGGTGTCTTGAGATTATCATCGGCTGAGGGTGGAAGC\n",
      "\n",
      "\n",
      "#### Clone #023 –       2 reads – 0.327%  \n",
      ">clone-023--IGL--0000002--0.327%--window \n",
      "GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGGAACC\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ2*01 IGLJ3*01 IGLJ3*02\n",
      ">clone-023--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:3700:1232-[0,74]-#1 - 75 bp (100% of 75.0 bp) + VJ\t 1 51 55 75\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 1.33e-06 1.10e-18/1.33e-06\n",
      "GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAGGAACC\n",
      "\n",
      "\n",
      "#### Clone #024 –       2 reads – 0.327%  \n",
      ">clone-024--IGL--0000002--0.327%--window \n",
      "GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGAGACC\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ2*01 IGLJ3*01 IGLJ3*02\n",
      ">clone-024--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1127:9570:1382-[0,74]-#1 - 75 bp (100% of 75.0 bp) + VJ\t 1 51 55 75\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 1.33e-06 1.10e-18/1.33e-06\n",
      "GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAGAGACC\n",
      "\n",
      "\n",
      "#### Clone #025 –       2 reads – 0.327%  \n",
      ">clone-025--IGL--0000002--0.327%--window \n",
      "GATCTGCCTATTACTGCTGTGCCCCATGGGGGGCTCCTTCCCTGCCTGAC\n",
      "[warning] (W69) Several genes with equal probability: IGLV10-67*01 IGLV10-67*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ7*02 IGLJ7*01\n",
      ">clone-025--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:10458:1104-[0,75]-#1 - 76 bp (100% of 76.0 bp) + VJ\t 30 44 48 47\tIGLV10-67*01 28/GTG/46 IGLJ7*01  IGL SEG_+ 7.05e-01 4.55e-01/2.50e-01\n",
      "ATGTGCTCTGAGCTTCCCTGAACCAGGATCTGCCTATTACTGCT\n",
      "GTG\n",
      "CCCCATGGGGGGCTCCTTCCCTGCCTGAC\n",
      "\n",
      "\n",
      "#### Clone #026 –       2 reads – 0.327%  \n",
      ">clone-026--TRG--0000002--0.327%--window \n",
      "GAGCAGCAGGAGGGGGGAGGGAAGGGAGCCAAGGGATGGACATCTTGTCA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-026--TRG--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:7743:1161-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly TRG @41  @32  TRG UNSEG too few V/J 7.51e+00 2.38e+00/5.14e+00\n",
      "AGGAGCAGCAGGAGGGGGGAGGGAAGGGAGCCAAGGGATGGACATCTTGTCATTTTTTTTCTTTGAATAAATGTCA\n",
      "\n",
      "\n",
      "#### Clone #027 –       2 reads – 0.327%  \n",
      ">clone-027--IGL--0000002--0.327%--window \n",
      "CCTCTGCCACCAGCCTCTTCTCCAACAGACTCTCCTTCCCCTTTTCCAGG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-027--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:8201:1155-[0,75]-#2 - 76 bp (101% of 75.0 bp) \t ! Possibly IGL @40  @9  IGL UNSEG too few V/J 9.30e+00 3.64e+00/5.65e+00\n",
      "CCACCTCCTCTTCGTCCTCTTCCTCCTCTGCCACCAGCCTCTTCTCCAACAGACTCTCCTTCCCCTTTTCCAGGTC\n",
      "\n",
      "\n",
      "#### Clone #028 –       2 reads – 0.327%  \n",
      ">clone-028--IGH--0000002--0.327%--window \n",
      "CAGGAAGGGGCTGGACTGGCTGTCCTTTCCAGTGGCCTGGCTCCGCTGTG\n",
      "[warning] (W69) Several genes with equal probability: IGHJ1P*02 IGHJ1P*01\n",
      ">clone-028--IGH--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:9291:1085-[0,75]-#2 - 76 bp (101% of 75.0 bp) + VDJ\t 6 21 22 27 28 22\tIGHV(II)-46-1*01 8//15 IGHD3/OR15-3a*01 10//34 IGHJ1P*01  IGH SEG_+ 4.11e-01 7.08e-02/3.40e-01\n",
      "CCTCACTCCAGGAAGGGGCTG\n",
      "GACTGG\n",
      "CTGTCCTTTCCAGTGGCCTGGCTCCGCTGTGTGGATGGGGAGATCGGGG\n",
      "\n",
      "\n",
      "#### Clone #029 –       2 reads – 0.327%  \n",
      ">clone-029--IGK--0000002--0.327%--window \n",
      "CAAGCCCCGGCTGAGCCTCAAGGGAATACCCACTCTGGTGTACAGACACG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-029--IGK--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:6890:1291-[0,75]-#2 - 76 bp (101% of 75.0 bp) \t ! Possibly IGK @56  @52  IGK UNSEG only J/3' 1.00e+01 9.92e+00/1.03e-01\n",
      "TCCAGGGCAAGCCCCGGCTGAGCCTCAAGGGAATACCCACTCTGGTGTACAGACACGACTGGAACTATGAACATCT\n",
      "\n",
      "\n",
      "#### Clone #030 –       2 reads – 0.327%  \n",
      ">clone-030--IGH--0000002--0.327%--window \n",
      "ATTTGCTCCCCTGAGTGTGTAGGTCCCTGAGACCTGCTCACCAGCCCCTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-030--IGH--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1201:10903:1363-[0,75]-#2 - 76 bp (101% of 75.0 bp) \t ! Possibly IGH @61  @40  IGH UNSEG too few V/J 1.29e+01 9.07e+00/3.85e+00\n",
      "GCTCTGGGCCATCCCAGGAGGGGGAATTTGCTCCCCTGAGTGTGTAGGTCCCTGAGACCTGCTCACCAGCCCCTCC\n",
      "\n",
      "\n",
      "#### Clone #031 –       2 reads – 0.327%  \n",
      ">clone-031--IGL--0000002--0.327%--window \n",
      "AGGGCCTCCTGGACACCTACAGCGTGTGGAGGAGTGGTCTCCCCAACTTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-031--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:4932:1253-[0,75]-#2 - 76 bp (100% of 75.5 bp) \t ! Possibly IGL @57  @44  IGL UNSEG too few V/J 6.47e+00 3.64e+00/2.83e+00\n",
      "CTTTTCCCCGGGCCTTCCAGGGCCTCCTGGACACCTACAGCGTGTGGAGGAGTGGTCTCCCCAACTTCCTAGCAGT\n",
      "\n",
      "\n",
      "#### Clone #032 –       2 reads – 0.327%  \n",
      ">clone-032--IGL--0000002--0.327%--window \n",
      "ACCATCAGTGTCAGCATTAGGGTTTTGGTTTTTGTTTCTTTTGGGTCTTT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-032--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:7815:1217-[0,75]-#2 - 76 bp (100% of 75.5 bp) \t ! Possibly IGL @21  @36  IGL UNSEG only J/3' 5.40e+00 5.15e+00/2.50e-01\n",
      "ACTGTTCAGGGTACTGACCATCAGTGTCAGCATTAGGGTTTTGGTTTTTGTTTCTTTTGGGTCTTTCTTTTTTGGC\n",
      "\n",
      "\n",
      "#### Clone #033 –       2 reads – 0.327%  \n",
      ">clone-033--IGH--0000002--0.327%--window \n",
      "AATTAGATCCCAGCCTGGACAAGGGCATGACTATTGCTGTTTGGGGACAC\n",
      "[warning] (W69) Several genes with equal probability: IGHV1/OR16-4*02 IGHV1/OR16-4*01 IGHV1/OR16-3*01 IGHV1/OR16-22*01\n",
      ">clone-033--IGH--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1201:4878:1117-[0,75]-#1 - 76 bp (100% of 76.0 bp) + VDJ\t 40 56 57 60 61 75\tIGHV1/OR16-22*01 13//3 IGHD4/OR15-4a*01 12//9 IGHJ3*01  IGH SEG_+ 1.30e-01 1.00e-01/3.01e-02\n",
      "GGAGGGAGTCAACGACCTATTGTATCAATTAGATCCCAGCCTGGACAAGGGCATGA\n",
      "CTAT\n",
      "TGCTGTTTGGGGACAC\n",
      "\n",
      "\n",
      "#### Clone #034 –       1 reads – 0.163%  \n",
      ">clone-034--IGL--0000001--0.163%--window \n",
      "TTTTTTTTTTTTTTCTTCCTTTTCCCTTTCTCGGTTCTCTTCTCTCCTCT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-034--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1202:9858:1155-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @72  @25  IGL UNSEG too few V/J 1.13e+01 7.29e+00/4.00e+00\n",
      "CAATACTAAGCTTTGCTTTTTTTTTTTTTTCTTCCTTTTCCCTTTCTCGGTTCTCTTCTCTCCTCTGAATGGCTGG\n",
      "\n",
      "\n",
      "#### Clone #035 –       1 reads – 0.163%  \n",
      ">clone-035--IGL--0000001--0.163%--window \n",
      "TTTTTTTTTTTTTGAAATGGAGTATTTCTCATTAGGTTTCACTGTCTTAT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-035--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1201:4711:1342-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @74  @62  IGL UNSEG only J/3' 1.10e+01 1.03e+01/7.07e-01\n",
      "CACCTTTTATATGAGTGGATTTTTTTTTTTTTTTTGAAATGGAGTATTTCTCATTAGGTTTCACTGTCTTATCCTT\n",
      "\n",
      "\n",
      "#### Clone #036 –       1 reads – 0.163%  \n",
      ">clone-036--IGL--0000001--0.163%--window \n",
      "TTTTTTTTTTTTTAGCTGAAGGATGGTTAGACAGCACAGCGAAGA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-036--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:2960:1395-[0,73]-#1 - 74 bp (100% of 74.0 bp) \t ! Possibly IGL @68  @7  IGL UNSEG only J/3' 1.05e+01 1.00e+01/4.86e-01\n",
      "TTTTTTTTTTTTTTTAGCTGAAGGATGGTTAGACAGCACAGCGAAGACTGCTAAATGCACTGACCCCCCCCATT\n",
      "\n",
      "\n",
      "#### Clone #037 –       1 reads – 0.163%  \n",
      ">clone-037--IGL--0000001--0.163%--window \n",
      "TTTTTTTTTTATTTTGCCTTTGTTTGTTTTTAACGTTTATCCTCTCCTCT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-037--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1201:6253:1105-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @11  @24  IGL UNSEG only J/3' 2.70e+00 2.58e+00/1.25e-01\n",
      "CTGATACTGATTTTTTTTTTTTTTTTTTTATTTTGCCTTTGTTTGTTTTTAACGTTTATCCTCTCCTCTTTGTGTG\n",
      "\n",
      "\n",
      "#### Clone #038 –       1 reads – 0.163%  \n",
      ">clone-038--IGL--0000001--0.163%--window \n",
      "TTTTTTTTTTATAATGAGTATGATCTATTTCTTTTCAAATAATCTTTGAG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-038--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:10969:1112-[0,74]-#1 - 75 bp (100% of 75.0 bp) \t ! Possibly IGL @37  @17  IGL UNSEG too few V/J 5.57e+00 3.60e+00/1.97e+00\n",
      "ATCTTGTTTTTTTTTTTTTTTTATAATGAGTATGATCTATTTCTTTTCAAATAATCTTTGAGATCCCAGGAAAAA\n",
      "\n",
      "\n",
      "#### Clone #039 –       1 reads – 0.163%  \n",
      ">clone-039--IGK--0000001--0.163%--window \n",
      "TTTCAATGGGATTCAACAGGGCCACCTTTTTCCACTGAAGTATTAGGAGG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-039--IGK--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1127:1340:1396-[0,74]-#1 - 75 bp (100% of 75.0 bp) \t ! Possibly IGK @45  @35  IGK UNSEG too few V/J 2.19e+01 1.96e+01/2.29e+00\n",
      "TCAAGTCTGGGTTTTCAATGGGATTCAACAGGGCCACCTTTTTCCACTGAAGTATTAGGAGGACGATGAAGACGA\n",
      "\n",
      "\n",
      "#### Clone #040 –       1 reads – 0.163%  \n",
      ">clone-040--TRD--0000001--0.163%--window \n",
      "TTTATTTTTTTTATTTTTTTATTTTTTATTTTTGTATATTTCTTTAAATT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-040--TRD--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1202:10947:1050-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly TRD @16  @66  TRD UNSEG only V/5' 6.73e+00 1.80e-02/6.71e+00\n",
      "ATTCTGTGCATGTGACCTTTTTTTTTTATTTTTTTTATTTTTTTATTTTTTATTTTTGTATATTTCTTTAAATTCC\n",
      "\n",
      "\n",
      "#### Clone #041 –       1 reads – 0.163%  \n",
      ">clone-041--IGL--0000001--0.163%--window \n",
      "TTGGGATTTTTTTTTTTTTTTTTGGCTTTTGTTTTTGTTTGTTTTTTTGT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-041--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1202:5423:1130-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @17  @46  IGL UNSEG only J/3' 1.85e+00 1.82e+00/3.12e-02\n",
      "AGTGTATATGGCTTGTGTTTTGGGATTTTTTTTTTTTTTTTTGGCTTTTGTTTTTGTTTGTTTTTTTGTTTCGTTT\n",
      "\n",
      "\n",
      "#### Clone #042 –       1 reads – 0.163%  \n",
      ">clone-042--TRG--0000001--0.163%--window \n",
      "TTGAAAATGCAAAGCGAGTGAGAAAAGAAAGTGAAAAAATCAAGG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-042--TRG--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:7455:1164-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly TRG @26  @65  TRG UNSEG only J/3' 4.81e+00 4.76e+00/5.67e-02\n",
      "AATGAAGTCAACAAAAGCACTGTTTACATTATTGAAAATGCAAAGCGAGTGAGAAAAGAAAGTGAAAAAATCAAGG\n",
      "\n",
      "\n",
      "#### Clone #043 –       1 reads – 0.163%  \n",
      ">clone-043--TRD--0000001--0.163%--window \n",
      "TTCTAAAGAAAGAAGGAGCATAGTACTCATTTATTTACTCCCGCATTCAG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-043--TRD--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:3842:1346-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly TRD @20  @39  TRD UNSEG too few V/J 1.59e+01 9.21e+00/6.71e+00\n",
      "CTTCTAAAGAAAGAAGGAGCATAGTACTCATTTATTTACTCCCGCATTCAGTACTTATTCCAATCACTTCATTAGG\n",
      "\n",
      "\n",
      "#### Clone #044 –       1 reads – 0.163%  \n",
      ">clone-044--IGL--0000001--0.163%--window \n",
      "TTCATATAGAGGCAGCGCCACTTTCGAGGGGGTGTTCGGCGGAGGGACCA\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-18*02 IGLV2-18*03 IGLV2-18*04 IGLV2-28*01\n",
      ">clone-044--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:2070:1410-[0,73]-#1 - 74 bp (100% of 74.0 bp) + VJ\t 1 31 36 73\tIGLV2-18*02 0/GAGG/2 IGLJ3*02  IGL SEG_+ 1.20e-06 1.20e-06/1.08e-16 {2(42)43 p CSSYRGSATFEGVF}\n",
      "CTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAGG\n",
      "GGGTGTTCGGCGGAGGGACCAAGGTGACCGTCCTACGTC\n",
      "\n",
      "\n",
      "#### Clone #045 –       1 reads – 0.163%  \n",
      ">clone-045--TRD--0000001--0.163%--window \n",
      "TTCAGCACCATCCCTGAGCTCATTAACTACCCTCAGCACAACTCTGCAGG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-045--TRD--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1201:7800:1203-[0,74]-#1 - 75 bp (100% of 75.0 bp) \t ! Possibly TRD @62  @34  TRD UNSEG only J/3' 9.09e+00 9.09e+00/8.08e-04\n",
      "TTCAGCACCATCCCTGAGCTCATTAACTACCCTCAGCACAACTCTGCAGGACTCATATCCAGGCTCAAATATCCA\n",
      "\n",
      "\n",
      "#### Clone #046 –       1 reads – 0.163%  \n",
      ">clone-046--IGK--0000001--0.163%--window \n",
      "TTAGATTACACTCATACCATTGTATCATCTTGCTATAAAAAGTTACTTAA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-046--IGK--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:7574:1305-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGK @50  @11  IGK UNSEG too few V/J 3.40e+00 1.75e+00/1.64e+00\n",
      "AAGTTTAGATTACACTCATACCATTGTATCATCTTGCTATAAAAAGTTACTTAAAACTATTTCTTTTGCATATAAA\n",
      "\n",
      "\n",
      "#### Clone #047 –       1 reads – 0.163%  \n",
      ">clone-047--IGL--0000001--0.163%--window \n",
      "TGTTTCTGTTTTTTTTTTTTTTTTGTGCCTCTTTTTCCCCTTCCCTACTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-047--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:9796:1410-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @76  @20  IGL UNSEG too few V/J 1.14e+00 6.44e-01/5.00e-01\n",
      "TGTCGACTGAAACTCACTTTTATGTTTCTGTTTTTTTTTTTTTTTTGTGCCTCTTTTTCCCCTTCCCTACTCATCT\n",
      "\n",
      "\n",
      "#### Clone #048 –       1 reads – 0.163%  \n",
      ">clone-048--IGK--0000001--0.163%--window \n",
      "TGTTATAGTCTGATTATTTCTGTGTTTTGAATATGATTTTCCTAATGCTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-048--IGK--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1202:4873:1086-[0,73]-#1 - 74 bp (100% of 74.0 bp) \t ! Possibly IGK @67  @1  IGK UNSEG only V/5' 6.47e+00 7.55e-02/6.40e+00\n",
      "TGTATACCACTATCTATGCAGCATTGTTATAGTCTGATTATTTCTGTGTTTTGAATATGATTTTCCTAATGCTC\n",
      "\n",
      "\n",
      "#### Clone #049 –       1 reads – 0.163%  \n",
      ">clone-049--TRD--0000001--0.163%--window \n",
      "TGTGGTGTCTGAAGACCAAGACTACACAGCTCATTTTGCTGATCCTGATA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-049--TRD--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1202:3024:1049-[0,74]-#1 - 75 bp (100% of 75.0 bp) \t ! Possibly TRD @71  @62  TRD UNSEG only J/3' 1.32e+01 1.29e+01/2.93e-01\n",
      "TTGTGGTGTCTGAAGACCAAGACTACACAGCTCATTTTGCTGATCCTGATACATTAGTTAACTGGGACTTTGTGG\n",
      "\n",
      "\n",
      "#### Clone #050 –       1 reads – 0.163%  \n",
      ">clone-050--TRG--0000001--0.163%--window \n",
      "TGTAGCAAAACAAAAAGTTTTATAAATTTCAGATAAGCTTTCTGTTTTTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-050--TRG--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1201:2533:1310-[0,74]-#1 - 75 bp (100% of 75.0 bp) \t ! Possibly TRG @59  @39  TRG UNSEG only J/3' 2.13e+02 2.12e+02/8.96e-01\n",
      "TATTCAAATACCTTCTGTTTTCTATGTAGCAAAACAAAAAGTTTTATAAATTTCAGATAAGCTTTCTGTTTTTCT\n",
      "\n",
      "\n",
      ".....\n",
      "\n",
      "#### Clones #510 to #192...\n",
      "#### end of clones\n",
      "\n",
      "Please review the 25 suggested edge(s) in tmp/vidjil//.edges\n",
      "Comparing clone consensus sequences 2 by 2\n",
      "    | 0001 0002 0003 0004 0005 0006 0007 0008 0009 0010 0011 0012 0013 0014 0015 \n",
      "001 | 0100 0070 0065 0054 0045 0053 0051 0053 0046 0046 0047 0079 0052 0043 0048    ST-E00107:437:h2gtvbbxx:8:1202:10776:1279-[0,75]-#9 - 76 bp (100% of 75.5 bp)\n",
      "002 |      0100 0093!0058 0046 0051 0052 0051 0048 0055 0043 0088 0050 0047 0047    ST-E00107:437:h2gtvbbxx:8:1202:5674:1191-[0,75]-#11 - 76 bp (100% of 75.7 bp)\n",
      "003 |           0100 0057 0048 0051 0052 0050 0047 0054 0044 0082 0053 0048 0048    ST-E00107:437:h2gtvbbxx:8:1128:8448:1226-[0,74]-#1 - 75 bp (100% of 75.0 bp)\n",
      "004 |                0100 0045 0054 0052 0057 0055 0049 0048 0058 0047 0051 0052    ST-E00107:437:h2gtvbbxx:8:1201:10955:1318-[0,75]-#1 - 76 bp (100% of 75.6 bp)\n",
      "005 |                     0100 0047 0043 0040 0041 0040 0044 0047 0052 0052 0048    ST-E00107:437:h2gtvbbxx:8:1202:3799:1263-[2,74]-#2 - 73 bp (96% of 75.3 bp)\n",
      "006 |                          0100 0052 0052 0047 0048 0048 0055 0053 0048 0050    ST-E00107:437:h2gtvbbxx:8:1202:9441:1163-[0,73]-#2 - 74 bp (99% of 74.4 bp)\n",
      "007 |                               0100 0048 0052 0055 0051 0053 0045 0048 0044    ST-E00107:437:h2gtvbbxx:8:1127:9863:1321-[0,75]-#1 - 76 bp (100% of 76.0 bp)\n",
      "008 |                                    0100 0048 0040 0051 0052 0053 0048 0056    ST-E00107:437:h2gtvbbxx:8:1202:7822:1115-[0,75]-#3 - 76 bp (100% of 75.5 bp)\n",
      "009 |                                         0100 0051 0049 0045 0044 0043 0050    ST-E00107:437:h2gtvbbxx:8:1202:5015:1261-[0,75]-#1 - 76 bp (100% of 76.0 bp)\n",
      "010 |                                              0100 0046 0054 0043 0043 0038    ST-E00107:437:h2gtvbbxx:8:1128:6463:1279-[0,74]-#1 - 75 bp (99% of 75.2 bp)\n",
      "011 |                                                   0100 0048 0041 0056 0051    ST-E00107:437:h2gtvbbxx:8:1128:6541:1010-[0,75]-#1 - 76 bp (100% of 76.0 bp)\n",
      "012 |                                                        0100 0048 0047 0046    ST-E00107:437:h2gtvbbxx:8:1128:7193:1162-[0,74]-#1 - 75 bp (100% of 75.0 bp)\n",
      "013 |                                                             0100 0052 0049    ST-E00107:437:h2gtvbbxx:8:1201:1772:1317-[0,75]-#2 - 76 bp (100% of 75.3 bp)\n",
      "014 |                                                                  0100 0051    ST-E00107:437:h2gtvbbxx:8:1201:8130:1079-[0,75]-#1 - 76 bp (100% of 76.0 bp)\n",
      "015 |                                                                       0100    ST-E00107:437:h2gtvbbxx:8:1128:10087:1165-[0,75]-#1 - 76 bp (100% of 76.0 bp)\n",
      "\n",
      "  ==> tmp/vidjil/result.tsv   \t(AIRR output)\n",
      "  ==> tmp/vidjil/result.vidjil\t(main output file, may be opened by the Vidjil web application)\n"
     ]
    }
   ],
   "source": [
    "!{igmap.VidjilWrapper().clones_cmd('tmp/vidjil/first_pass.fa', 'tmp/vidjil')}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Have a glimpse on \"windows\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: 240--window--1\n",
      "Name: 240--window--1\n",
      "Description: 240--window--1\n",
      "Number of features: 0\n",
      "Seq('GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGGGACC')\n",
      "ID: 95--window--2\n",
      "Name: 95--window--2\n",
      "Description: 95--window--2\n",
      "Number of features: 0\n",
      "Seq('CTGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAG')\n",
      "ID: 28--window--3\n",
      "Name: 28--window--3\n",
      "Description: 28--window--3\n",
      "Number of features: 0\n",
      "Seq('TATTACTGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGG')\n"
     ]
    }
   ],
   "source": [
    "for i, x in enumerate(SeqIO.parse(\"tmp/vidjil/result.windows.fa\", \"fasta\")):\n",
    "    if i < 3:\n",
    "        print(x)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Same two commands as above, but in one run"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p16 - \n",
      "# 2023-10-02 09:56:44\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p16.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.02%\n",
      "  ==> junction detected in 23 reads (0.0188%)\n",
      "  ==> found 9 windows in 23 reads (0.0188% of 122066 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        1    76.0          1   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       21    75.7          7   0.333\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        1    76.0          1   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        0       -          0       -\n",
      "\n",
      "  SEG               ->       23    75.7\n",
      "  SEG_+             ->       11    75.9\n",
      "  SEG_-             ->       12    75.5\n",
      "  SEG changed w     ->        5    75.2\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23070    75.6\n",
      "  UNSEG too few V/J ->    89989    75.5\n",
      "  UNSEG only V/5'   ->     7969    75.6\n",
      "  UNSEG only J/3'   ->     1015    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p16.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p13 - \n",
      "# 2023-10-02 09:56:44\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p13.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 38 reads (0.0311%)\n",
      "  ==> found 18 windows in 38 reads (0.0311% of 122071 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        1    76.0          1   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        2    76.0          2   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       32    75.5         12   0.375\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        1    76.0          1   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        2    75.5          2   1.000\n",
      "\n",
      "  SEG               ->       38    75.6\n",
      "  SEG_+             ->       14    75.6\n",
      "  SEG_-             ->       24    75.5\n",
      "  SEG changed w     ->       19    75.6\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23195    75.6\n",
      "  UNSEG too few V/J ->    89931    75.5\n",
      "  UNSEG only V/5'   ->     7926    75.5\n",
      "  UNSEG only J/3'   ->      981    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p13.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p9 - \n",
      "# 2023-10-02 09:56:44\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p9.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.02%\n",
      "  ==> junction detected in 24 reads (0.0197%)\n",
      "  ==> found 13 windows in 24 reads (0.0197% of 122064 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        1    76.0          1   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        3    75.7          3   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       20    75.4          9   0.450\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        0       -          0       -\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        0       -          0       -\n",
      "\n",
      "  SEG               ->       24    75.5\n",
      "  SEG_+             ->       11    75.2\n",
      "  SEG_-             ->       13    75.7\n",
      "  SEG changed w     ->        8    75.4\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23137    75.6\n",
      "  UNSEG too few V/J ->    89938    75.5\n",
      "  UNSEG only V/5'   ->     7966    75.6\n",
      "  UNSEG only J/3'   ->      999    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p9.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p3 - \n",
      "# 2023-10-02 09:56:44\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p3.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.05%\n",
      "  ==> junction detected in 57 reads (0.0467%)\n",
      "  ==> found 22 windows in 57 reads (0.0467% of 122062 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        3    76.0          3   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       48    75.6         13   0.271\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        1    76.0          1   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        2    75.5          2   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        3    75.3          3   1.000\n",
      "\n",
      "  SEG               ->       57    75.6\n",
      "  SEG_+             ->       20    75.7\n",
      "  SEG_-             ->       37    75.5\n",
      "  SEG changed w     ->       23    75.8\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23045    75.6\n",
      "  UNSEG too few V/J ->    89899    75.5\n",
      "  UNSEG only V/5'   ->     8030    75.6\n",
      "  UNSEG only J/3'   ->     1031    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p3.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p15 - \n",
      "# 2023-10-02 09:56:44\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p15.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 33 reads (0.027%)\n",
      "  ==> found 18 windows in 33 reads (0.027% of 122063 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        1    76.0          1   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       26    75.7         11   0.423\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        0       -          0       -\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    76.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        5    76.0          5   1.000\n",
      "\n",
      "  SEG               ->       33    75.7\n",
      "  SEG_+             ->       13    75.5\n",
      "  SEG_-             ->       20    75.8\n",
      "  SEG changed w     ->       11    75.7\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23338    75.6\n",
      "  UNSEG too few V/J ->    89818    75.5\n",
      "  UNSEG only V/5'   ->     7834    75.6\n",
      "  UNSEG only J/3'   ->     1040    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p15.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p1 - \n",
      "# 2023-10-02 09:56:44\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p1.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 35 reads (0.0287%)\n",
      "  ==> found 14 windows in 35 reads (0.0287% of 122063 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        1    76.0          1   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       28    75.6          8   0.286\n",
      "  TRA               ->        2    74.0          1   0.500\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        0       -          0       -\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        2    75.5          2   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        2    76.0          2   1.000\n",
      "\n",
      "  SEG               ->       35    75.5\n",
      "  SEG_+             ->       20    75.7\n",
      "  SEG_-             ->       15    75.3\n",
      "  SEG changed w     ->       15    75.4\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23169    75.6\n",
      "  UNSEG too few V/J ->    89909    75.5\n",
      "  UNSEG only V/5'   ->     7908    75.6\n",
      "  UNSEG only J/3'   ->     1042    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p1.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p14 - \n",
      "# 2023-10-02 09:56:44\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p14.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.04%\n",
      "  ==> junction detected in 44 reads (0.0357%)\n",
      "  ==> found 19 windows in 44 reads (0.0357% of 123271 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        1    74.0          1   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        1    76.0          1   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       36    75.6         13   0.361\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        4    75.0          2   0.500\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    76.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        1    76.0          1   1.000\n",
      "\n",
      "  SEG               ->       44    75.5\n",
      "  SEG_+             ->       17    75.5\n",
      "  SEG_-             ->       27    75.5\n",
      "  SEG changed w     ->       17    75.6\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23293    75.6\n",
      "  UNSEG too few V/J ->    90820    75.5\n",
      "  UNSEG only V/5'   ->     8045    75.6\n",
      "  UNSEG only J/3'   ->     1069    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p14.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p2 - \n",
      "# 2023-10-02 09:56:44\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p2.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 41 reads (0.0322%)\n",
      "  ==> found 14 windows in 41 reads (0.0322% of 127145 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        1    76.0          1   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       34    75.4          8   0.235\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        1    76.0          1   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    76.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        4    75.2          3   0.750\n",
      "\n",
      "  SEG               ->       41    75.4\n",
      "  SEG_+             ->       21    75.2\n",
      "  SEG_-             ->       20    75.7\n",
      "  SEG changed w     ->       11    75.5\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24210    75.6\n",
      "  UNSEG too few V/J ->    93457    75.5\n",
      "  UNSEG only V/5'   ->     8367    75.6\n",
      "  UNSEG only J/3'   ->     1070    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p2.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p6 - \n",
      "# 2023-10-02 09:56:44\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p6.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 43 reads (0.0338%)\n",
      "  ==> found 21 windows in 43 reads (0.0338% of 127152 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        1    75.0          1   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       32    75.4         12   0.375\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        1    76.0          1   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    76.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        8    75.9          6   0.750\n",
      "\n",
      "  SEG               ->       43    75.5\n",
      "  SEG_+             ->       21    75.4\n",
      "  SEG_-             ->       22    75.6\n",
      "  SEG changed w     ->       15    75.5\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24244    75.6\n",
      "  UNSEG too few V/J ->    93571    75.5\n",
      "  UNSEG only V/5'   ->     8179    75.6\n",
      "  UNSEG only J/3'   ->     1115    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p6.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p4 - \n",
      "# 2023-10-02 09:56:44\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p4.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 42 reads (0.033%)\n",
      "  ==> found 20 windows in 42 reads (0.033% of 127147 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        1    74.0          1   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        4    75.8          4   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       33    75.4         13   0.394\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        2    75.0          1   0.500\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        2    76.0          1   0.500\n",
      "\n",
      "  SEG               ->       42    75.4\n",
      "  SEG_+             ->       15    75.3\n",
      "  SEG_-             ->       27    75.5\n",
      "  SEG changed w     ->       12    75.6\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24247    75.6\n",
      "  UNSEG too few V/J ->    93610    75.5\n",
      "  UNSEG only V/5'   ->     8200    75.6\n",
      "  UNSEG only J/3'   ->     1047    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        1    76.0\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p4.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p5 - \n",
      "# 2023-10-02 09:56:44\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p5.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 43 reads (0.0338%)\n",
      "  ==> found 13 windows in 43 reads (0.0338% of 127150 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        1    76.0          1   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        2    76.0          2   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       37    75.5          7   0.189\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        0       -          0       -\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        3    75.7          3   1.000\n",
      "\n",
      "  SEG               ->       43    75.6\n",
      "  SEG_+             ->       14    75.2\n",
      "  SEG_-             ->       29    75.7\n",
      "  SEG changed w     ->       13    75.7\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24310    75.6\n",
      "  UNSEG too few V/J ->    93527    75.5\n",
      "  UNSEG only V/5'   ->     8180    75.5\n",
      "  UNSEG only J/3'   ->     1090    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p5.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p10 - \n",
      "# 2023-10-02 09:56:44\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p10.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 37 reads (0.0291%)\n",
      "  ==> found 20 windows in 37 reads (0.0291% of 127149 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        3    75.7          3   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       27    75.3         12   0.444\n",
      "  TRA               ->        2    74.0          1   0.500\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        2    75.5          1   0.500\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        2    75.5          2   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        1    76.0          1   1.000\n",
      "\n",
      "  SEG               ->       37    75.3\n",
      "  SEG_+             ->       24    75.1\n",
      "  SEG_-             ->       13    75.6\n",
      "  SEG changed w     ->        9    75.1\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24087    75.6\n",
      "  UNSEG too few V/J ->    93791    75.5\n",
      "  UNSEG only V/5'   ->     8167    75.6\n",
      "  UNSEG only J/3'   ->     1067    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p10.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p11 - \n",
      "# 2023-10-02 09:56:44\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p11.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 43 reads (0.0338%)\n",
      "  ==> found 23 windows in 43 reads (0.0338% of 127147 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        1    76.0          1   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        2    76.0          2   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       36    75.5         17   0.472\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        3    75.7          2   0.667\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        1    75.0          1   1.000\n",
      "\n",
      "  SEG               ->       43    75.6\n",
      "  SEG_+             ->       19    75.4\n",
      "  SEG_-             ->       24    75.7\n",
      "  SEG changed w     ->        8    75.5\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24031    75.6\n",
      "  UNSEG too few V/J ->    93680    75.5\n",
      "  UNSEG only V/5'   ->     8304    75.6\n",
      "  UNSEG only J/3'   ->     1089    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p11.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p12 - \n",
      "# 2023-10-02 09:56:44\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p12.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.02%\n",
      "  ==> junction detected in 27 reads (0.0212%)\n",
      "  ==> found 14 windows in 27 reads (0.0212% of 127148 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        2    74.0          2   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       21    75.5          8   0.381\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        1    76.0          1   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    74.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        2    75.0          2   1.000\n",
      "\n",
      "  SEG               ->       27    75.3\n",
      "  SEG_+             ->       12    75.2\n",
      "  SEG_-             ->       15    75.5\n",
      "  SEG changed w     ->        7    75.1\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23817    75.6\n",
      "  UNSEG too few V/J ->    93910    75.5\n",
      "  UNSEG only V/5'   ->     8314    75.6\n",
      "  UNSEG only J/3'   ->     1080    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p12.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p8 - \n",
      "# 2023-10-02 09:56:44\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p8.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.04%\n",
      "  ==> junction detected in 54 reads (0.0425%)\n",
      "  ==> found 25 windows in 54 reads (0.0425% of 127147 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        2    75.0          2   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        5    75.0          5   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       44    75.6         15   0.341\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        0       -          0       -\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        2    75.5          2   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        1    72.0          1   1.000\n",
      "\n",
      "  SEG               ->       54    75.5\n",
      "  SEG_+             ->       28    75.6\n",
      "  SEG_-             ->       26    75.4\n",
      "  SEG changed w     ->       19    75.4\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23771    75.6\n",
      "  UNSEG too few V/J ->    93970    75.5\n",
      "  UNSEG only V/5'   ->     8269    75.6\n",
      "  UNSEG only J/3'   ->     1083    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p8.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p7 - \n",
      "# 2023-10-02 09:56:44\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p7.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 35 reads (0.0275%)\n",
      "  ==> found 22 windows in 35 reads (0.0275% of 127155 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        1    76.0          1   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        0       -          0       -\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       27    75.7         14   0.519\n",
      "  TRA               ->        1    76.0          1   1.000\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        3    76.0          3   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    75.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        2    75.5          2   1.000\n",
      "\n",
      "  SEG               ->       35    75.7\n",
      "  SEG_+             ->       22    75.6\n",
      "  SEG_-             ->       13    75.8\n",
      "  SEG changed w     ->       14    75.8\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24352    75.5\n",
      "  UNSEG too few V/J ->    93486    75.5\n",
      "  UNSEG only V/5'   ->     8201    75.5\n",
      "  UNSEG only J/3'   ->     1081    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p7.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations and analyzing clones\n",
      "Command line: /Users/mikesh/vcs/pyigmap/igmap/external/bin/vidjil-algo_darwin -g /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens.g -U -c clones --all -o tmp/vidjil -b result - \n",
      "# 2023-10-02 09:57:27\n",
      "\n",
      "* WARNING: vidjil-algo was run with '--all' option or with a large '--max-designations/-z' option\n",
      "* vidjil-algo efficiently extracts windows overlapping the CDR3\n",
      "* to cluster reads into clones ('-c clones').\n",
      "* Computing accurate V(D)J designations for many sequences ('-c designations' or large '-z' values)\n",
      "* is not as efficient as the default '-c clones' command.\n",
      "* More information is provided in doc/vidjil-algo.md.\n",
      "\n",
      "Load germlines and build Kmer indexes\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /Users/mikesh/vcs/pyigmap/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//result.detected.vdj.fa\n",
      "\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "  ==> junction detected in 612 reads (98.9%)\n",
      "  ==> found 192 windows in 612 reads (98.9% of 619 reads)\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        8    75.2          5   0.625\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->       31    75.6         28   0.903\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->      497    75.5        113   0.227\n",
      "  TRA               ->        5    74.4          1   0.200\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->       20    75.6          6   0.300\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->       14    75.5         13   0.929\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->       37    75.6         26   0.703\n",
      "\n",
      "  SEG               ->      612    75.5\n",
      "  SEG_+             ->      612    75.5\n",
      "  SEG_-             ->        0       -\n",
      "  SEG changed w     ->      202    75.5\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->        0       -\n",
      "  UNSEG too few V/J ->        1    76.0\n",
      "  UNSEG only V/5'   ->        6    75.0\n",
      "  UNSEG only J/3'   ->        0       -\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "Sort windows by number of occurrences\n",
      "  ==> tmp/vidjil//result.windows.fa\n",
      "\n",
      "Diversity measures      IGH   IGK   IGL   TRA   TRB   TRD   TRG   all   \n",
      "  index_H_entropy       1.560 3.300 2.423 0.000 1.228 2.540 3.086 3.224 \n",
      "  index_E_equitability  0.750 0.961 0.390 0.000 0.410 0.962 0.855 0.502 \n",
      "  index_Ds_diversity    0.893 0.994 0.728 0.000 0.621 0.989 0.970 0.820 \n",
      "Considering labeled windows and windows with >= 1 reads and with a ratio >= 0.000 (0)\n",
      "  ==> keep 192 windows in 612 reads (98.869%)  \n",
      "  ==> 192 clones\n",
      "No clustering\n",
      "\n",
      "Detailed analysis of all clones\n",
      "  ==> suggested edges in tmp/vidjil//result.edges\n",
      "#### Clone #001 –     240 reads – 39.2%  \n",
      ">clone-001--IGL--0000240--39.2%--window \n",
      "GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGGGACC\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ3*02 IGLJ3*01 IGLJ2*01\n",
      ">clone-001--IGL--0000240--39.2%--ST-E00107:437:h2gtvbbxx:8:1202:1759:1134-[0,75]-#1 - 76 bp (100% of 75.5 bp) + VJ\t 1 43 47 76\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 5.21e-12 7.32e-14/5.14e-12\n",
      "GGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAGGGACCAAGGTGACC\n",
      "\n",
      "\n",
      "#### Clone #002 –      95 reads – 15.5%  \n",
      ">clone-002--IGL--0000095--15.5%--window \n",
      "CTGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAG\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ3*02 IGLJ3*01 IGLJ2*01\n",
      ">clone-002--IGL--0000095--15.5%--ST-E00107:437:h2gtvbbxx:8:1127:10449:1316-[0,75]-#15 - 76 bp (100% of 75.7 bp) + VJ\t 1 56 60 76\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 1.08e-05 1.09e-21/1.08e-05\n",
      "AGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAGG\n",
      "\n",
      "\n",
      "#### Clone #003 –      28 reads – 4.58%  \n",
      ">clone-003--IGL--0000028--4.58%--window \n",
      "TATTACTGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGG\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ3*02 IGLJ3*01 IGLJ2*01\n",
      ">clone-003--IGL--0000028--4.58%--ST-E00107:437:h2gtvbbxx:8:1128:9630:1088-[0,74]-#3 - 75 bp (100% of 75.0 bp) + VJ\t 1 58 62 75\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 6.81e-04 6.73e-23/6.81e-04\n",
      "CCAGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGG\n",
      "\n",
      "\n",
      "#### Clone #004 –      12 reads – 1.96%  \n",
      ">clone-004--TRB--0000012--1.96%--window \n",
      "TCTGCAGTGCTAGAGAGTCGACTAGCGATCCAAAAAATGAGCAGTTCTTC\n",
      "[warning] (W69) Several genes with equal probability: TRBV20/OR9-2*01 TRBV20-1*01 TRBV20/OR9-2*03\n",
      ">clone-004--TRB--0000012--1.96%--ST-E00107:437:h2gtvbbxx:8:1201:10955:1318-[0,75]-#3 - 76 bp (100% of 75.6 bp) + VDJ\t 1 33 37 44 53 76\tTRBV20/OR9-2*01 0/GTC/2 TRBD2*01 6/ATCCAAAA/7 TRBJ2-1*01  TRB SEG_+ 2.68e-10 2.43e-15/2.68e-10 {20(48)67 p CSARESTSDPKNEQFF}\n",
      "AGACAGCAGCTTCTACATCTGCAGTGCTAGAGA\n",
      "GTCGACTAGCGATCCAAAA\n",
      "AATGAGCAGTTCTTCGGGCCAGGG\n",
      "\n",
      "\n",
      "#### Clone #005 –       6 reads – 0.98%  \n",
      ">clone-005--IGL--0000006--0.98%--window \n",
      "ATGGTAGGCAGCCCTCCCTCTCCATCCCCTCACCCCACCCCTTAGCCACA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-005--IGL--0000006--0.98%--ST-E00107:437:h2gtvbbxx:8:1202:3799:1263-[2,74]-#1 - 73 bp (96% of 75.3 bp) \t ! Possibly IGL @34  @8  IGL UNSEG only J/3' 1.42e+01 1.40e+01/1.70e-01\n",
      "CCCAGACCATGGTAGGCAGCCCTCCCTCTCCATCCCCTCACCCCACCCCTTAGCCACAGTGAAGGGAATGGAA\n",
      "\n",
      "\n",
      "#### Clone #006 –       5 reads – 0.817%  \n",
      ">clone-006--TRA--0000005--0.817%--window \n",
      "TGCGAAGAAGCCACTTGCAGTGACACCAGTGAGAGCATTTTGGAAGAGGA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-006--TRA--0000005--0.817%--ST-E00107:437:h2gtvbbxx:8:1202:9441:1163-[0,73]-#2 - 74 bp (99% of 74.4 bp) \t ! Possibly TRA @55  @37  TRA UNSEG only J/3' 3.07e+00 3.03e+00/4.29e-02\n",
      "GGAGTTTTGAGGAGTATCAGCTGCGAAGAAGCCACTTGCAGTGACACCAGTGAGAGCATTTTGGAAGAGGAACC\n",
      "\n",
      "\n",
      "#### Clone #007 –       4 reads – 0.654%  \n",
      ">clone-007--TRG--0000004--0.654%--window \n",
      "TATCAGAAAGAAGACATTACTATTTATTACGGCAATTCAAGATATTTTCT\n",
      ">clone-007--TRG--0000004--0.654%--ST-E00107:437:h2gtvbbxx:8:1127:9863:1321-[0,75]-#1 - 76 bp (100% of 76.0 bp) + VJ\t 13 38 44 60\tTRGV10*02 18/GGCAA/14 TRGJP1*01  TRG SEG_+ 1.06e-01 1.05e-01/6.27e-04\n",
      "GGAATGGCTATCAGAAAGAAGACATTACTATTTATTAC\n",
      "GGCAA\n",
      "TTCAAGATATTTTCTGATTTCCGGAAGTCAGGG\n",
      "\n",
      "\n",
      "#### Clone #008 –       4 reads – 0.654%  \n",
      ">clone-008--TRB--0000004--0.654%--window \n",
      "CTTCTGTGCCATCAGTGAGCCGACAGGGATCAGAAGAGACCCAGTACTTC\n",
      "[warning] (W69) Several genes with equal probability: TRBV10-3*01 TRBV10-3*02\n",
      ">clone-008--TRB--0000004--0.654%--ST-E00107:437:h2gtvbbxx:8:1202:9770:1109-[0,75]-#1 - 76 bp (100% of 75.5 bp) + VDJ\t 1 24 27 33 39 76\tTRBV10-3*01 2/CC/2 TRBD1*01 3/ATCAG/3 TRBJ2-5*01  TRB SEG_+ 1.28e-09 1.28e-09/2.50e-19 {10(46)55 u CAISEPTGIR#ETQYF}\n",
      "GTGTACTTCTGTGCCATCAGTGAG\n",
      "CCGACAGGGATCAG\n",
      "AAGAGACCCAGTACTTCGGGCCAGGCACGCGGCTCCTG\n",
      "\n",
      "\n",
      "#### Clone #009 –       4 reads – 0.654%  \n",
      ">clone-009--TRG--0000004--0.654%--window \n",
      "CAGATTGTGACTCCTGGAGATTTGGATATATGTGAATATAATTTTAGAAG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-009--TRG--0000004--0.654%--ST-E00107:437:h2gtvbbxx:8:1202:10862:1259-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly TRG @30  @49  TRG UNSEG only J/3' 3.29e+00 2.38e+00/9.08e-01\n",
      "ATAGAGCAACAGATTGTGACTCCTGGAGATTTGGATATATGTGAATATAATTTTAGAAGCTTATCATTTACCTATG\n",
      "\n",
      "\n",
      "#### Clone #010 –       4 reads – 0.654%  \n",
      ">clone-010--TRG--0000004--0.654%--window \n",
      "AGAAAAGCAATAATTTTACAGGCATTTTGAGGTGTCTCTTTCTTTGGGTT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-010--TRG--0000004--0.654%--ST-E00107:437:h2gtvbbxx:8:1128:6463:1279-[0,74]-#2 - 75 bp (99% of 75.2 bp) \t ! Possibly TRG @43  @18  TRG UNSEG too few V/J 1.02e+01 6.64e+00/3.58e+00\n",
      "CTTAGAGAAAGAAAAGCAATAATTTTACAGGCATTTTGAGGTGTCTCTTTCTTTGGGTTCTTTCTGTTTGAAAGG\n",
      "\n",
      "\n",
      "#### Clone #011 –       4 reads – 0.654%  \n",
      ">clone-011--IGL--0000004--0.654%--window \n",
      "AAGCCCTCAGCACCACTTTGAACAGAATTCCATCATCCCTGGCCCTCCCT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-011--IGL--0000004--0.654%--ST-E00107:437:h2gtvbbxx:8:1202:8645:1182-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @34  @23  IGL UNSEG only V/5' 4.04e+00 4.03e-02/4.00e+00\n",
      "CAGGAACAGACTTGGGAACAAACACAAGCCCTCAGCACCACTTTGAACAGAATTCCATCATCCCTGGCCCTCCCTT\n",
      "\n",
      "\n",
      "#### Clone #012 –       3 reads – 0.49%  \n",
      ">clone-012--IGL--0000003--0.49%--window \n",
      "GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAAGGACC\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ2*01 IGLJ3*01 IGLJ3*02\n",
      ">clone-012--IGL--0000003--0.49%--ST-E00107:437:h2gtvbbxx:8:1128:7193:1162-[0,74]-#1 - 75 bp (100% of 75.0 bp) + VJ\t 1 51 55 75\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 1.33e-06 1.10e-18/1.33e-06\n",
      "GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAAGGACC\n",
      "\n",
      "\n",
      "#### Clone #013 –       3 reads – 0.49%  \n",
      ">clone-013--IGL--0000003--0.49%--window \n",
      "CCCGTGCTGAAGAGCAGCCTGGCCTTGGGCCTGGAGCTGCGGGCCGCAGC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-013--IGL--0000003--0.49%--ST-E00107:437:h2gtvbbxx:8:1201:1772:1317-[0,75]-#2 - 76 bp (100% of 75.3 bp) \t ! Possibly IGL @25  @29  IGL UNSEG too few V/J 9.15e+00 5.15e+00/4.00e+00\n",
      "GATGCCCGTGCTGAAGAGCAGCCTGGCCTTGGGCCTGGAGCTGCGGGCCGCAGCCGGGAGCCACTTTGATGCTGCG\n",
      "\n",
      "\n",
      "#### Clone #014 –       3 reads – 0.49%  \n",
      ">clone-014--IGL--0000003--0.49%--window \n",
      "ACCCTCAGTGCCAGCCAGCCTCTGCCAGACCTCTCTTTCCCTCTTCTCCC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-014--IGL--0000003--0.49%--ST-E00107:437:h2gtvbbxx:8:1201:3263:1063-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @35  @1  IGL UNSEG only V/5' 4.00e+00 9.83e-06/4.00e+00\n",
      "GCACAGCTCCAGCTCGGACAGCACCCTCAGTGCCAGCCAGCCTCTGCCAGACCTCTCTTTCCCTCTTCTCCCCAAC\n",
      "\n",
      "\n",
      "#### Clone #015 –       2 reads – 0.327%  \n",
      ">clone-015--IGK--0000002--0.327%--window \n",
      "TTCCCTGTGACCCAGCCTGTTCCATAGCCAAAGGGACCAAAGTGAACGAT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-015--IGK--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:2714:1052-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGK @31  @52  IGK UNSEG only J/3' 1.99e+01 1.98e+01/9.07e-03\n",
      "CTCCAGCCTCCAGCCTCACTGGCAGTTCCCTGTGACCCAGCCTGTTCCATAGCCAAAGGGACCAAAGTGAACGATC\n",
      "\n",
      "\n",
      "#### Clone #016 –       2 reads – 0.327%  \n",
      ">clone-016--IGK--0000002--0.327%--window \n",
      "TTACTTATCAAATGGATTATCTCCCCCTCTTTAAAAGTAAGTTCATCTTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-016--IGK--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:3099:1077-[0,75]-#1 - 76 bp (100% of 75.5 bp) \t ! Possibly IGK @51  @67  IGK UNSEG too few V/J 5.04e+00 1.75e+00/3.28e+00\n",
      "CACCAGCCAGCTTCTCCAGTCTCCTTACTTATCAAATGGATTATCTCCCCCTCTTTAAAAGTAAGTTCATCTTCAT\n",
      "\n",
      "\n",
      "#### Clone #017 –       2 reads – 0.327%  \n",
      ">clone-017--IGL--0000002--0.327%--window \n",
      "TGTGCCTGGACGGGAAGATGGCAAGGAGGCACCGGAGGAGGCACAGCTCG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-017--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:2863:1247-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @50  @59  IGL UNSEG only J/3' 1.46e+01 1.46e+01/5.52e-03\n",
      "CAGGCCCAGATCAAGGGGCCCGTGGATGTGCCTGGACGGGAAGATGGCAAGGAGGCACCGGAGGAGGCACAGCTCG\n",
      "\n",
      "\n",
      "#### Clone #018 –       2 reads – 0.327%  \n",
      ">clone-018--IGL--0000002--0.327%--window \n",
      "TGGGGGACACAGGCACGCTGCTTTCCCTCCCAAGCCAGTGGGCCCCCACC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-018--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:10264:1397-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @45  @4  IGL UNSEG too few V/J 5.28e+00 1.29e+00/4.00e+00\n",
      "CTCCTGGGGGACACAGGCACGCTGCTTTCCCTCCCAAGCCAGTGGGCCCCCACCCGACTCCCCACTCTGCAGCTTC\n",
      "\n",
      "\n",
      "#### Clone #019 –       2 reads – 0.327%  \n",
      ">clone-019--TRG--0000002--0.327%--window \n",
      "TGAAAATATCTATTGAAAATGAGCAACTTGTGATTGGATCATATAGTCAG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-019--TRG--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:4510:1071-[0,75]-#1 - 76 bp (102% of 74.0 bp) \t ! Possibly TRG @36  @44  TRG UNSEG only J/3' 2.38e+00 2.38e+00/1.25e-03\n",
      "AAAGTACAGACTTCTGAAAATATCTATTGAAAATGAGCAACTTGTGATTGGATCATATAGTCAGCCTTCAGATTCC\n",
      "\n",
      "\n",
      "#### Clone #020 –       2 reads – 0.327%  \n",
      ">clone-020--TRD--0000002--0.327%--window \n",
      "TCTCAGAAAGCCTTTCTGACTGATCTCAGCATTACCTCTTTGGAAAAGGA\n",
      ">clone-020--TRD--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:7854:1109-[0,74]-#1 - 75 bp (100% of 75.0 bp) + VJ\t 46 60 61 72\tTRAV14/DV4*04 0//18 TRDJ1*01  TRD SEG_+ 8.04e-01 8.04e-01/2.86e-04\n",
      "CCCAGAAATCCAAGCCAGTTTTCTCAGAAAGCCTTTCTGACTGATCTCAGCATTACCTCT\n",
      "\n",
      "TTGGAAAAGGAAGGT\n",
      "\n",
      "\n",
      "#### Clone #021 –       2 reads – 0.327%  \n",
      ">clone-021--IGL--0000002--0.327%--window \n",
      "TACTGCAGTTCATATAGAGGCACCGCCACTTTCGAGGTGGTGTTCGGCGG\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ3*02 IGLJ3*01 IGLJ2*01\n",
      ">clone-021--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1201:9019:1421-[0,75]-#1 - 76 bp (100% of 76.0 bp) + VJ\t 1 40 44 76\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 1.50e-10 1.50e-10/8.03e-14\n",
      "TGATTATTACTGCAGTTCATATAGAGGCACCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAGGGACCAAGGTGACCGTC\n",
      "\n",
      "\n",
      "#### Clone #022 –       2 reads – 0.327%  \n",
      ">clone-022--IGL--0000002--0.327%--window \n",
      "GTTTTTCTGCCTCAGCCTCCTGAGTAGCTAAGGGAGGTGTCTTGAGATTA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-022--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:6128:1156-[0,75]-#1 - 76 bp (100% of 75.5 bp) \t ! Possibly IGL @31  @59  IGL UNSEG too few V/J 1.14e+00 6.44e-01/5.00e-01\n",
      "GCGTGGTTTTTCTGCCTCAGCCTCCTGAGTAGCTAAGGGAGGTGTCTTGAGATTATCATCGGCTGAGGGTGGAAGC\n",
      "\n",
      "\n",
      "#### Clone #023 –       2 reads – 0.327%  \n",
      ">clone-023--IGL--0000002--0.327%--window \n",
      "GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGGAACC\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ2*01 IGLJ3*01 IGLJ3*02\n",
      ">clone-023--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:10796:1236-[0,74]-#1 - 75 bp (100% of 75.0 bp) + VJ\t 1 51 55 75\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 1.33e-06 1.10e-18/1.33e-06\n",
      "GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAGGAACC\n",
      "\n",
      "\n",
      "#### Clone #024 –       2 reads – 0.327%  \n",
      ">clone-024--IGL--0000002--0.327%--window \n",
      "GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGAGACC\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ2*01 IGLJ3*01 IGLJ3*02\n",
      ">clone-024--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:4142:1107-[0,74]-#1 - 75 bp (100% of 75.0 bp) + VJ\t 1 51 55 75\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 1.33e-06 1.10e-18/1.33e-06\n",
      "GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAGAGACC\n",
      "\n",
      "\n",
      "#### Clone #025 –       2 reads – 0.327%  \n",
      ">clone-025--IGL--0000002--0.327%--window \n",
      "GATCTGCCTATTACTGCTGTGCCCCATGGGGGGCTCCTTCCCTGCCTGAC\n",
      "[warning] (W69) Several genes with equal probability: IGLV10-67*01 IGLV10-67*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ7*02 IGLJ7*01\n",
      ">clone-025--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:10458:1104-[0,75]-#1 - 76 bp (100% of 76.0 bp) + VJ\t 30 44 48 47\tIGLV10-67*01 28/GTG/46 IGLJ7*01  IGL SEG_+ 7.05e-01 4.55e-01/2.50e-01\n",
      "ATGTGCTCTGAGCTTCCCTGAACCAGGATCTGCCTATTACTGCT\n",
      "GTG\n",
      "CCCCATGGGGGGCTCCTTCCCTGCCTGAC\n",
      "\n",
      "\n",
      "#### Clone #026 –       2 reads – 0.327%  \n",
      ">clone-026--TRG--0000002--0.327%--window \n",
      "GAGCAGCAGGAGGGGGGAGGGAAGGGAGCCAAGGGATGGACATCTTGTCA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-026--TRG--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:7743:1161-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly TRG @41  @32  TRG UNSEG too few V/J 7.51e+00 2.38e+00/5.14e+00\n",
      "AGGAGCAGCAGGAGGGGGGAGGGAAGGGAGCCAAGGGATGGACATCTTGTCATTTTTTTTCTTTGAATAAATGTCA\n",
      "\n",
      "\n",
      "#### Clone #027 –       2 reads – 0.327%  \n",
      ">clone-027--IGL--0000002--0.327%--window \n",
      "CCTCTGCCACCAGCCTCTTCTCCAACAGACTCTCCTTCCCCTTTTCCAGG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-027--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:8201:1155-[0,75]-#1 - 76 bp (101% of 75.0 bp) \t ! Possibly IGL @40  @9  IGL UNSEG too few V/J 9.30e+00 3.64e+00/5.65e+00\n",
      "CCACCTCCTCTTCGTCCTCTTCCTCCTCTGCCACCAGCCTCTTCTCCAACAGACTCTCCTTCCCCTTTTCCAGGTC\n",
      "\n",
      "\n",
      "#### Clone #028 –       2 reads – 0.327%  \n",
      ">clone-028--IGH--0000002--0.327%--window \n",
      "CAGGAAGGGGCTGGACTGGCTGTCCTTTCCAGTGGCCTGGCTCCGCTGTG\n",
      "[warning] (W69) Several genes with equal probability: IGHJ1P*02 IGHJ1P*01\n",
      ">clone-028--IGH--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:9291:1085-[0,75]-#1 - 76 bp (101% of 75.0 bp) + VDJ\t 6 21 22 27 28 22\tIGHV(II)-46-1*01 8//15 IGHD3/OR15-3a*01 10//34 IGHJ1P*01  IGH SEG_+ 4.11e-01 7.08e-02/3.40e-01\n",
      "CCTCACTCCAGGAAGGGGCTG\n",
      "GACTGG\n",
      "CTGTCCTTTCCAGTGGCCTGGCTCCGCTGTGTGGATGGGGAGATCGGGG\n",
      "\n",
      "\n",
      "#### Clone #029 –       2 reads – 0.327%  \n",
      ">clone-029--IGK--0000002--0.327%--window \n",
      "CAAGCCCCGGCTGAGCCTCAAGGGAATACCCACTCTGGTGTACAGACACG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-029--IGK--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:6890:1291-[0,75]-#1 - 76 bp (101% of 75.0 bp) \t ! Possibly IGK @56  @52  IGK UNSEG only J/3' 1.00e+01 9.92e+00/1.03e-01\n",
      "TCCAGGGCAAGCCCCGGCTGAGCCTCAAGGGAATACCCACTCTGGTGTACAGACACGACTGGAACTATGAACATCT\n",
      "\n",
      "\n",
      "#### Clone #030 –       2 reads – 0.327%  \n",
      ">clone-030--IGH--0000002--0.327%--window \n",
      "ATTTGCTCCCCTGAGTGTGTAGGTCCCTGAGACCTGCTCACCAGCCCCTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-030--IGH--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1201:10903:1363-[0,75]-#2 - 76 bp (101% of 75.0 bp) \t ! Possibly IGH @61  @40  IGH UNSEG too few V/J 1.29e+01 9.07e+00/3.85e+00\n",
      "GCTCTGGGCCATCCCAGGAGGGGGAATTTGCTCCCCTGAGTGTGTAGGTCCCTGAGACCTGCTCACCAGCCCCTCC\n",
      "\n",
      "\n",
      "#### Clone #031 –       2 reads – 0.327%  \n",
      ">clone-031--IGL--0000002--0.327%--window \n",
      "AGGGCCTCCTGGACACCTACAGCGTGTGGAGGAGTGGTCTCCCCAACTTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-031--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:4932:1253-[0,75]-#2 - 76 bp (100% of 75.5 bp) \t ! Possibly IGL @57  @44  IGL UNSEG too few V/J 6.47e+00 3.64e+00/2.83e+00\n",
      "CTTTTCCCCGGGCCTTCCAGGGCCTCCTGGACACCTACAGCGTGTGGAGGAGTGGTCTCCCCAACTTCCTAGCAGT\n",
      "\n",
      "\n",
      "#### Clone #032 –       2 reads – 0.327%  \n",
      ">clone-032--IGL--0000002--0.327%--window \n",
      "ACCATCAGTGTCAGCATTAGGGTTTTGGTTTTTGTTTCTTTTGGGTCTTT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-032--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:7815:1217-[0,75]-#2 - 76 bp (100% of 75.5 bp) \t ! Possibly IGL @21  @36  IGL UNSEG only J/3' 5.40e+00 5.15e+00/2.50e-01\n",
      "ACTGTTCAGGGTACTGACCATCAGTGTCAGCATTAGGGTTTTGGTTTTTGTTTCTTTTGGGTCTTTCTTTTTTGGC\n",
      "\n",
      "\n",
      "#### Clone #033 –       2 reads – 0.327%  \n",
      ">clone-033--IGH--0000002--0.327%--window \n",
      "AATTAGATCCCAGCCTGGACAAGGGCATGACTATTGCTGTTTGGGGACAC\n",
      "[warning] (W69) Several genes with equal probability: IGHV1/OR16-4*02 IGHV1/OR16-4*01 IGHV1/OR16-3*01 IGHV1/OR16-22*01\n",
      ">clone-033--IGH--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1201:9240:1123-[0,75]-#1 - 76 bp (100% of 76.0 bp) + VDJ\t 40 56 57 60 61 75\tIGHV1/OR16-22*01 13//3 IGHD4/OR15-4a*01 12//9 IGHJ3*01  IGH SEG_+ 1.30e-01 1.00e-01/3.01e-02\n",
      "GGAGGGAGTCAACGACCTATTGTATCAATTAGATCCCAGCCTGGACAAGGGCATGA\n",
      "CTAT\n",
      "TGCTGTTTGGGGACAC\n",
      "\n",
      "\n",
      "#### Clone #034 –       1 reads – 0.163%  \n",
      ">clone-034--IGL--0000001--0.163%--window \n",
      "TTTTTTTTTTTTTTCTTCCTTTTCCCTTTCTCGGTTCTCTTCTCTCCTCT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-034--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1202:9858:1155-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @72  @25  IGL UNSEG too few V/J 1.13e+01 7.29e+00/4.00e+00\n",
      "CAATACTAAGCTTTGCTTTTTTTTTTTTTTCTTCCTTTTCCCTTTCTCGGTTCTCTTCTCTCCTCTGAATGGCTGG\n",
      "\n",
      "\n",
      "#### Clone #035 –       1 reads – 0.163%  \n",
      ">clone-035--IGL--0000001--0.163%--window \n",
      "TTTTTTTTTTTTTGAAATGGAGTATTTCTCATTAGGTTTCACTGTCTTAT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-035--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1201:4711:1342-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @74  @62  IGL UNSEG only J/3' 1.10e+01 1.03e+01/7.07e-01\n",
      "CACCTTTTATATGAGTGGATTTTTTTTTTTTTTTTGAAATGGAGTATTTCTCATTAGGTTTCACTGTCTTATCCTT\n",
      "\n",
      "\n",
      "#### Clone #036 –       1 reads – 0.163%  \n",
      ">clone-036--IGL--0000001--0.163%--window \n",
      "TTTTTTTTTTTTTAGCTGAAGGATGGTTAGACAGCACAGCGAAGA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-036--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:2960:1395-[0,73]-#1 - 74 bp (100% of 74.0 bp) \t ! Possibly IGL @68  @7  IGL UNSEG only J/3' 1.05e+01 1.00e+01/4.86e-01\n",
      "TTTTTTTTTTTTTTTAGCTGAAGGATGGTTAGACAGCACAGCGAAGACTGCTAAATGCACTGACCCCCCCCATT\n",
      "\n",
      "\n",
      "#### Clone #037 –       1 reads – 0.163%  \n",
      ">clone-037--IGL--0000001--0.163%--window \n",
      "TTTTTTTTTTATTTTGCCTTTGTTTGTTTTTAACGTTTATCCTCTCCTCT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-037--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1201:6253:1105-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @11  @24  IGL UNSEG only J/3' 2.70e+00 2.58e+00/1.25e-01\n",
      "CTGATACTGATTTTTTTTTTTTTTTTTTTATTTTGCCTTTGTTTGTTTTTAACGTTTATCCTCTCCTCTTTGTGTG\n",
      "\n",
      "\n",
      "#### Clone #038 –       1 reads – 0.163%  \n",
      ">clone-038--IGL--0000001--0.163%--window \n",
      "TTTTTTTTTTATAATGAGTATGATCTATTTCTTTTCAAATAATCTTTGAG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-038--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:10969:1112-[0,74]-#1 - 75 bp (100% of 75.0 bp) \t ! Possibly IGL @37  @17  IGL UNSEG too few V/J 5.57e+00 3.60e+00/1.97e+00\n",
      "ATCTTGTTTTTTTTTTTTTTTTATAATGAGTATGATCTATTTCTTTTCAAATAATCTTTGAGATCCCAGGAAAAA\n",
      "\n",
      "\n",
      "#### Clone #039 –       1 reads – 0.163%  \n",
      ">clone-039--IGK--0000001--0.163%--window \n",
      "TTTCAATGGGATTCAACAGGGCCACCTTTTTCCACTGAAGTATTAGGAGG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-039--IGK--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1127:1340:1396-[0,74]-#1 - 75 bp (100% of 75.0 bp) \t ! Possibly IGK @45  @35  IGK UNSEG too few V/J 2.19e+01 1.96e+01/2.29e+00\n",
      "TCAAGTCTGGGTTTTCAATGGGATTCAACAGGGCCACCTTTTTCCACTGAAGTATTAGGAGGACGATGAAGACGA\n",
      "\n",
      "\n",
      "#### Clone #040 –       1 reads – 0.163%  \n",
      ">clone-040--TRD--0000001--0.163%--window \n",
      "TTTATTTTTTTTATTTTTTTATTTTTTATTTTTGTATATTTCTTTAAATT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-040--TRD--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1202:10947:1050-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly TRD @16  @66  TRD UNSEG only V/5' 6.73e+00 1.80e-02/6.71e+00\n",
      "ATTCTGTGCATGTGACCTTTTTTTTTTATTTTTTTTATTTTTTTATTTTTTATTTTTGTATATTTCTTTAAATTCC\n",
      "\n",
      "\n",
      "#### Clone #041 –       1 reads – 0.163%  \n",
      ">clone-041--IGL--0000001--0.163%--window \n",
      "TTGGGATTTTTTTTTTTTTTTTTGGCTTTTGTTTTTGTTTGTTTTTTTGT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-041--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1202:5423:1130-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @17  @46  IGL UNSEG only J/3' 1.85e+00 1.82e+00/3.12e-02\n",
      "AGTGTATATGGCTTGTGTTTTGGGATTTTTTTTTTTTTTTTTGGCTTTTGTTTTTGTTTGTTTTTTTGTTTCGTTT\n",
      "\n",
      "\n",
      "#### Clone #042 –       1 reads – 0.163%  \n",
      ">clone-042--TRG--0000001--0.163%--window \n",
      "TTGAAAATGCAAAGCGAGTGAGAAAAGAAAGTGAAAAAATCAAGG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-042--TRG--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:7455:1164-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly TRG @26  @65  TRG UNSEG only J/3' 4.81e+00 4.76e+00/5.67e-02\n",
      "AATGAAGTCAACAAAAGCACTGTTTACATTATTGAAAATGCAAAGCGAGTGAGAAAAGAAAGTGAAAAAATCAAGG\n",
      "\n",
      "\n",
      "#### Clone #043 –       1 reads – 0.163%  \n",
      ">clone-043--TRD--0000001--0.163%--window \n",
      "TTCTAAAGAAAGAAGGAGCATAGTACTCATTTATTTACTCCCGCATTCAG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-043--TRD--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:3842:1346-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly TRD @20  @39  TRD UNSEG too few V/J 1.59e+01 9.21e+00/6.71e+00\n",
      "CTTCTAAAGAAAGAAGGAGCATAGTACTCATTTATTTACTCCCGCATTCAGTACTTATTCCAATCACTTCATTAGG\n",
      "\n",
      "\n",
      "#### Clone #044 –       1 reads – 0.163%  \n",
      ">clone-044--IGL--0000001--0.163%--window \n",
      "TTCATATAGAGGCAGCGCCACTTTCGAGGGGGTGTTCGGCGGAGGGACCA\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-18*02 IGLV2-18*03 IGLV2-18*04 IGLV2-28*01\n",
      ">clone-044--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:2070:1410-[0,73]-#1 - 74 bp (100% of 74.0 bp) + VJ\t 1 31 36 73\tIGLV2-18*02 0/GAGG/2 IGLJ3*02  IGL SEG_+ 1.20e-06 1.20e-06/1.08e-16 {2(42)43 p CSSYRGSATFEGVF}\n",
      "CTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAGG\n",
      "GGGTGTTCGGCGGAGGGACCAAGGTGACCGTCCTACGTC\n",
      "\n",
      "\n",
      "#### Clone #045 –       1 reads – 0.163%  \n",
      ">clone-045--TRD--0000001--0.163%--window \n",
      "TTCAGCACCATCCCTGAGCTCATTAACTACCCTCAGCACAACTCTGCAGG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-045--TRD--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1201:7800:1203-[0,74]-#1 - 75 bp (100% of 75.0 bp) \t ! Possibly TRD @62  @34  TRD UNSEG only J/3' 9.09e+00 9.09e+00/8.08e-04\n",
      "TTCAGCACCATCCCTGAGCTCATTAACTACCCTCAGCACAACTCTGCAGGACTCATATCCAGGCTCAAATATCCA\n",
      "\n",
      "\n",
      "#### Clone #046 –       1 reads – 0.163%  \n",
      ">clone-046--IGK--0000001--0.163%--window \n",
      "TTAGATTACACTCATACCATTGTATCATCTTGCTATAAAAAGTTACTTAA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-046--IGK--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:7574:1305-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGK @50  @11  IGK UNSEG too few V/J 3.40e+00 1.75e+00/1.64e+00\n",
      "AAGTTTAGATTACACTCATACCATTGTATCATCTTGCTATAAAAAGTTACTTAAAACTATTTCTTTTGCATATAAA\n",
      "\n",
      "\n",
      "#### Clone #047 –       1 reads – 0.163%  \n",
      ">clone-047--IGL--0000001--0.163%--window \n",
      "TGTTTCTGTTTTTTTTTTTTTTTTGTGCCTCTTTTTCCCCTTCCCTACTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-047--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:9796:1410-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @76  @20  IGL UNSEG too few V/J 1.14e+00 6.44e-01/5.00e-01\n",
      "TGTCGACTGAAACTCACTTTTATGTTTCTGTTTTTTTTTTTTTTTTGTGCCTCTTTTTCCCCTTCCCTACTCATCT\n",
      "\n",
      "\n",
      "#### Clone #048 –       1 reads – 0.163%  \n",
      ">clone-048--IGK--0000001--0.163%--window \n",
      "TGTTATAGTCTGATTATTTCTGTGTTTTGAATATGATTTTCCTAATGCTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-048--IGK--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1202:4873:1086-[0,73]-#1 - 74 bp (100% of 74.0 bp) \t ! Possibly IGK @67  @1  IGK UNSEG only V/5' 6.47e+00 7.55e-02/6.40e+00\n",
      "TGTATACCACTATCTATGCAGCATTGTTATAGTCTGATTATTTCTGTGTTTTGAATATGATTTTCCTAATGCTC\n",
      "\n",
      "\n",
      "#### Clone #049 –       1 reads – 0.163%  \n",
      ">clone-049--TRD--0000001--0.163%--window \n",
      "TGTGGTGTCTGAAGACCAAGACTACACAGCTCATTTTGCTGATCCTGATA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-049--TRD--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1202:3024:1049-[0,74]-#1 - 75 bp (100% of 75.0 bp) \t ! Possibly TRD @71  @62  TRD UNSEG only J/3' 1.32e+01 1.29e+01/2.93e-01\n",
      "TTGTGGTGTCTGAAGACCAAGACTACACAGCTCATTTTGCTGATCCTGATACATTAGTTAACTGGGACTTTGTGG\n",
      "\n",
      "\n",
      "#### Clone #050 –       1 reads – 0.163%  \n",
      ">clone-050--TRG--0000001--0.163%--window \n",
      "TGTAGCAAAACAAAAAGTTTTATAAATTTCAGATAAGCTTTCTGTTTTTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-050--TRG--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1201:2533:1310-[0,74]-#1 - 75 bp (100% of 75.0 bp) \t ! Possibly TRG @59  @39  TRG UNSEG only J/3' 2.13e+02 2.12e+02/8.96e-01\n",
      "TATTCAAATACCTTCTGTTTTCTATGTAGCAAAACAAAAAGTTTTATAAATTTCAGATAAGCTTTCTGTTTTTCT\n",
      "\n",
      "\n",
      ".....\n",
      "\n",
      "#### Clones #510 to #192...\n",
      "#### end of clones\n",
      "\n",
      "Please review the 25 suggested edge(s) in tmp/vidjil//.edges\n",
      "Comparing clone consensus sequences 2 by 2\n",
      "    | 0001 0002 0003 0004 0005 0006 0007 0008 0009 0010 0011 0012 0013 0014 0015 \n",
      "001 | 0100 0070 0065 0054 0045 0053 0051 0053 0046 0046 0047 0079 0052 0043 0053    ST-E00107:437:h2gtvbbxx:8:1202:1759:1134-[0,75]-#1 - 76 bp (100% of 75.5 bp)\n",
      "002 |      0100 0093!0058 0046 0051 0052 0051 0048 0055 0043 0088 0050 0047 0048    ST-E00107:437:h2gtvbbxx:8:1127:10449:1316-[0,75]-#15 - 76 bp (100% of 75.7 bp)\n",
      "003 |           0100 0057 0048 0051 0052 0050 0047 0054 0044 0082 0053 0048 0048    ST-E00107:437:h2gtvbbxx:8:1128:9630:1088-[0,74]-#3 - 75 bp (100% of 75.0 bp)\n",
      "004 |                0100 0045 0054 0052 0057 0055 0049 0048 0058 0047 0051 0049    ST-E00107:437:h2gtvbbxx:8:1201:10955:1318-[0,75]-#3 - 76 bp (100% of 75.6 bp)\n",
      "005 |                     0100 0047 0043 0040 0041 0040 0044 0047 0052 0052 0057    ST-E00107:437:h2gtvbbxx:8:1202:3799:1263-[2,74]-#1 - 73 bp (96% of 75.3 bp)\n",
      "006 |                          0100 0052 0052 0047 0048 0048 0055 0053 0048 0050    ST-E00107:437:h2gtvbbxx:8:1202:9441:1163-[0,73]-#2 - 74 bp (99% of 74.4 bp)\n",
      "007 |                               0100 0048 0052 0055 0051 0053 0045 0048 0040    ST-E00107:437:h2gtvbbxx:8:1127:9863:1321-[0,75]-#1 - 76 bp (100% of 76.0 bp)\n",
      "008 |                                    0100 0048 0040 0051 0052 0053 0048 0047    ST-E00107:437:h2gtvbbxx:8:1202:9770:1109-[0,75]-#1 - 76 bp (100% of 75.5 bp)\n",
      "009 |                                         0100 0051 0049 0045 0044 0043 0047    ST-E00107:437:h2gtvbbxx:8:1202:10862:1259-[0,75]-#1 - 76 bp (100% of 76.0 bp)\n",
      "010 |                                              0100 0046 0054 0043 0043 0042    ST-E00107:437:h2gtvbbxx:8:1128:6463:1279-[0,74]-#2 - 75 bp (99% of 75.2 bp)\n",
      "011 |                                                   0100 0048 0041 0056 0046    ST-E00107:437:h2gtvbbxx:8:1202:8645:1182-[0,75]-#1 - 76 bp (100% of 76.0 bp)\n",
      "012 |                                                        0100 0048 0047 0051    ST-E00107:437:h2gtvbbxx:8:1128:7193:1162-[0,74]-#1 - 75 bp (100% of 75.0 bp)\n",
      "013 |                                                             0100 0052 0051    ST-E00107:437:h2gtvbbxx:8:1201:1772:1317-[0,75]-#2 - 76 bp (100% of 75.3 bp)\n",
      "014 |                                                                  0100 0051    ST-E00107:437:h2gtvbbxx:8:1201:3263:1063-[0,75]-#1 - 76 bp (100% of 76.0 bp)\n",
      "015 |                                                                       0100    ST-E00107:437:h2gtvbbxx:8:1202:2714:1052-[0,75]-#1 - 76 bp (100% of 76.0 bp)\n",
      "\n",
      "  ==> tmp/vidjil/result.tsv   \t(AIRR output)\n",
      "  ==> tmp/vidjil/result.vidjil\t(main output file, may be opened by the Vidjil web application)\n"
     ]
    }
   ],
   "source": [
    "!rm -rf tmp/vidjil/\n",
    "!{igmap.VidjilWrapper().run_cmd('assets/sample_UHRR_R*.fastq.gz', 'tmp/vidjil')}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Have a glimpse at raw output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 192 entries, 0 to 191\n",
      "Data columns (total 30 columns):\n",
      " #   Column               Non-Null Count  Dtype  \n",
      "---  ------               --------------  -----  \n",
      " 0   locus                192 non-null    object \n",
      " 1   duplicate_count      192 non-null    int64  \n",
      " 2   v_call               53 non-null     object \n",
      " 3   d_call               5 non-null      object \n",
      " 4   j_call               53 non-null     object \n",
      " 5   sequence_id          192 non-null    object \n",
      " 6   sequence             192 non-null    object \n",
      " 7   productive           5 non-null      object \n",
      " 8   vj_in_frame          5 non-null      object \n",
      " 9   stop_codon           4 non-null      object \n",
      " 10  junction_aa          5 non-null      object \n",
      " 11  junction             0 non-null      float64\n",
      " 12  cdr3_aa              5 non-null      object \n",
      " 13  warnings             188 non-null    object \n",
      " 14  v_sequence_start     19 non-null     float64\n",
      " 15  v_sequence_end       53 non-null     float64\n",
      " 16  d_sequence_start     5 non-null      float64\n",
      " 17  d_sequence_end       5 non-null      float64\n",
      " 18  j_sequence_start     53 non-null     float64\n",
      " 19  j_sequence_end       27 non-null     float64\n",
      " 20  cdr3_sequence_start  5 non-null      float64\n",
      " 21  cdr3_sequence_end    5 non-null      float64\n",
      " 22  v_support            192 non-null    float64\n",
      " 23  j_support            192 non-null    float64\n",
      " 24  rev_comp             0 non-null      float64\n",
      " 25  sequence_alignment   0 non-null      float64\n",
      " 26  germline_alignment   0 non-null      float64\n",
      " 27  v_cigar              0 non-null      float64\n",
      " 28  d_cigar              0 non-null      float64\n",
      " 29  j_cigar              0 non-null      float64\n",
      "dtypes: float64(17), int64(1), object(12)\n",
      "memory usage: 45.1+ KB\n",
      "None\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>duplicate_count</th>\n",
       "      <th>v_call</th>\n",
       "      <th>j_call</th>\n",
       "      <th>junction_aa</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>240</td>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>95</td>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>28</td>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156</th>\n",
       "      <td>12</td>\n",
       "      <td>TRBV20/OR9-2*01</td>\n",
       "      <td>TRBJ2-1*01</td>\n",
       "      <td>CSARESTSDPKNEQFF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>4</td>\n",
       "      <td>TRBV10-3*01</td>\n",
       "      <td>TRBJ2-5*01</td>\n",
       "      <td>CAISEPTGIR#ETQYF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>4</td>\n",
       "      <td>TRGV10*02</td>\n",
       "      <td>TRGJP1*01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>3</td>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>2</td>\n",
       "      <td>TRAV14/DV4*04</td>\n",
       "      <td>TRDJ1*01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>2</td>\n",
       "      <td>IGLV10-67*01</td>\n",
       "      <td>IGLJ7*01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>2</td>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     duplicate_count           v_call      j_call       junction_aa\n",
       "120              240      IGLV2-28*01    IGLJ2*01               NaN\n",
       "75                95      IGLV2-28*01    IGLJ2*01               NaN\n",
       "140               28      IGLV2-28*01    IGLJ2*01               NaN\n",
       "156               12  TRBV20/OR9-2*01  TRBJ2-1*01  CSARESTSDPKNEQFF\n",
       "80                 4      TRBV10-3*01  TRBJ2-5*01  CAISEPTGIR#ETQYF\n",
       "138                4        TRGV10*02   TRGJP1*01               NaN\n",
       "116                3      IGLV2-28*01    IGLJ2*01               NaN\n",
       "153                2    TRAV14/DV4*04    TRDJ1*01               NaN\n",
       "89                 2     IGLV10-67*01    IGLJ7*01               NaN\n",
       "118                2      IGLV2-28*01    IGLJ2*01               NaN"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('tmp/vidjil/result.tsv',\n",
    "                 sep='\\t',\n",
    "                 low_memory=False, \n",
    "                 usecols=lambda c: not c.startswith('Unnamed:'))\n",
    "print(df.info())\n",
    "df.sort_values('duplicate_count', inplace=True, ascending=False)\n",
    "df.dropna(subset=['v_call', 'j_call'], inplace=True)\n",
    "df[['duplicate_count', 'v_call', 'j_call', 'junction_aa']][0:10]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load vidjil tsv output using built-in function. Note same CDR3 corresponding to different sequences (i.e. offsets/groups of reads), this may be used to cound 'events' or 'molecules' in the same way as UMI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>duplicate_count</th>\n",
       "      <th>v_call</th>\n",
       "      <th>j_call</th>\n",
       "      <th>sequence</th>\n",
       "      <th>junction_aa</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>240</td>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>GGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTCGAG...</td>\n",
       "      <td>CSSYRGSATFEVVF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>95</td>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>AGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAG...</td>\n",
       "      <td>CSSYRGSATFEVVF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>28</td>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>CCAGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGC...</td>\n",
       "      <td>CSSYRGSATFEVVF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156</th>\n",
       "      <td>12</td>\n",
       "      <td>TRBV20/OR9-2*01</td>\n",
       "      <td>TRBJ2-1*01</td>\n",
       "      <td>AGACAGCAGCTTCTACATCTGCAGTGCTAGAGAGTCGACTAGCGAT...</td>\n",
       "      <td>CSARESTSDPKNEQFF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>4</td>\n",
       "      <td>TRBV10-3*01</td>\n",
       "      <td>TRBJ2-5*01</td>\n",
       "      <td>GTGTACTTCTGTGCCATCAGTGAGCCGACAGGGATCAGAAGAGACC...</td>\n",
       "      <td>CTSVPSVSRQGSEETQYF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>4</td>\n",
       "      <td>TRGV10*02</td>\n",
       "      <td>TRGJP1*01</td>\n",
       "      <td>GGAATGGCTATCAGAAAGAAGACATTACTATTTATTACGGCAATTC...</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>3</td>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCA...</td>\n",
       "      <td>CSSYRGSATFEVVF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>2</td>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCA...</td>\n",
       "      <td>CSSYRGSATFEVVF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>2</td>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCA...</td>\n",
       "      <td>CSSYRGSATFEVVF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>2</td>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>TGATTATTACTGCAGTTCATATAGAGGCACCGCCACTTTCGAGGTG...</td>\n",
       "      <td>CSSYRGTATFEVVF</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     duplicate_count           v_call      j_call  \\\n",
       "120              240      IGLV2-28*01    IGLJ2*01   \n",
       "75                95      IGLV2-28*01    IGLJ2*01   \n",
       "140               28      IGLV2-28*01    IGLJ2*01   \n",
       "156               12  TRBV20/OR9-2*01  TRBJ2-1*01   \n",
       "80                 4      TRBV10-3*01  TRBJ2-5*01   \n",
       "138                4        TRGV10*02   TRGJP1*01   \n",
       "116                3      IGLV2-28*01    IGLJ2*01   \n",
       "118                2      IGLV2-28*01    IGLJ2*01   \n",
       "117                2      IGLV2-28*01    IGLJ2*01   \n",
       "135                2      IGLV2-28*01    IGLJ2*01   \n",
       "\n",
       "                                              sequence         junction_aa  \n",
       "120  GGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTCGAG...      CSSYRGSATFEVVF  \n",
       "75   AGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAG...      CSSYRGSATFEVVF  \n",
       "140  CCAGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGC...      CSSYRGSATFEVVF  \n",
       "156  AGACAGCAGCTTCTACATCTGCAGTGCTAGAGAGTCGACTAGCGAT...    CSARESTSDPKNEQFF  \n",
       "80   GTGTACTTCTGTGCCATCAGTGAGCCGACAGGGATCAGAAGAGACC...  CTSVPSVSRQGSEETQYF  \n",
       "138  GGAATGGCTATCAGAAAGAAGACATTACTATTTATTACGGCAATTC...                      \n",
       "116  GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCA...      CSSYRGSATFEVVF  \n",
       "118  GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCA...      CSSYRGSATFEVVF  \n",
       "117  GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCA...      CSSYRGSATFEVVF  \n",
       "135  TGATTATTACTGCAGTTCATATAGAGGCACCGCCACTTTCGAGGTG...      CSSYRGTATFEVVF  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = igmap.read_vidjil('tmp/vidjil/result.tsv')\n",
    "df[['duplicate_count', 'v_call', 'j_call', 'sequence', 'junction_aa']][0:10]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load and parse into a more concise format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>v_call</th>\n",
       "      <th>j_call</th>\n",
       "      <th>junction</th>\n",
       "      <th>junction_aa</th>\n",
       "      <th>duplicate_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>TGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTC</td>\n",
       "      <td>CSSYRGSATFEVVF</td>\n",
       "      <td>379</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>TRBV20/OR9-2*01</td>\n",
       "      <td>TRBJ2-1*01</td>\n",
       "      <td>TGCAGTGCTAGAGAGTCGACTAGCGATCCAAAAAATGAGCAGTTCTTC</td>\n",
       "      <td>CSARESTSDPKNEQFF</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>TRBV10-3*01</td>\n",
       "      <td>TRBJ2-5*01</td>\n",
       "      <td>TGTACTTCTGTGCCATCAGTGAGCCGACAGGGATCAGAAGAGACCC...</td>\n",
       "      <td>CTSVPSVSRQGSEETQYF</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>TGCAGTTCATATAGAGGCACCGCCACTTTCGAGGTGGTGTTC</td>\n",
       "      <td>CSSYRGTATFEVVF</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>IGLV10-67*01</td>\n",
       "      <td>IGLJ7*01</td>\n",
       "      <td>TGTGCTCTGAGCTTCCCTGAACCAGGATCTGCCTATTACTGCTGTG...</td>\n",
       "      <td>CALSFPEPGSAYYCCAPW</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>TGCAGTTCATATATAGGCAGCGCCACTTTCGAGGTGGTGTTC</td>\n",
       "      <td>CSSYIGSATFEVVF</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ7*01</td>\n",
       "      <td>TGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTC</td>\n",
       "      <td>CSSYRGSATFEVVF</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>TRDV3*01</td>\n",
       "      <td>TRDJ3*01</td>\n",
       "      <td>TGCCACTTACTAGCTGGGAC</td>\n",
       "      <td>CHL_SWD</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ3*02</td>\n",
       "      <td>TGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGAGG</td>\n",
       "      <td>CSSYRG_APLSRR</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>TGCCGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTC</td>\n",
       "      <td>CRSYRGSATFEVVF</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>TRGV11*01</td>\n",
       "      <td>TRGJ1*01</td>\n",
       "      <td>TGTGCCTGCCAGTACTACTGTTT</td>\n",
       "      <td>CAC_VLLF</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>TRGV9*01</td>\n",
       "      <td>TRGJP2*01</td>\n",
       "      <td>TGTAATCCCCATAATCCCCATGTGTCTAGGGAGAGACCTGGTGGGAGGT</td>\n",
       "      <td>CNPHNPHV_LGRDLVGG</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>TGCAGTTCATATCGAGGCAGCGCCACTTTCGAGGTGGTGTTC</td>\n",
       "      <td>CSSYRGSATFEVVF</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>TGCAGTTCATATAGAGGCAGCGCCACTTTCGATGTGGTGTTC</td>\n",
       "      <td>CSSYRGSATFDVVF</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>TGCAGTTCATATAGATGCAGCGCCACTTTCGAGGTGGTGTTC</td>\n",
       "      <td>CSSYRCSATFEVVF</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>TGCAGTTCATATAGATCCAGCGCCACTTTCGAGGTGGTGTTC</td>\n",
       "      <td>CSSYRSSATFEVVF</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>TGCAGTTCATATAGAGGCAGAGACACTGTCGAGGTGGTGTTC</td>\n",
       "      <td>CSSYRGRDTVEVVF</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>TGCAGTTCATATAGAGCCAGCGCCACTTTCGAGGTGGTGTTC</td>\n",
       "      <td>CSSYRASATFEVVF</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>IGLV2-28*01</td>\n",
       "      <td>IGLJ-C/OR18*01</td>\n",
       "      <td>TGTTCTCAGCAGGGCTCTCCGCAGGGAGGGGGCCGGCTCCTTCCCT...</td>\n",
       "      <td>CSQQGSPQG_GPAPSLAATS</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>IGLV2-18*02</td>\n",
       "      <td>IGLJ3*02</td>\n",
       "      <td>TGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGGGGTGTTC</td>\n",
       "      <td>CSSYRGSATFEGVF</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>IGLV2-18*02</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>TGCAGTTCATATAGAGGTAGCGCCACTTTCGAGGTGGTGTTC</td>\n",
       "      <td>CSSYRGSATFEVVF</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>IGLV2-18*02</td>\n",
       "      <td>IGLJ2*01</td>\n",
       "      <td>TGCAGCTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTC</td>\n",
       "      <td>CSSYRGSATFEVVF</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IGKV2-23*01</td>\n",
       "      <td>IGKJ2*02</td>\n",
       "      <td>TGCTCGACCTTTCTCTCCAGTGGAGATTTGCCAACCTGCCCAACAA...</td>\n",
       "      <td>CSTFLSSG_FANLPNNA</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>TRGVA*01</td>\n",
       "      <td>TRGJP*01</td>\n",
       "      <td>TGTTGGGAGGGAGGCTGGCACATCAACTGTCTTGGGGCCCTGA</td>\n",
       "      <td>CWEGGWH_STVLGP*</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             v_call          j_call  \\\n",
       "18      IGLV2-28*01        IGLJ2*01   \n",
       "31  TRBV20/OR9-2*01      TRBJ2-1*01   \n",
       "29      TRBV10-3*01      TRBJ2-5*01   \n",
       "16      IGLV2-28*01        IGLJ2*01   \n",
       "8      IGLV10-67*01        IGLJ7*01   \n",
       "22      IGLV2-28*01        IGLJ2*01   \n",
       "26      IGLV2-28*01        IGLJ7*01   \n",
       "32         TRDV3*01        TRDJ3*01   \n",
       "25      IGLV2-28*01        IGLJ3*02   \n",
       "24      IGLV2-28*01        IGLJ2*01   \n",
       "35        TRGV11*01        TRGJ1*01   \n",
       "36         TRGV9*01       TRGJP2*01   \n",
       "23      IGLV2-28*01        IGLJ2*01   \n",
       "19      IGLV2-28*01        IGLJ2*01   \n",
       "21      IGLV2-28*01        IGLJ2*01   \n",
       "20      IGLV2-28*01        IGLJ2*01   \n",
       "17      IGLV2-28*01        IGLJ2*01   \n",
       "15      IGLV2-28*01        IGLJ2*01   \n",
       "13      IGLV2-28*01  IGLJ-C/OR18*01   \n",
       "12      IGLV2-18*02        IGLJ3*02   \n",
       "11      IGLV2-18*02        IGLJ2*01   \n",
       "10      IGLV2-18*02        IGLJ2*01   \n",
       "4       IGKV2-23*01        IGKJ2*02   \n",
       "37         TRGVA*01        TRGJP*01   \n",
       "\n",
       "                                             junction           junction_aa  \\\n",
       "18         TGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTC        CSSYRGSATFEVVF   \n",
       "31   TGCAGTGCTAGAGAGTCGACTAGCGATCCAAAAAATGAGCAGTTCTTC      CSARESTSDPKNEQFF   \n",
       "29  TGTACTTCTGTGCCATCAGTGAGCCGACAGGGATCAGAAGAGACCC...    CTSVPSVSRQGSEETQYF   \n",
       "16         TGCAGTTCATATAGAGGCACCGCCACTTTCGAGGTGGTGTTC        CSSYRGTATFEVVF   \n",
       "8   TGTGCTCTGAGCTTCCCTGAACCAGGATCTGCCTATTACTGCTGTG...    CALSFPEPGSAYYCCAPW   \n",
       "22         TGCAGTTCATATATAGGCAGCGCCACTTTCGAGGTGGTGTTC        CSSYIGSATFEVVF   \n",
       "26         TGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTC        CSSYRGSATFEVVF   \n",
       "32                               TGCCACTTACTAGCTGGGAC               CHL_SWD   \n",
       "25              TGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGAGG         CSSYRG_APLSRR   \n",
       "24         TGCCGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTC        CRSYRGSATFEVVF   \n",
       "35                            TGTGCCTGCCAGTACTACTGTTT              CAC_VLLF   \n",
       "36  TGTAATCCCCATAATCCCCATGTGTCTAGGGAGAGACCTGGTGGGAGGT     CNPHNPHV_LGRDLVGG   \n",
       "23         TGCAGTTCATATCGAGGCAGCGCCACTTTCGAGGTGGTGTTC        CSSYRGSATFEVVF   \n",
       "19         TGCAGTTCATATAGAGGCAGCGCCACTTTCGATGTGGTGTTC        CSSYRGSATFDVVF   \n",
       "21         TGCAGTTCATATAGATGCAGCGCCACTTTCGAGGTGGTGTTC        CSSYRCSATFEVVF   \n",
       "20         TGCAGTTCATATAGATCCAGCGCCACTTTCGAGGTGGTGTTC        CSSYRSSATFEVVF   \n",
       "17         TGCAGTTCATATAGAGGCAGAGACACTGTCGAGGTGGTGTTC        CSSYRGRDTVEVVF   \n",
       "15         TGCAGTTCATATAGAGCCAGCGCCACTTTCGAGGTGGTGTTC        CSSYRASATFEVVF   \n",
       "13  TGTTCTCAGCAGGGCTCTCCGCAGGGAGGGGGCCGGCTCCTTCCCT...  CSQQGSPQG_GPAPSLAATS   \n",
       "12         TGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGGGGTGTTC        CSSYRGSATFEGVF   \n",
       "11         TGCAGTTCATATAGAGGTAGCGCCACTTTCGAGGTGGTGTTC        CSSYRGSATFEVVF   \n",
       "10         TGCAGCTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTC        CSSYRGSATFEVVF   \n",
       "4   TGCTCGACCTTTCTCTCCAGTGGAGATTTGCCAACCTGCCCAACAA...     CSTFLSSG_FANLPNNA   \n",
       "37        TGTTGGGAGGGAGGCTGGCACATCAACTGTCTTGGGGCCCTGA       CWEGGWH_STVLGP*   \n",
       "\n",
       "    duplicate_count  \n",
       "18              379  \n",
       "31               12  \n",
       "29                4  \n",
       "16                2  \n",
       "8                 2  \n",
       "22                1  \n",
       "26                1  \n",
       "32                1  \n",
       "25                1  \n",
       "24                1  \n",
       "35                1  \n",
       "36                1  \n",
       "23                1  \n",
       "19                1  \n",
       "21                1  \n",
       "20                1  \n",
       "17                1  \n",
       "15                1  \n",
       "13                1  \n",
       "12                1  \n",
       "11                1  \n",
       "10                1  \n",
       "4                 1  \n",
       "37                1  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "igmap.read_vidjil('tmp/vidjil/result.tsv', concise=True)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
