{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import pandas as pd\n",
    "sys.path.append(\"../src/\")\n",
    "from igmap.vidjil import VidjilWrapper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p1 - \n",
      "# 2023-10-02 00:40:32\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p1.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.02%\n",
      "  ==> junction detected in 25 reads (0.0205%)\n",
      "  ==> found 12 windows in 25 reads (0.0205% of 122063 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        1    76.0          1   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        0       -          0       -\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       24    75.5         11   0.458\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        0       -          0       -\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        0       -          0       -\n",
      "\n",
      "  SEG               ->       25    75.5\n",
      "  SEG_+             ->       12    75.2\n",
      "  SEG_-             ->       13    75.8\n",
      "  SEG changed w     ->        5    75.4\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23287    75.6\n",
      "  UNSEG too few V/J ->    89759    75.5\n",
      "  UNSEG only V/5'   ->     7945    75.6\n",
      "  UNSEG only J/3'   ->     1047    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p1.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p10 - \n",
      "# 2023-10-02 00:40:32\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p10.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 33 reads (0.027%)\n",
      "  ==> found 15 windows in 33 reads (0.027% of 122066 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        4    75.2          4   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       23    75.6          5   0.217\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        2    76.0          2   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        2    75.0          2   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        2    75.5          2   1.000\n",
      "\n",
      "  SEG               ->       33    75.5\n",
      "  SEG_+             ->       14    75.7\n",
      "  SEG_-             ->       19    75.4\n",
      "  SEG changed w     ->       11    75.5\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23144    75.6\n",
      "  UNSEG too few V/J ->    89955    75.5\n",
      "  UNSEG only V/5'   ->     7918    75.5\n",
      "  UNSEG only J/3'   ->     1016    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p10.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p5 - \n",
      "# 2023-10-02 00:40:32\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p5.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.04%\n",
      "  ==> junction detected in 46 reads (0.0377%)\n",
      "  ==> found 17 windows in 46 reads (0.0377% of 122067 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        1    76.0          1   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       41    75.6         12   0.293\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        1    76.0          1   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        3    75.3          3   1.000\n",
      "\n",
      "  SEG               ->       46    75.6\n",
      "  SEG_+             ->       19    75.6\n",
      "  SEG_-             ->       27    75.6\n",
      "  SEG changed w     ->       11    75.5\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23290    75.6\n",
      "  UNSEG too few V/J ->    89962    75.5\n",
      "  UNSEG only V/5'   ->     7786    75.6\n",
      "  UNSEG only J/3'   ->      983    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p5.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p15 - \n",
      "# 2023-10-02 00:40:32\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p15.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 35 reads (0.0287%)\n",
      "  ==> found 17 windows in 35 reads (0.0287% of 122063 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        1    76.0          1   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       31    75.4         13   0.419\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        1    76.0          1   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        2    75.5          2   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        0       -          0       -\n",
      "\n",
      "  SEG               ->       35    75.4\n",
      "  SEG_+             ->       16    75.5\n",
      "  SEG_-             ->       19    75.3\n",
      "  SEG changed w     ->       11    75.5\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23048    75.6\n",
      "  UNSEG too few V/J ->    90158    75.5\n",
      "  UNSEG only V/5'   ->     7824    75.6\n",
      "  UNSEG only J/3'   ->      998    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p15.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p12 - \n",
      "# 2023-10-02 00:40:32\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p12.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 38 reads (0.0311%)\n",
      "  ==> found 17 windows in 38 reads (0.0311% of 122062 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        1    76.0          1   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        1    76.0          1   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       30    75.6          9   0.300\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        1    74.0          1   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    76.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        4    75.5          4   1.000\n",
      "\n",
      "  SEG               ->       38    75.6\n",
      "  SEG_+             ->        8    75.4\n",
      "  SEG_-             ->       30    75.6\n",
      "  SEG changed w     ->       17    75.7\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    22929    75.6\n",
      "  UNSEG too few V/J ->    90010    75.5\n",
      "  UNSEG only V/5'   ->     7992    75.6\n",
      "  UNSEG only J/3'   ->     1093    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p12.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p9 - \n",
      "# 2023-10-02 00:40:32\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p9.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 37 reads (0.0303%)\n",
      "  ==> found 15 windows in 37 reads (0.0303% of 122068 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        3    75.7          3   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       28    75.7          7   0.250\n",
      "  TRA               ->        2    74.0          1   0.500\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        0       -          0       -\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        4    76.0          4   1.000\n",
      "\n",
      "  SEG               ->       37    75.6\n",
      "  SEG_+             ->       22    75.7\n",
      "  SEG_-             ->       15    75.5\n",
      "  SEG changed w     ->       13    75.2\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23167    75.6\n",
      "  UNSEG too few V/J ->    89969    75.5\n",
      "  UNSEG only V/5'   ->     7860    75.5\n",
      "  UNSEG only J/3'   ->     1035    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p9.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p16 - \n",
      "# 2023-10-02 00:40:32\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p16.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 32 reads (0.026%)\n",
      "  ==> found 18 windows in 32 reads (0.026% of 123273 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        3    75.7          3   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       23    75.6          9   0.391\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        2    76.0          2   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        2    75.5          2   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        2    76.0          2   1.000\n",
      "\n",
      "  SEG               ->       32    75.6\n",
      "  SEG_+             ->       16    75.6\n",
      "  SEG_-             ->       16    75.6\n",
      "  SEG changed w     ->       11    75.7\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23392    75.6\n",
      "  UNSEG too few V/J ->    90782    75.5\n",
      "  UNSEG only V/5'   ->     7999    75.6\n",
      "  UNSEG only J/3'   ->     1068    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p16.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p13 - \n",
      "# 2023-10-02 00:40:32\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p13.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 43 reads (0.0338%)\n",
      "  ==> found 17 windows in 43 reads (0.0338% of 127146 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        0       -          0       -\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       35    75.4         12   0.343\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        0       -          0       -\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    76.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        7    75.6          4   0.571\n",
      "\n",
      "  SEG               ->       43    75.4\n",
      "  SEG_+             ->       21    75.2\n",
      "  SEG_-             ->       22    75.6\n",
      "  SEG changed w     ->       13    75.7\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24073    75.6\n",
      "  UNSEG too few V/J ->    93560    75.5\n",
      "  UNSEG only V/5'   ->     8374    75.5\n",
      "  UNSEG only J/3'   ->     1096    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p13.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p11 - \n",
      "# 2023-10-02 00:40:32\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p11.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 39 reads (0.0307%)\n",
      "  ==> found 17 windows in 39 reads (0.0307% of 127144 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        2    75.0          2   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        0       -          0       -\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       34    75.5         12   0.353\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        1    75.0          1   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    76.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        1    75.0          1   1.000\n",
      "\n",
      "  SEG               ->       39    75.5\n",
      "  SEG_+             ->       15    75.1\n",
      "  SEG_-             ->       24    75.8\n",
      "  SEG changed w     ->       15    75.7\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24120    75.6\n",
      "  UNSEG too few V/J ->    93614    75.5\n",
      "  UNSEG only V/5'   ->     8324    75.6\n",
      "  UNSEG only J/3'   ->     1047    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p11.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p6 - \n",
      "# 2023-10-02 00:40:32\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p6.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.04%\n",
      "  ==> junction detected in 48 reads (0.0378%)\n",
      "  ==> found 21 windows in 48 reads (0.0378% of 127152 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        2    76.0          2   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        5    75.4          5   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       37    75.4         11   0.297\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        2    76.0          1   0.500\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        2    76.0          2   1.000\n",
      "\n",
      "  SEG               ->       48    75.5\n",
      "  SEG_+             ->       26    75.4\n",
      "  SEG_-             ->       22    75.6\n",
      "  SEG changed w     ->       17    75.4\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24133    75.6\n",
      "  UNSEG too few V/J ->    93627    75.5\n",
      "  UNSEG only V/5'   ->     8292    75.6\n",
      "  UNSEG only J/3'   ->     1051    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        1    76.0\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p6.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p4 - \n",
      "# 2023-10-02 00:40:32\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p4.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 42 reads (0.033%)\n",
      "  ==> found 23 windows in 42 reads (0.033% of 127150 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        2    76.0          2   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       31    75.5         14   0.452\n",
      "  TRA               ->        2    74.0          1   0.500\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        1    75.0          1   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    75.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        5    76.0          4   0.800\n",
      "\n",
      "  SEG               ->       42    75.5\n",
      "  SEG_+             ->       20    75.2\n",
      "  SEG_-             ->       22    75.6\n",
      "  SEG changed w     ->       14    75.4\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24040    75.6\n",
      "  UNSEG too few V/J ->    93666    75.5\n",
      "  UNSEG only V/5'   ->     8314    75.5\n",
      "  UNSEG only J/3'   ->     1088    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p4.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p8 - \n",
      "# 2023-10-02 00:40:32\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p8.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.02%\n",
      "  ==> junction detected in 31 reads (0.0244%)\n",
      "  ==> found 12 windows in 31 reads (0.0244% of 127150 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        0       -          0       -\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       26    75.6          8   0.308\n",
      "  TRA               ->        1    76.0          1   1.000\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        3    76.0          2   0.667\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    75.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        0       -          0       -\n",
      "\n",
      "  SEG               ->       31    75.6\n",
      "  SEG_+             ->       14    75.8\n",
      "  SEG_-             ->       17    75.5\n",
      "  SEG changed w     ->       12    75.7\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24164    75.6\n",
      "  UNSEG too few V/J ->    93696    75.5\n",
      "  UNSEG only V/5'   ->     8228    75.5\n",
      "  UNSEG only J/3'   ->     1031    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p8.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p7 - \n",
      "# 2023-10-02 00:40:32\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p7.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.04%\n",
      "  ==> junction detected in 49 reads (0.0385%)\n",
      "  ==> found 23 windows in 49 reads (0.0385% of 127151 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        5    75.8          5   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       42    75.5         16   0.381\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        0       -          0       -\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    76.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        1    75.0          1   1.000\n",
      "\n",
      "  SEG               ->       49    75.5\n",
      "  SEG_+             ->       24    75.2\n",
      "  SEG_-             ->       25    75.7\n",
      "  SEG changed w     ->       17    75.7\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24073    75.5\n",
      "  UNSEG too few V/J ->    93726    75.5\n",
      "  UNSEG only V/5'   ->     8201    75.6\n",
      "  UNSEG only J/3'   ->     1102    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p7.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p14 - \n",
      "# 2023-10-02 00:40:32\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p14.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 35 reads (0.0275%)\n",
      "  ==> found 16 windows in 35 reads (0.0275% of 127150 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        2    75.0          2   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        1    76.0          1   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       31    75.6         12   0.387\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        0       -          0       -\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    75.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        0       -          0       -\n",
      "\n",
      "  SEG               ->       35    75.5\n",
      "  SEG_+             ->       16    75.8\n",
      "  SEG_-             ->       19    75.4\n",
      "  SEG changed w     ->       11    75.4\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24272    75.6\n",
      "  UNSEG too few V/J ->    93610    75.5\n",
      "  UNSEG only V/5'   ->     8159    75.6\n",
      "  UNSEG only J/3'   ->     1074    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p14.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p3 - \n",
      "# 2023-10-02 00:40:32\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p3.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 43 reads (0.0338%)\n",
      "  ==> found 21 windows in 43 reads (0.0338% of 127146 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        2    75.0          2   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       35    75.7         16   0.457\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        4    75.5          1   0.250\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        2    75.5          2   1.000\n",
      "\n",
      "  SEG               ->       43    75.7\n",
      "  SEG_+             ->       20    75.5\n",
      "  SEG_-             ->       23    75.8\n",
      "  SEG changed w     ->       16    75.6\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23903    75.6\n",
      "  UNSEG too few V/J ->    93673    75.5\n",
      "  UNSEG only V/5'   ->     8448    75.6\n",
      "  UNSEG only J/3'   ->     1079    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p3.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p2 - \n",
      "# 2023-10-02 00:40:32\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p2.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 43 reads (0.0338%)\n",
      "  ==> found 25 windows in 43 reads (0.0338% of 127149 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        1    74.0          1   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        4    75.5          4   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       31    75.6         13   0.419\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        2    75.0          2   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    76.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        4    74.8          4   1.000\n",
      "\n",
      "  SEG               ->       43    75.4\n",
      "  SEG_+             ->       19    75.3\n",
      "  SEG_-             ->       24    75.5\n",
      "  SEG changed w     ->       12    75.7\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24281    75.6\n",
      "  UNSEG too few V/J ->    93539    75.5\n",
      "  UNSEG only V/5'   ->     8195    75.6\n",
      "  UNSEG only J/3'   ->     1091    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p2.vidjil\t(only metadata, no clone output)\n"
     ]
    }
   ],
   "source": [
    "!{VidjilWrapper().detect_cmd('assets/sample*.fastq.gz', 'tmp/vidjil')}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations and analyzing clones\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c clones --all -o tmp/vidjil -b result - \n",
      "# 2023-10-02 00:41:16\n",
      "\n",
      "* WARNING: vidjil-algo was run with '--all' option or with a large '--max-designations/-z' option\n",
      "* vidjil-algo efficiently extracts windows overlapping the CDR3\n",
      "* to cluster reads into clones ('-c clones').\n",
      "* Computing accurate V(D)J designations for many sequences ('-c designations' or large '-z' values)\n",
      "* is not as efficient as the default '-c clones' command.\n",
      "* More information is provided in doc/vidjil-algo.md.\n",
      "\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//result.detected.vdj.fa\n",
      "\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "  ==> junction detected in 612 reads (98.9%)\n",
      "  ==> found 192 windows in 612 reads (98.9% of 619 reads)\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        8    75.2          5   0.625\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->       31    75.6         28   0.903\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->      497    75.5        113   0.227\n",
      "  TRA               ->        5    74.4          1   0.200\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->       20    75.6          6   0.300\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->       14    75.5         13   0.929\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->       37    75.6         26   0.703\n",
      "\n",
      "  SEG               ->      612    75.5\n",
      "  SEG_+             ->      612    75.5\n",
      "  SEG_-             ->        0       -\n",
      "  SEG changed w     ->      202    75.5\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->        0       -\n",
      "  UNSEG too few V/J ->        1    76.0\n",
      "  UNSEG only V/5'   ->        6    75.0\n",
      "  UNSEG only J/3'   ->        0       -\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "Sort windows by number of occurrences\n",
      "  ==> tmp/vidjil//result.windows.fa\n",
      "\n",
      "Diversity measures      IGH   IGK   IGL   TRA   TRB   TRD   TRG   all   \n",
      "  index_H_entropy       1.560 3.300 2.423 0.000 1.228 2.540 3.086 3.224 \n",
      "  index_E_equitability  0.750 0.961 0.390 0.000 0.410 0.962 0.855 0.502 \n",
      "  index_Ds_diversity    0.893 0.994 0.728 0.000 0.621 0.989 0.970 0.820 \n",
      "Considering labeled windows and windows with >= 1 reads and with a ratio >= 0.000 (0)\n",
      "  ==> keep 192 windows in 612 reads (98.869%)  \n",
      "  ==> 192 clones\n",
      "No clustering\n",
      "\n",
      "Detailed analysis of all clones\n",
      "  ==> suggested edges in tmp/vidjil//result.edges\n",
      "#### Clone #001 –     240 reads – 39.2%  \n",
      ">clone-001--IGL--0000240--39.2%--window \n",
      "GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGGGACC\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ3*02 IGLJ3*01 IGLJ2*01\n",
      ">clone-001--IGL--0000240--39.2%--ST-E00107:437:h2gtvbbxx:8:1128:7893:1076-[0,75]-#3 - 76 bp (100% of 75.5 bp) + VJ\t 1 43 47 76\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 5.21e-12 7.32e-14/5.14e-12\n",
      "GGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAGGGACCAAGGTGACC\n",
      "\n",
      "\n",
      "#### Clone #002 –      95 reads – 15.5%  \n",
      ">clone-002--IGL--0000095--15.5%--window \n",
      "CTGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAG\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ3*02 IGLJ3*01 IGLJ2*01\n",
      ">clone-002--IGL--0000095--15.5%--ST-E00107:437:h2gtvbbxx:8:1201:5613:1397-[0,75]-#10 - 76 bp (100% of 75.7 bp) + VJ\t 1 56 60 76\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 1.08e-05 1.09e-21/1.08e-05\n",
      "AGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAGG\n",
      "\n",
      "\n",
      "#### Clone #003 –      28 reads – 4.58%  \n",
      ">clone-003--IGL--0000028--4.58%--window \n",
      "TATTACTGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGG\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ3*02 IGLJ3*01 IGLJ2*01\n",
      ">clone-003--IGL--0000028--4.58%--ST-E00107:437:h2gtvbbxx:8:1202:5934:1201-[0,74]-#1 - 75 bp (100% of 75.0 bp) + VJ\t 1 58 62 75\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 6.81e-04 6.73e-23/6.81e-04\n",
      "CCAGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGG\n",
      "\n",
      "\n",
      "#### Clone #004 –      12 reads – 1.96%  \n",
      ">clone-004--TRB--0000012--1.96%--window \n",
      "TCTGCAGTGCTAGAGAGTCGACTAGCGATCCAAAAAATGAGCAGTTCTTC\n",
      "[warning] (W69) Several genes with equal probability: TRBV20/OR9-2*01 TRBV20-1*01 TRBV20/OR9-2*03\n",
      ">clone-004--TRB--0000012--1.96%--ST-E00107:437:h2gtvbbxx:8:1201:10955:1318-[0,75]-#1 - 76 bp (100% of 75.6 bp) + VDJ\t 1 33 37 44 53 76\tTRBV20/OR9-2*01 0/GTC/2 TRBD2*01 6/ATCCAAAA/7 TRBJ2-1*01  TRB SEG_+ 2.68e-10 2.43e-15/2.68e-10 {20(48)67 p CSARESTSDPKNEQFF}\n",
      "AGACAGCAGCTTCTACATCTGCAGTGCTAGAGA\n",
      "GTCGACTAGCGATCCAAAA\n",
      "AATGAGCAGTTCTTCGGGCCAGGG\n",
      "\n",
      "\n",
      "#### Clone #005 –       6 reads – 0.98%  \n",
      ">clone-005--IGL--0000006--0.98%--window \n",
      "ATGGTAGGCAGCCCTCCCTCTCCATCCCCTCACCCCACCCCTTAGCCACA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-005--IGL--0000006--0.98%--ST-E00107:437:h2gtvbbxx:8:1202:3799:1263-[2,74]-#3 - 73 bp (96% of 75.3 bp) \t ! Possibly IGL @34  @8  IGL UNSEG only J/3' 1.42e+01 1.40e+01/1.70e-01\n",
      "CCCAGACCATGGTAGGCAGCCCTCCCTCTCCATCCCCTCACCCCACCCCTTAGCCACAGTGAAGGGAATGGAA\n",
      "\n",
      "\n",
      "#### Clone #006 –       5 reads – 0.817%  \n",
      ">clone-006--TRA--0000005--0.817%--window \n",
      "TGCGAAGAAGCCACTTGCAGTGACACCAGTGAGAGCATTTTGGAAGAGGA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-006--TRA--0000005--0.817%--ST-E00107:437:h2gtvbbxx:8:1202:9441:1163-[0,73]-#2 - 74 bp (99% of 74.4 bp) \t ! Possibly TRA @55  @37  TRA UNSEG only J/3' 3.07e+00 3.03e+00/4.29e-02\n",
      "GGAGTTTTGAGGAGTATCAGCTGCGAAGAAGCCACTTGCAGTGACACCAGTGAGAGCATTTTGGAAGAGGAACC\n",
      "\n",
      "\n",
      "#### Clone #007 –       4 reads – 0.654%  \n",
      ">clone-007--TRG--0000004--0.654%--window \n",
      "TATCAGAAAGAAGACATTACTATTTATTACGGCAATTCAAGATATTTTCT\n",
      ">clone-007--TRG--0000004--0.654%--ST-E00107:437:h2gtvbbxx:8:1127:9645:1324-[0,75]-#2 - 76 bp (100% of 76.0 bp) + VJ\t 13 38 44 60\tTRGV10*02 18/GGCAA/14 TRGJP1*01  TRG SEG_+ 1.06e-01 1.05e-01/6.27e-04\n",
      "GGAATGGCTATCAGAAAGAAGACATTACTATTTATTAC\n",
      "GGCAA\n",
      "TTCAAGATATTTTCTGATTTCCGGAAGTCAGGG\n",
      "\n",
      "\n",
      "#### Clone #008 –       4 reads – 0.654%  \n",
      ">clone-008--TRB--0000004--0.654%--window \n",
      "CTTCTGTGCCATCAGTGAGCCGACAGGGATCAGAAGAGACCCAGTACTTC\n",
      "[warning] (W69) Several genes with equal probability: TRBV10-3*01 TRBV10-3*02\n",
      ">clone-008--TRB--0000004--0.654%--ST-E00107:437:h2gtvbbxx:8:1202:9770:1109-[0,75]-#1 - 76 bp (100% of 75.5 bp) + VDJ\t 1 24 27 33 39 76\tTRBV10-3*01 2/CC/2 TRBD1*01 3/ATCAG/3 TRBJ2-5*01  TRB SEG_+ 1.28e-09 1.28e-09/2.50e-19 {10(46)55 u CAISEPTGIR#ETQYF}\n",
      "GTGTACTTCTGTGCCATCAGTGAG\n",
      "CCGACAGGGATCAG\n",
      "AAGAGACCCAGTACTTCGGGCCAGGCACGCGGCTCCTG\n",
      "\n",
      "\n",
      "#### Clone #009 –       4 reads – 0.654%  \n",
      ">clone-009--TRG--0000004--0.654%--window \n",
      "CAGATTGTGACTCCTGGAGATTTGGATATATGTGAATATAATTTTAGAAG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-009--TRG--0000004--0.654%--ST-E00107:437:h2gtvbbxx:8:1202:10862:1259-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly TRG @30  @49  TRG UNSEG only J/3' 3.29e+00 2.38e+00/9.08e-01\n",
      "ATAGAGCAACAGATTGTGACTCCTGGAGATTTGGATATATGTGAATATAATTTTAGAAGCTTATCATTTACCTATG\n",
      "\n",
      "\n",
      "#### Clone #010 –       4 reads – 0.654%  \n",
      ">clone-010--TRG--0000004--0.654%--window \n",
      "AGAAAAGCAATAATTTTACAGGCATTTTGAGGTGTCTCTTTCTTTGGGTT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-010--TRG--0000004--0.654%--ST-E00107:437:h2gtvbbxx:8:1202:10719:1141-[0,74]-#1 - 75 bp (99% of 75.2 bp) \t ! Possibly TRG @43  @18  TRG UNSEG too few V/J 1.02e+01 6.64e+00/3.58e+00\n",
      "CTTAGAGAAAGAAAAGCAATAATTTTACAGGCATTTTGAGGTGTCTCTTTCTTTGGGTTCTTTCTGTTTGAAAGG\n",
      "\n",
      "\n",
      "#### Clone #011 –       4 reads – 0.654%  \n",
      ">clone-011--IGL--0000004--0.654%--window \n",
      "AAGCCCTCAGCACCACTTTGAACAGAATTCCATCATCCCTGGCCCTCCCT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-011--IGL--0000004--0.654%--ST-E00107:437:h2gtvbbxx:8:1128:5256:1403-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @32  @21  IGL UNSEG only V/5' 4.04e+00 4.03e-02/4.00e+00\n",
      "GGAACAGACTTGGGAACAAACACAAGCCCTCAGCACCACTTTGAACAGAATTCCATCATCCCTGGCCCTCCCTTCC\n",
      "\n",
      "\n",
      "#### Clone #012 –       3 reads – 0.49%  \n",
      ">clone-012--IGL--0000003--0.49%--window \n",
      "GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAAGGACC\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ2*01 IGLJ3*01 IGLJ3*02\n",
      ">clone-012--IGL--0000003--0.49%--ST-E00107:437:h2gtvbbxx:8:1128:7193:1162-[0,74]-#1 - 75 bp (100% of 75.0 bp) + VJ\t 1 51 55 75\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 1.33e-06 1.10e-18/1.33e-06\n",
      "GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAAGGACC\n",
      "\n",
      "\n",
      "#### Clone #013 –       3 reads – 0.49%  \n",
      ">clone-013--IGL--0000003--0.49%--window \n",
      "CCCGTGCTGAAGAGCAGCCTGGCCTTGGGCCTGGAGCTGCGGGCCGCAGC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-013--IGL--0000003--0.49%--ST-E00107:437:h2gtvbbxx:8:1201:1772:1317-[0,75]-#2 - 76 bp (100% of 75.3 bp) \t ! Possibly IGL @25  @29  IGL UNSEG too few V/J 9.15e+00 5.15e+00/4.00e+00\n",
      "GATGCCCGTGCTGAAGAGCAGCCTGGCCTTGGGCCTGGAGCTGCGGGCCGCAGCCGGGAGCCACTTTGATGCTGCG\n",
      "\n",
      "\n",
      "#### Clone #014 –       3 reads – 0.49%  \n",
      ">clone-014--IGL--0000003--0.49%--window \n",
      "ACCCTCAGTGCCAGCCAGCCTCTGCCAGACCTCTCTTTCCCTCTTCTCCC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-014--IGL--0000003--0.49%--ST-E00107:437:h2gtvbbxx:8:1201:3263:1063-[0,75]-#2 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @35  @1  IGL UNSEG only V/5' 4.00e+00 9.83e-06/4.00e+00\n",
      "GCACAGCTCCAGCTCGGACAGCACCCTCAGTGCCAGCCAGCCTCTGCCAGACCTCTCTTTCCCTCTTCTCCCCAAC\n",
      "\n",
      "\n",
      "#### Clone #015 –       2 reads – 0.327%  \n",
      ">clone-015--IGK--0000002--0.327%--window \n",
      "TTCCCTGTGACCCAGCCTGTTCCATAGCCAAAGGGACCAAAGTGAACGAT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-015--IGK--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:10087:1165-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGK @68  @34  IGK UNSEG only J/3' 7.03e+00 7.02e+00/9.07e-03\n",
      "CTGGCAGTTCCCTGTGACCCAGCCTGTTCCATAGCCAAAGGGACCAAAGTGAACGATCAGAGTGCCCCGCAACTAT\n",
      "\n",
      "\n",
      "#### Clone #016 –       2 reads – 0.327%  \n",
      ">clone-016--IGK--0000002--0.327%--window \n",
      "TTACTTATCAAATGGATTATCTCCCCCTCTTTAAAAGTAAGTTCATCTTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-016--IGK--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:3099:1077-[0,75]-#2 - 76 bp (100% of 75.5 bp) \t ! Possibly IGK @51  @67  IGK UNSEG too few V/J 5.04e+00 1.75e+00/3.28e+00\n",
      "CACCAGCCAGCTTCTCCAGTCTCCTTACTTATCAAATGGATTATCTCCCCCTCTTTAAAAGTAAGTTCATCTTCAT\n",
      "\n",
      "\n",
      "#### Clone #017 –       2 reads – 0.327%  \n",
      ">clone-017--IGL--0000002--0.327%--window \n",
      "TGTGCCTGGACGGGAAGATGGCAAGGAGGCACCGGAGGAGGCACAGCTCG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-017--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:5015:1246-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @50  @59  IGL UNSEG only J/3' 1.46e+01 1.46e+01/5.52e-03\n",
      "CAGGCCCAGATCAAGGGGCCCGTGGATGTGCCTGGACGGGAAGATGGCAAGGAGGCACCGGAGGAGGCACAGCTCG\n",
      "\n",
      "\n",
      "#### Clone #018 –       2 reads – 0.327%  \n",
      ">clone-018--IGL--0000002--0.327%--window \n",
      "TGGGGGACACAGGCACGCTGCTTTCCCTCCCAAGCCAGTGGGCCCCCACC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-018--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:10264:1397-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @45  @4  IGL UNSEG too few V/J 5.28e+00 1.29e+00/4.00e+00\n",
      "CTCCTGGGGGACACAGGCACGCTGCTTTCCCTCCCAAGCCAGTGGGCCCCCACCCGACTCCCCACTCTGCAGCTTC\n",
      "\n",
      "\n",
      "#### Clone #019 –       2 reads – 0.327%  \n",
      ">clone-019--TRG--0000002--0.327%--window \n",
      "TGAAAATATCTATTGAAAATGAGCAACTTGTGATTGGATCATATAGTCAG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-019--TRG--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:4510:1071-[0,75]-#1 - 76 bp (102% of 74.0 bp) \t ! Possibly TRG @36  @44  TRG UNSEG only J/3' 2.38e+00 2.38e+00/1.25e-03\n",
      "AAAGTACAGACTTCTGAAAATATCTATTGAAAATGAGCAACTTGTGATTGGATCATATAGTCAGCCTTCAGATTCC\n",
      "\n",
      "\n",
      "#### Clone #020 –       2 reads – 0.327%  \n",
      ">clone-020--TRD--0000002--0.327%--window \n",
      "TCTCAGAAAGCCTTTCTGACTGATCTCAGCATTACCTCTTTGGAAAAGGA\n",
      ">clone-020--TRD--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:1495:1101-[0,74]-#1 - 75 bp (100% of 75.0 bp) + VJ\t 46 60 61 72\tTRAV14/DV4*04 0//18 TRDJ1*01  TRD SEG_+ 8.04e-01 8.04e-01/2.86e-04\n",
      "CCCAGAAATCCAAGCCAGTTTTCTCAGAAAGCCTTTCTGACTGATCTCAGCATTACCTCT\n",
      "\n",
      "TTGGAAAAGGAAGGT\n",
      "\n",
      "\n",
      "#### Clone #021 –       2 reads – 0.327%  \n",
      ">clone-021--IGL--0000002--0.327%--window \n",
      "TACTGCAGTTCATATAGAGGCACCGCCACTTTCGAGGTGGTGTTCGGCGG\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ3*02 IGLJ3*01 IGLJ2*01\n",
      ">clone-021--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1201:7769:1266-[0,75]-#1 - 76 bp (100% of 76.0 bp) + VJ\t 1 56 60 76\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 1.08e-05 3.49e-20/1.08e-05\n",
      "AGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCACCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAGG\n",
      "\n",
      "\n",
      "#### Clone #022 –       2 reads – 0.327%  \n",
      ">clone-022--IGL--0000002--0.327%--window \n",
      "GTTTTTCTGCCTCAGCCTCCTGAGTAGCTAAGGGAGGTGTCTTGAGATTA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-022--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:6128:1156-[0,75]-#2 - 76 bp (100% of 75.5 bp) \t ! Possibly IGL @31  @59  IGL UNSEG too few V/J 1.14e+00 6.44e-01/5.00e-01\n",
      "GCGTGGTTTTTCTGCCTCAGCCTCCTGAGTAGCTAAGGGAGGTGTCTTGAGATTATCATCGGCTGAGGGTGGAAGC\n",
      "\n",
      "\n",
      "#### Clone #023 –       2 reads – 0.327%  \n",
      ">clone-023--IGL--0000002--0.327%--window \n",
      "GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGGAACC\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ2*01 IGLJ3*01 IGLJ3*02\n",
      ">clone-023--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:3700:1232-[0,74]-#1 - 75 bp (100% of 75.0 bp) + VJ\t 1 51 55 75\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 1.33e-06 1.10e-18/1.33e-06\n",
      "GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAGGAACC\n",
      "\n",
      "\n",
      "#### Clone #024 –       2 reads – 0.327%  \n",
      ">clone-024--IGL--0000002--0.327%--window \n",
      "GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGAGACC\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ2*01 IGLJ3*01 IGLJ3*02\n",
      ">clone-024--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:4142:1107-[0,74]-#1 - 75 bp (100% of 75.0 bp) + VJ\t 1 51 55 75\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 1.33e-06 1.10e-18/1.33e-06\n",
      "GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAGAGACC\n",
      "\n",
      "\n",
      "#### Clone #025 –       2 reads – 0.327%  \n",
      ">clone-025--IGL--0000002--0.327%--window \n",
      "GATCTGCCTATTACTGCTGTGCCCCATGGGGGGCTCCTTCCCTGCCTGAC\n",
      "[warning] (W69) Several genes with equal probability: IGLV10-67*01 IGLV10-67*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ7*02 IGLJ7*01\n",
      ">clone-025--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:10458:1104-[0,75]-#1 - 76 bp (100% of 76.0 bp) + VJ\t 30 44 48 47\tIGLV10-67*01 28/GTG/46 IGLJ7*01  IGL SEG_+ 7.05e-01 4.55e-01/2.50e-01\n",
      "ATGTGCTCTGAGCTTCCCTGAACCAGGATCTGCCTATTACTGCT\n",
      "GTG\n",
      "CCCCATGGGGGGCTCCTTCCCTGCCTGAC\n",
      "\n",
      "\n",
      "#### Clone #026 –       2 reads – 0.327%  \n",
      ">clone-026--TRG--0000002--0.327%--window \n",
      "GAGCAGCAGGAGGGGGGAGGGAAGGGAGCCAAGGGATGGACATCTTGTCA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-026--TRG--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1201:5113:1293-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly TRG @42  @33  TRG UNSEG too few V/J 7.51e+00 2.38e+00/5.14e+00\n",
      "GAGGAGCAGCAGGAGGGGGGAGGGAAGGGAGCCAAGGGATGGACATCTTGTCATTTTTTTTCTTTGAATAAATGTC\n",
      "\n",
      "\n",
      "#### Clone #027 –       2 reads – 0.327%  \n",
      ">clone-027--IGL--0000002--0.327%--window \n",
      "CCTCTGCCACCAGCCTCTTCTCCAACAGACTCTCCTTCCCCTTTTCCAGG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-027--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:8201:1155-[0,75]-#2 - 76 bp (101% of 75.0 bp) \t ! Possibly IGL @40  @9  IGL UNSEG too few V/J 9.30e+00 3.64e+00/5.65e+00\n",
      "CCACCTCCTCTTCGTCCTCTTCCTCCTCTGCCACCAGCCTCTTCTCCAACAGACTCTCCTTCCCCTTTTCCAGGTC\n",
      "\n",
      "\n",
      "#### Clone #028 –       2 reads – 0.327%  \n",
      ">clone-028--IGH--0000002--0.327%--window \n",
      "CAGGAAGGGGCTGGACTGGCTGTCCTTTCCAGTGGCCTGGCTCCGCTGTG\n",
      "[warning] (W69) Several genes with equal probability: IGHJ1P*02 IGHJ1P*01\n",
      ">clone-028--IGH--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:9291:1085-[0,75]-#1 - 76 bp (101% of 75.0 bp) + VDJ\t 6 21 22 27 28 22\tIGHV(II)-46-1*01 8//15 IGHD3/OR15-3a*01 10//34 IGHJ1P*01  IGH SEG_+ 4.11e-01 7.08e-02/3.40e-01\n",
      "CCTCACTCCAGGAAGGGGCTG\n",
      "GACTGG\n",
      "CTGTCCTTTCCAGTGGCCTGGCTCCGCTGTGTGGATGGGGAGATCGGGG\n",
      "\n",
      "\n",
      "#### Clone #029 –       2 reads – 0.327%  \n",
      ">clone-029--IGK--0000002--0.327%--window \n",
      "CAAGCCCCGGCTGAGCCTCAAGGGAATACCCACTCTGGTGTACAGACACG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-029--IGK--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:6890:1291-[0,75]-#1 - 76 bp (101% of 75.0 bp) \t ! Possibly IGK @56  @52  IGK UNSEG only J/3' 1.00e+01 9.92e+00/1.03e-01\n",
      "TCCAGGGCAAGCCCCGGCTGAGCCTCAAGGGAATACCCACTCTGGTGTACAGACACGACTGGAACTATGAACATCT\n",
      "\n",
      "\n",
      "#### Clone #030 –       2 reads – 0.327%  \n",
      ">clone-030--IGH--0000002--0.327%--window \n",
      "ATTTGCTCCCCTGAGTGTGTAGGTCCCTGAGACCTGCTCACCAGCCCCTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-030--IGH--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1201:10903:1363-[0,75]-#2 - 76 bp (101% of 75.0 bp) \t ! Possibly IGH @61  @40  IGH UNSEG too few V/J 1.29e+01 9.07e+00/3.85e+00\n",
      "GCTCTGGGCCATCCCAGGAGGGGGAATTTGCTCCCCTGAGTGTGTAGGTCCCTGAGACCTGCTCACCAGCCCCTCC\n",
      "\n",
      "\n",
      "#### Clone #031 –       2 reads – 0.327%  \n",
      ">clone-031--IGL--0000002--0.327%--window \n",
      "AGGGCCTCCTGGACACCTACAGCGTGTGGAGGAGTGGTCTCCCCAACTTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-031--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:4932:1253-[0,75]-#2 - 76 bp (100% of 75.5 bp) \t ! Possibly IGL @57  @44  IGL UNSEG too few V/J 6.47e+00 3.64e+00/2.83e+00\n",
      "CTTTTCCCCGGGCCTTCCAGGGCCTCCTGGACACCTACAGCGTGTGGAGGAGTGGTCTCCCCAACTTCCTAGCAGT\n",
      "\n",
      "\n",
      "#### Clone #032 –       2 reads – 0.327%  \n",
      ">clone-032--IGL--0000002--0.327%--window \n",
      "ACCATCAGTGTCAGCATTAGGGTTTTGGTTTTTGTTTCTTTTGGGTCTTT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-032--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:7815:1217-[0,75]-#2 - 76 bp (100% of 75.5 bp) \t ! Possibly IGL @21  @36  IGL UNSEG only J/3' 5.40e+00 5.15e+00/2.50e-01\n",
      "ACTGTTCAGGGTACTGACCATCAGTGTCAGCATTAGGGTTTTGGTTTTTGTTTCTTTTGGGTCTTTCTTTTTTGGC\n",
      "\n",
      "\n",
      "#### Clone #033 –       2 reads – 0.327%  \n",
      ">clone-033--IGH--0000002--0.327%--window \n",
      "AATTAGATCCCAGCCTGGACAAGGGCATGACTATTGCTGTTTGGGGACAC\n",
      "[warning] (W69) Several genes with equal probability: IGHV1/OR16-4*02 IGHV1/OR16-4*01 IGHV1/OR16-3*01 IGHV1/OR16-22*01\n",
      ">clone-033--IGH--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1201:4878:1117-[0,75]-#1 - 76 bp (100% of 76.0 bp) + VDJ\t 40 56 57 60 61 75\tIGHV1/OR16-22*01 13//3 IGHD4/OR15-4a*01 12//9 IGHJ3*01  IGH SEG_+ 1.30e-01 1.00e-01/3.01e-02\n",
      "GGAGGGAGTCAACGACCTATTGTATCAATTAGATCCCAGCCTGGACAAGGGCATGA\n",
      "CTAT\n",
      "TGCTGTTTGGGGACAC\n",
      "\n",
      "\n",
      "#### Clone #034 –       1 reads – 0.163%  \n",
      ">clone-034--IGL--0000001--0.163%--window \n",
      "TTTTTTTTTTTTTTCTTCCTTTTCCCTTTCTCGGTTCTCTTCTCTCCTCT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-034--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1202:9858:1155-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @72  @25  IGL UNSEG too few V/J 1.13e+01 7.29e+00/4.00e+00\n",
      "CAATACTAAGCTTTGCTTTTTTTTTTTTTTCTTCCTTTTCCCTTTCTCGGTTCTCTTCTCTCCTCTGAATGGCTGG\n",
      "\n",
      "\n",
      "#### Clone #035 –       1 reads – 0.163%  \n",
      ">clone-035--IGL--0000001--0.163%--window \n",
      "TTTTTTTTTTTTTGAAATGGAGTATTTCTCATTAGGTTTCACTGTCTTAT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-035--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1201:4711:1342-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @74  @62  IGL UNSEG only J/3' 1.10e+01 1.03e+01/7.07e-01\n",
      "CACCTTTTATATGAGTGGATTTTTTTTTTTTTTTTGAAATGGAGTATTTCTCATTAGGTTTCACTGTCTTATCCTT\n",
      "\n",
      "\n",
      "#### Clone #036 –       1 reads – 0.163%  \n",
      ">clone-036--IGL--0000001--0.163%--window \n",
      "TTTTTTTTTTTTTAGCTGAAGGATGGTTAGACAGCACAGCGAAGA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-036--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:2960:1395-[0,73]-#1 - 74 bp (100% of 74.0 bp) \t ! Possibly IGL @68  @7  IGL UNSEG only J/3' 1.05e+01 1.00e+01/4.86e-01\n",
      "TTTTTTTTTTTTTTTAGCTGAAGGATGGTTAGACAGCACAGCGAAGACTGCTAAATGCACTGACCCCCCCCATT\n",
      "\n",
      "\n",
      "#### Clone #037 –       1 reads – 0.163%  \n",
      ">clone-037--IGL--0000001--0.163%--window \n",
      "TTTTTTTTTTATTTTGCCTTTGTTTGTTTTTAACGTTTATCCTCTCCTCT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-037--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1201:6253:1105-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @11  @24  IGL UNSEG only J/3' 2.70e+00 2.58e+00/1.25e-01\n",
      "CTGATACTGATTTTTTTTTTTTTTTTTTTATTTTGCCTTTGTTTGTTTTTAACGTTTATCCTCTCCTCTTTGTGTG\n",
      "\n",
      "\n",
      "#### Clone #038 –       1 reads – 0.163%  \n",
      ">clone-038--IGL--0000001--0.163%--window \n",
      "TTTTTTTTTTATAATGAGTATGATCTATTTCTTTTCAAATAATCTTTGAG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-038--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:10969:1112-[0,74]-#1 - 75 bp (100% of 75.0 bp) \t ! Possibly IGL @37  @17  IGL UNSEG too few V/J 5.57e+00 3.60e+00/1.97e+00\n",
      "ATCTTGTTTTTTTTTTTTTTTTATAATGAGTATGATCTATTTCTTTTCAAATAATCTTTGAGATCCCAGGAAAAA\n",
      "\n",
      "\n",
      "#### Clone #039 –       1 reads – 0.163%  \n",
      ">clone-039--IGK--0000001--0.163%--window \n",
      "TTTCAATGGGATTCAACAGGGCCACCTTTTTCCACTGAAGTATTAGGAGG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-039--IGK--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1127:1340:1396-[0,74]-#1 - 75 bp (100% of 75.0 bp) \t ! Possibly IGK @45  @35  IGK UNSEG too few V/J 2.19e+01 1.96e+01/2.29e+00\n",
      "TCAAGTCTGGGTTTTCAATGGGATTCAACAGGGCCACCTTTTTCCACTGAAGTATTAGGAGGACGATGAAGACGA\n",
      "\n",
      "\n",
      "#### Clone #040 –       1 reads – 0.163%  \n",
      ">clone-040--TRD--0000001--0.163%--window \n",
      "TTTATTTTTTTTATTTTTTTATTTTTTATTTTTGTATATTTCTTTAAATT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-040--TRD--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1202:10947:1050-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly TRD @16  @66  TRD UNSEG only V/5' 6.73e+00 1.80e-02/6.71e+00\n",
      "ATTCTGTGCATGTGACCTTTTTTTTTTATTTTTTTTATTTTTTTATTTTTTATTTTTGTATATTTCTTTAAATTCC\n",
      "\n",
      "\n",
      "#### Clone #041 –       1 reads – 0.163%  \n",
      ">clone-041--IGL--0000001--0.163%--window \n",
      "TTGGGATTTTTTTTTTTTTTTTTGGCTTTTGTTTTTGTTTGTTTTTTTGT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-041--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1202:5423:1130-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @17  @46  IGL UNSEG only J/3' 1.85e+00 1.82e+00/3.12e-02\n",
      "AGTGTATATGGCTTGTGTTTTGGGATTTTTTTTTTTTTTTTTGGCTTTTGTTTTTGTTTGTTTTTTTGTTTCGTTT\n",
      "\n",
      "\n",
      "#### Clone #042 –       1 reads – 0.163%  \n",
      ">clone-042--TRG--0000001--0.163%--window \n",
      "TTGAAAATGCAAAGCGAGTGAGAAAAGAAAGTGAAAAAATCAAGG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-042--TRG--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:7455:1164-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly TRG @26  @65  TRG UNSEG only J/3' 4.81e+00 4.76e+00/5.67e-02\n",
      "AATGAAGTCAACAAAAGCACTGTTTACATTATTGAAAATGCAAAGCGAGTGAGAAAAGAAAGTGAAAAAATCAAGG\n",
      "\n",
      "\n",
      "#### Clone #043 –       1 reads – 0.163%  \n",
      ">clone-043--TRD--0000001--0.163%--window \n",
      "TTCTAAAGAAAGAAGGAGCATAGTACTCATTTATTTACTCCCGCATTCAG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-043--TRD--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:3842:1346-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly TRD @20  @39  TRD UNSEG too few V/J 1.59e+01 9.21e+00/6.71e+00\n",
      "CTTCTAAAGAAAGAAGGAGCATAGTACTCATTTATTTACTCCCGCATTCAGTACTTATTCCAATCACTTCATTAGG\n",
      "\n",
      "\n",
      "#### Clone #044 –       1 reads – 0.163%  \n",
      ">clone-044--IGL--0000001--0.163%--window \n",
      "TTCATATAGAGGCAGCGCCACTTTCGAGGGGGTGTTCGGCGGAGGGACCA\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-18*02 IGLV2-18*03 IGLV2-18*04 IGLV2-28*01\n",
      ">clone-044--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:2070:1410-[0,73]-#1 - 74 bp (100% of 74.0 bp) + VJ\t 1 31 36 73\tIGLV2-18*02 0/GAGG/2 IGLJ3*02  IGL SEG_+ 1.20e-06 1.20e-06/1.08e-16 {2(42)43 p CSSYRGSATFEGVF}\n",
      "CTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAGG\n",
      "GGGTGTTCGGCGGAGGGACCAAGGTGACCGTCCTACGTC\n",
      "\n",
      "\n",
      "#### Clone #045 –       1 reads – 0.163%  \n",
      ">clone-045--TRD--0000001--0.163%--window \n",
      "TTCAGCACCATCCCTGAGCTCATTAACTACCCTCAGCACAACTCTGCAGG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-045--TRD--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1201:7800:1203-[0,74]-#1 - 75 bp (100% of 75.0 bp) \t ! Possibly TRD @62  @34  TRD UNSEG only J/3' 9.09e+00 9.09e+00/8.08e-04\n",
      "TTCAGCACCATCCCTGAGCTCATTAACTACCCTCAGCACAACTCTGCAGGACTCATATCCAGGCTCAAATATCCA\n",
      "\n",
      "\n",
      "#### Clone #046 –       1 reads – 0.163%  \n",
      ">clone-046--IGK--0000001--0.163%--window \n",
      "TTAGATTACACTCATACCATTGTATCATCTTGCTATAAAAAGTTACTTAA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-046--IGK--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:7574:1305-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGK @50  @11  IGK UNSEG too few V/J 3.40e+00 1.75e+00/1.64e+00\n",
      "AAGTTTAGATTACACTCATACCATTGTATCATCTTGCTATAAAAAGTTACTTAAAACTATTTCTTTTGCATATAAA\n",
      "\n",
      "\n",
      "#### Clone #047 –       1 reads – 0.163%  \n",
      ">clone-047--IGL--0000001--0.163%--window \n",
      "TGTTTCTGTTTTTTTTTTTTTTTTGTGCCTCTTTTTCCCCTTCCCTACTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-047--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:9796:1410-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @76  @20  IGL UNSEG too few V/J 1.14e+00 6.44e-01/5.00e-01\n",
      "TGTCGACTGAAACTCACTTTTATGTTTCTGTTTTTTTTTTTTTTTTGTGCCTCTTTTTCCCCTTCCCTACTCATCT\n",
      "\n",
      "\n",
      "#### Clone #048 –       1 reads – 0.163%  \n",
      ">clone-048--IGK--0000001--0.163%--window \n",
      "TGTTATAGTCTGATTATTTCTGTGTTTTGAATATGATTTTCCTAATGCTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-048--IGK--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1202:4873:1086-[0,73]-#1 - 74 bp (100% of 74.0 bp) \t ! Possibly IGK @67  @1  IGK UNSEG only V/5' 6.47e+00 7.55e-02/6.40e+00\n",
      "TGTATACCACTATCTATGCAGCATTGTTATAGTCTGATTATTTCTGTGTTTTGAATATGATTTTCCTAATGCTC\n",
      "\n",
      "\n",
      "#### Clone #049 –       1 reads – 0.163%  \n",
      ">clone-049--TRD--0000001--0.163%--window \n",
      "TGTGGTGTCTGAAGACCAAGACTACACAGCTCATTTTGCTGATCCTGATA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-049--TRD--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1202:3024:1049-[0,74]-#1 - 75 bp (100% of 75.0 bp) \t ! Possibly TRD @71  @62  TRD UNSEG only J/3' 1.32e+01 1.29e+01/2.93e-01\n",
      "TTGTGGTGTCTGAAGACCAAGACTACACAGCTCATTTTGCTGATCCTGATACATTAGTTAACTGGGACTTTGTGG\n",
      "\n",
      "\n",
      "#### Clone #050 –       1 reads – 0.163%  \n",
      ">clone-050--TRG--0000001--0.163%--window \n",
      "TGTAGCAAAACAAAAAGTTTTATAAATTTCAGATAAGCTTTCTGTTTTTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-050--TRG--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1201:2533:1310-[0,74]-#1 - 75 bp (100% of 75.0 bp) \t ! Possibly TRG @59  @39  TRG UNSEG only J/3' 2.13e+02 2.12e+02/8.96e-01\n",
      "TATTCAAATACCTTCTGTTTTCTATGTAGCAAAACAAAAAGTTTTATAAATTTCAGATAAGCTTTCTGTTTTTCT\n",
      "\n",
      "\n",
      ".....\n",
      "\n",
      "#### Clones #510 to #192...\n",
      "#### end of clones\n",
      "\n",
      "Please review the 25 suggested edge(s) in tmp/vidjil//.edges\n",
      "Comparing clone consensus sequences 2 by 2\n",
      "    | 0001 0002 0003 0004 0005 0006 0007 0008 0009 0010 0011 0012 0013 0014 0015 \n",
      "001 | 0100 0070 0065 0054 0045 0053 0051 0053 0046 0046 0047 0079 0052 0043 0048    ST-E00107:437:h2gtvbbxx:8:1128:7893:1076-[0,75]-#3 - 76 bp (100% of 75.5 bp)\n",
      "002 |      0100 0093!0058 0046 0051 0052 0051 0048 0055 0044 0088 0050 0047 0047    ST-E00107:437:h2gtvbbxx:8:1201:5613:1397-[0,75]-#10 - 76 bp (100% of 75.7 bp)\n",
      "003 |           0100 0057 0048 0051 0052 0050 0047 0054 0046 0082 0053 0048 0048    ST-E00107:437:h2gtvbbxx:8:1202:5934:1201-[0,74]-#1 - 75 bp (100% of 75.0 bp)\n",
      "004 |                0100 0045 0054 0052 0057 0055 0049 0048 0058 0047 0051 0052    ST-E00107:437:h2gtvbbxx:8:1201:10955:1318-[0,75]-#1 - 76 bp (100% of 75.6 bp)\n",
      "005 |                     0100 0047 0043 0040 0041 0040 0044 0047 0052 0052 0048    ST-E00107:437:h2gtvbbxx:8:1202:3799:1263-[2,74]-#3 - 73 bp (96% of 75.3 bp)\n",
      "006 |                          0100 0052 0052 0047 0048 0047 0055 0053 0048 0050    ST-E00107:437:h2gtvbbxx:8:1202:9441:1163-[0,73]-#2 - 74 bp (99% of 74.4 bp)\n",
      "007 |                               0100 0048 0052 0055 0050 0053 0045 0048 0044    ST-E00107:437:h2gtvbbxx:8:1127:9645:1324-[0,75]-#2 - 76 bp (100% of 76.0 bp)\n",
      "008 |                                    0100 0048 0040 0052 0052 0053 0048 0056    ST-E00107:437:h2gtvbbxx:8:1202:9770:1109-[0,75]-#1 - 76 bp (100% of 75.5 bp)\n",
      "009 |                                         0100 0051 0047 0045 0044 0043 0050    ST-E00107:437:h2gtvbbxx:8:1202:10862:1259-[0,75]-#1 - 76 bp (100% of 76.0 bp)\n",
      "010 |                                              0100 0045 0054 0043 0043 0038    ST-E00107:437:h2gtvbbxx:8:1202:10719:1141-[0,74]-#1 - 75 bp (99% of 75.2 bp)\n",
      "011 |                                                   0100 0047 0046 0057 0049    ST-E00107:437:h2gtvbbxx:8:1128:5256:1403-[0,75]-#1 - 76 bp (100% of 76.0 bp)\n",
      "012 |                                                        0100 0048 0047 0046    ST-E00107:437:h2gtvbbxx:8:1128:7193:1162-[0,74]-#1 - 75 bp (100% of 75.0 bp)\n",
      "013 |                                                             0100 0052 0049    ST-E00107:437:h2gtvbbxx:8:1201:1772:1317-[0,75]-#2 - 76 bp (100% of 75.3 bp)\n",
      "014 |                                                                  0100 0051    ST-E00107:437:h2gtvbbxx:8:1201:3263:1063-[0,75]-#2 - 76 bp (100% of 76.0 bp)\n",
      "015 |                                                                       0100    ST-E00107:437:h2gtvbbxx:8:1128:10087:1165-[0,75]-#1 - 76 bp (100% of 76.0 bp)\n",
      "\n",
      "  ==> tmp/vidjil/result.tsv   \t(AIRR output)\n",
      "  ==> tmp/vidjil/result.vidjil\t(main output file, may be opened by the Vidjil web application)\n"
     ]
    }
   ],
   "source": [
    "!{VidjilWrapper().clones_cmd('tmp/vidjil/_p*.fa', 'tmp/vidjil')}\n",
    "!rm -r tmp/vidjil/_p*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p13 - \n",
      "# 2023-10-02 00:41:31\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p13.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 32 reads (0.0262%)\n",
      "  ==> found 18 windows in 32 reads (0.0262% of 122065 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        1    76.0          1   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        1    76.0          1   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       27    75.5         13   0.481\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        1    76.0          1   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    76.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        1    76.0          1   1.000\n",
      "\n",
      "  SEG               ->       32    75.6\n",
      "  SEG_+             ->       12    75.4\n",
      "  SEG_-             ->       20    75.7\n",
      "  SEG changed w     ->       10    75.8\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23149    75.6\n",
      "  UNSEG too few V/J ->    89808    75.5\n",
      "  UNSEG only V/5'   ->     8052    75.5\n",
      "  UNSEG only J/3'   ->     1024    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p13.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p15 - \n",
      "# 2023-10-02 00:41:31\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p15.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 35 reads (0.0287%)\n",
      "  ==> found 15 windows in 35 reads (0.0287% of 122065 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        1    74.0          1   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        2    76.0          2   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       29    75.6          9   0.310\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        2    76.0          2   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        1    75.0          1   1.000\n",
      "\n",
      "  SEG               ->       35    75.5\n",
      "  SEG_+             ->       16    75.7\n",
      "  SEG_-             ->       19    75.4\n",
      "  SEG changed w     ->       11    75.5\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23291    75.6\n",
      "  UNSEG too few V/J ->    89918    75.5\n",
      "  UNSEG only V/5'   ->     7833    75.5\n",
      "  UNSEG only J/3'   ->      988    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p15.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p3 - \n",
      "# 2023-10-02 00:41:31\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p3.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.02%\n",
      "  ==> junction detected in 28 reads (0.0229%)\n",
      "  ==> found 14 windows in 28 reads (0.0229% of 122063 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        1    76.0          1   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        2    76.0          2   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       23    75.6          9   0.391\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        0       -          0       -\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        2    76.0          2   1.000\n",
      "\n",
      "  SEG               ->       28    75.7\n",
      "  SEG_+             ->       12    75.8\n",
      "  SEG_-             ->       16    75.6\n",
      "  SEG changed w     ->       10    75.7\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    22880    75.6\n",
      "  UNSEG too few V/J ->    90199    75.5\n",
      "  UNSEG only V/5'   ->     7930    75.5\n",
      "  UNSEG only J/3'   ->     1026    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p3.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p2 - \n",
      "# 2023-10-02 00:41:31\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p2.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 36 reads (0.0295%)\n",
      "  ==> found 17 windows in 36 reads (0.0295% of 122059 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        2    75.5          2   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       29    75.4         10   0.345\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        2    76.0          2   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        2    75.0          2   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        1    76.0          1   1.000\n",
      "\n",
      "  SEG               ->       36    75.4\n",
      "  SEG_+             ->       16    75.4\n",
      "  SEG_-             ->       20    75.4\n",
      "  SEG changed w     ->       11    75.6\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    22992    75.6\n",
      "  UNSEG too few V/J ->    90022    75.5\n",
      "  UNSEG only V/5'   ->     7932    75.6\n",
      "  UNSEG only J/3'   ->     1077    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p2.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p7 - \n",
      "# 2023-10-02 00:41:31\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p7.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.02%\n",
      "  ==> junction detected in 24 reads (0.0197%)\n",
      "  ==> found 16 windows in 24 reads (0.0197% of 122069 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        0       -          0       -\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       19    75.7         11   0.579\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        0       -          0       -\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        2    76.0          2   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        3    76.0          3   1.000\n",
      "\n",
      "  SEG               ->       24    75.8\n",
      "  SEG_+             ->       10    75.6\n",
      "  SEG_-             ->       14    75.9\n",
      "  SEG changed w     ->        9    75.7\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23164    75.6\n",
      "  UNSEG too few V/J ->    89950    75.5\n",
      "  UNSEG only V/5'   ->     7853    75.6\n",
      "  UNSEG only J/3'   ->     1078    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p7.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p9 - \n",
      "# 2023-10-02 00:41:31\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p9.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.04%\n",
      "  ==> junction detected in 43 reads (0.0352%)\n",
      "  ==> found 13 windows in 43 reads (0.0352% of 122067 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        3    75.7          3   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       35    75.7          6   0.171\n",
      "  TRA               ->        2    74.0          1   0.500\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        0       -          0       -\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        3    76.0          3   1.000\n",
      "\n",
      "  SEG               ->       43    75.6\n",
      "  SEG_+             ->       21    75.7\n",
      "  SEG_-             ->       22    75.5\n",
      "  SEG changed w     ->       14    75.4\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23133    75.5\n",
      "  UNSEG too few V/J ->    89993    75.5\n",
      "  UNSEG only V/5'   ->     7870    75.6\n",
      "  UNSEG only J/3'   ->     1028    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p9.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p4 - \n",
      "# 2023-10-02 00:41:31\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p4.detected.vdj.fa\n",
      "....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.04%\n",
      "  ==> junction detected in 45 reads (0.0365%)\n",
      "  ==> found 20 windows in 45 reads (0.0365% of 123277 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        1    76.0          1   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        3    75.0          3   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       34    75.5         11   0.324\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        2    75.0          1   0.500\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        5    75.6          4   0.800\n",
      "\n",
      "  SEG               ->       45    75.5\n",
      "  SEG_+             ->       22    75.4\n",
      "  SEG_-             ->       23    75.6\n",
      "  SEG changed w     ->       14    75.2\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23342    75.6\n",
      "  UNSEG too few V/J ->    90897    75.5\n",
      "  UNSEG only V/5'   ->     7956    75.6\n",
      "  UNSEG only J/3'   ->     1037    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p4.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p16 - \n",
      "# 2023-10-02 00:41:31\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p16.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 44 reads (0.0346%)\n",
      "  ==> found 14 windows in 44 reads (0.0346% of 127142 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        0       -          0       -\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       36    75.4          9   0.250\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        0       -          0       -\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    76.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        7    75.6          4   0.571\n",
      "\n",
      "  SEG               ->       44    75.4\n",
      "  SEG_+             ->       23    75.2\n",
      "  SEG_-             ->       21    75.7\n",
      "  SEG changed w     ->       12    75.7\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24053    75.6\n",
      "  UNSEG too few V/J ->    93596    75.5\n",
      "  UNSEG only V/5'   ->     8377    75.6\n",
      "  UNSEG only J/3'   ->     1072    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p16.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p1 - \n",
      "# 2023-10-02 00:41:31\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p1.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.04%\n",
      "  ==> junction detected in 52 reads (0.0409%)\n",
      "  ==> found 25 windows in 52 reads (0.0409% of 127151 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        3    75.0          3   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       41    75.6         15   0.366\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        2    76.0          1   0.500\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        2    76.0          2   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        4    74.8          4   1.000\n",
      "\n",
      "  SEG               ->       52    75.6\n",
      "  SEG_+             ->       31    75.5\n",
      "  SEG_-             ->       21    75.7\n",
      "  SEG changed w     ->       16    75.7\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24417    75.6\n",
      "  UNSEG too few V/J ->    93430    75.5\n",
      "  UNSEG only V/5'   ->     8155    75.6\n",
      "  UNSEG only J/3'   ->     1096    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        1    76.0\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p1.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p11 - \n",
      "# 2023-10-02 00:41:31\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p11.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 36 reads (0.0283%)\n",
      "  ==> found 22 windows in 36 reads (0.0283% of 127144 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        1    76.0          1   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        2    76.0          2   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       30    75.6         16   0.533\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        2    75.5          2   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        1    75.0          1   1.000\n",
      "\n",
      "  SEG               ->       36    75.6\n",
      "  SEG_+             ->       13    75.5\n",
      "  SEG_-             ->       23    75.7\n",
      "  SEG changed w     ->        9    75.6\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24208    75.6\n",
      "  UNSEG too few V/J ->    93714    75.5\n",
      "  UNSEG only V/5'   ->     8155    75.6\n",
      "  UNSEG only J/3'   ->     1031    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p11.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p6 - \n",
      "# 2023-10-02 00:41:31\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p6.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 36 reads (0.0283%)\n",
      "  ==> found 16 windows in 36 reads (0.0283% of 127147 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        2    75.0          2   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        5    75.6          5   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       27    75.2          7   0.259\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        0       -          0       -\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        2    76.0          2   1.000\n",
      "\n",
      "  SEG               ->       36    75.3\n",
      "  SEG_+             ->       16    74.9\n",
      "  SEG_-             ->       20    75.6\n",
      "  SEG changed w     ->       11    75.5\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24204    75.6\n",
      "  UNSEG too few V/J ->    93602    75.5\n",
      "  UNSEG only V/5'   ->     8231    75.6\n",
      "  UNSEG only J/3'   ->     1074    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p6.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p8 - \n",
      "# 2023-10-02 00:41:31\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p8.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 44 reads (0.0346%)\n",
      "  ==> found 21 windows in 44 reads (0.0346% of 127151 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        2    75.0          2   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       35    75.4         13   0.371\n",
      "  TRA               ->        2    74.0          1   0.500\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        2    74.5          2   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        2    74.5          2   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        1    76.0          1   1.000\n",
      "\n",
      "  SEG               ->       44    75.3\n",
      "  SEG_+             ->       20    75.3\n",
      "  SEG_-             ->       24    75.2\n",
      "  SEG changed w     ->       16    75.2\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24305    75.6\n",
      "  UNSEG too few V/J ->    93428    75.5\n",
      "  UNSEG only V/5'   ->     8277    75.6\n",
      "  UNSEG only J/3'   ->     1097    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p8.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p5 - \n",
      "# 2023-10-02 00:41:31\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p5.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 43 reads (0.0338%)\n",
      "  ==> found 18 windows in 43 reads (0.0338% of 127150 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        2    76.0          2   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       37    75.5         12   0.324\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        1    76.0          1   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    76.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        2    75.5          2   1.000\n",
      "\n",
      "  SEG               ->       43    75.5\n",
      "  SEG_+             ->       22    75.5\n",
      "  SEG_-             ->       21    75.6\n",
      "  SEG changed w     ->       16    75.4\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24228    75.6\n",
      "  UNSEG too few V/J ->    93455    75.5\n",
      "  UNSEG only V/5'   ->     8338    75.6\n",
      "  UNSEG only J/3'   ->     1086    75.6\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p5.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p10 - \n",
      "# 2023-10-02 00:41:31\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p10.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 32 reads (0.0252%)\n",
      "  ==> found 16 windows in 32 reads (0.0252% of 127146 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        1    76.0          1   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        1    76.0          1   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       28    75.8         12   0.429\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        1    76.0          1   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        0       -          0       -\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        1    75.0          1   1.000\n",
      "\n",
      "  SEG               ->       32    75.8\n",
      "  SEG_+             ->       11    75.9\n",
      "  SEG_-             ->       21    75.7\n",
      "  SEG changed w     ->       12    75.8\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23930    75.6\n",
      "  UNSEG too few V/J ->    93736    75.5\n",
      "  UNSEG only V/5'   ->     8378    75.6\n",
      "  UNSEG only J/3'   ->     1070    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p10.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p12 - \n",
      "# 2023-10-02 00:41:31\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p12.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.03%\n",
      "  ==> junction detected in 43 reads (0.0338%)\n",
      "  ==> found 22 windows in 43 reads (0.0338% of 127155 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        1    74.0          1   1.000\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        2    76.0          2   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       34    75.6         14   0.412\n",
      "  TRA               ->        0       -          0       -\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        3    75.3          2   0.667\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        1    75.0          1   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        2    75.0          2   1.000\n",
      "\n",
      "  SEG               ->       43    75.5\n",
      "  SEG_+             ->       19    75.4\n",
      "  SEG_-             ->       24    75.6\n",
      "  SEG changed w     ->       16    75.6\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    23922    75.6\n",
      "  UNSEG too few V/J ->    93850    75.5\n",
      "  UNSEG only V/5'   ->     8281    75.6\n",
      "  UNSEG only J/3'   ->     1059    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p12.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c detect -o tmp/vidjil -b _p14 - \n",
      "# 2023-10-02 00:41:31\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//_p14.detected.vdj.fa\n",
      ".....\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "[warning] (W20) Very few V(D)J recombinations found: 0.04%\n",
      "  ==> junction detected in 46 reads (0.0362%)\n",
      "  ==> found 20 windows in 46 reads (0.0362% of 127149 reads)\n",
      "  ! There are not so many CDR3 windows found in this set of reads.\n",
      "  ! Please check the causes below and refer to doc/vidjil-algo.md.\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        0       -          0       -\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->        2    75.5          2   1.000\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->       38    75.5         12   0.316\n",
      "  TRA               ->        1    76.0          1   1.000\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->        2    76.0          2   1.000\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->        2    75.5          2   1.000\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->        1    76.0          1   1.000\n",
      "\n",
      "  SEG               ->       46    75.5\n",
      "  SEG_+             ->       18    75.5\n",
      "  SEG_-             ->       28    75.6\n",
      "  SEG changed w     ->       19    75.6\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->    24098    75.5\n",
      "  UNSEG too few V/J ->    93708    75.5\n",
      "  UNSEG only V/5'   ->     8241    75.6\n",
      "  UNSEG only J/3'   ->     1056    75.5\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "  ==> tmp/vidjil/_p14.vidjil\t(only metadata, no clone output)\n",
      "# vidjil-algo -- V(D)J recombinations analysis <http://www.vidjil.org/>\n",
      "# Copyright (C) 2011-2022 by the Vidjil team\n",
      "# Bonsai bioinformatics at CRIStAL (UMR CNRS 9189, Université Lille) and Inria Lille\n",
      "# VidjilNet consortium\n",
      "\n",
      "# vidjil-algo is free software, and you are welcome to redistribute it\n",
      "# under certain conditions -- see http://git.vidjil.org/blob/master/doc/LICENSE\n",
      "# No lymphocyte was harmed in the making of this software,\n",
      "# however this software is for research use only and comes with no warranty.\n",
      "\n",
      "# Please cite http://biomedcentral.com/1471-2164/15/409 if you use vidjil-algo.\n",
      "\n",
      "# development version\n",
      "# git: 200cc9a70 (2023-08-14)\n",
      "# reading from stdin, estimating 100 reads\n",
      "Detecting V(D)J recombinations and analyzing clones\n",
      "Command line: /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/bin/vidjil-algo_darwin -g /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens.g -U -c clones --all -o tmp/vidjil -b result - \n",
      "# 2023-10-02 00:42:15\n",
      "\n",
      "* WARNING: vidjil-algo was run with '--all' option or with a large '--max-designations/-z' option\n",
      "* vidjil-algo efficiently extracts windows overlapping the CDR3\n",
      "* to cluster reads into clones ('-c clones').\n",
      "* Computing accurate V(D)J designations for many sequences ('-c designations' or large '-z' values)\n",
      "* is not as efficient as the default '-c clones' command.\n",
      "* More information is provided in doc/vidjil-algo.md.\n",
      "\n",
      "Load germlines and build Kmer indexes\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHV.fa\t156380 bp in 550 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD.fa\t  1070 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKJ+down.fa\t   885 bp in   9 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLV.fa\t 44430 bp in 151 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGLJ+down.fa\t  1077 bp in  11 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAV.fa\t 37936 bp in 135 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBV.fa\t 47267 bp in 168 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD.fa\t    44 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGV.fa\t  7249 bp in  24 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRGJ+down.fa\t   692 bp in   6 sequences\n",
      "\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHD+up.fa\t  3160 bp in  44 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGHJ+down.fa\t  2074 bp in  19 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGKV.fa\t 42786 bp in 148 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-INTRON.fa\t 42958 bp in 149 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/IGK-KDE.fa\t   290 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD+up.fa\t   210 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRAJ+down.fa\t  8597 bp in  71 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBD+up.fa\t   224 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRBJ+down.fa\t  1754 bp in  16 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDV.fa\t  7022 bp in  25 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD.fa\t    30 bp in   3 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDJ+down.fa\t   452 bp in   4 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD2+up.fa\t    69 bp in   1 sequences\n",
      " <== /usr/local/Caskroom/miniconda/base/lib/python3.8/site-packages/igmap-0.0.1-py3.8.egg/igmap/external/vidjil-germline/homo-sapiens/TRDD3+down.fa\t    73 bp in   1 sequences\n",
      "Germlines loaded: Homo sapiens (9606)\n",
      "   IGH   'H'   0x01  0.149% l13 k12 ######-######   0.002% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK   'K'   0x02  0.844% l11 k10 #####-#####     0.040% l11 k10 #####-#####    \n",
      "   IGL   'L'   0x03  1.411% l11 k10 #####-#####     0.055% l11 k10 #####-#####    \n",
      "   TRA   'A'   0x04  0.025% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB   'B'   0x05  0.094% l13 k12 ######-######   0.000% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD   'D'   0x06  0.222% l11 k10 #####-#####     0.000% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   TRG   'G'   0x07  0.276% l11 k10 #####-#####     0.039% l11 k10 #####-#####    \n",
      "   IGH+  'h'   0x08  0.008% l13 k12 ######-######   0.006% l13 k12 ######-######  \n",
      "   IGK+  'k'   0x09  0.859% l11 k10 #####-#####     0.026% l11 k10 #####-#####    \n",
      "   TRA+D 'a'   0x0a  0.004% l14 k13 #######-######  0.027% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRA+D 'a'   0x0b  0.000% l14 k13 #######-######  0.010% l14 k13 #######-###### \n",
      "   TRB+  'b'   0x0c  0.001% l13 k12 ######-######   0.008% l13 k12 ######-######  \n",
      "   TRD+  'd'   0x0d  0.864% l9  k9  #########       0.023% l9  k9  #########      \n",
      "   TRD+  'd'   0x0e  0.023% l9  k9  #########       0.002% l9  k9  #########       0.157% l9  k9  #########      \n",
      "   TRD+  'd'   0x0f  0.023% l9  k9  #########       0.025% l9  k9  #########      \n",
      "\n",
      "\n",
      "Loop through reads, detecting V(D)J recombinations while extracting windows up to 50bp\n",
      "  ==> tmp/vidjil//result.detected.vdj.fa\n",
      "\n",
      "[warning] (W21) Bad e-value multiplier.\n",
      "  ! The estimated number of reads was far below the actual number of reads\n",
      "  ! There may be false positives, you should run with an higher --read-number\n",
      "  ==> junction detected in 612 reads (98.9%)\n",
      "  ==> found 192 windows in 612 reads (98.9% of 619 reads)\n",
      "                          reads av. len     clones clo/rds\n",
      "  IGH               ->        8    75.2          5   0.625\n",
      "  IGH+              ->        0       -          0       -\n",
      "  IGK               ->       31    75.6         28   0.903\n",
      "  IGK+              ->        0       -          0       -\n",
      "  IGL               ->      497    75.5        113   0.227\n",
      "  TRA               ->        5    74.4          1   0.200\n",
      "  TRA+D             ->        0       -          0       -\n",
      "  TRB               ->       20    75.6          6   0.300\n",
      "  TRB+              ->        0       -          0       -\n",
      "  TRD               ->       14    75.5         13   0.929\n",
      "  TRD+              ->        0       -          0       -\n",
      "  TRG               ->       37    75.6         26   0.703\n",
      "\n",
      "  SEG               ->      612    75.5\n",
      "  SEG_+             ->      612    75.5\n",
      "  SEG_-             ->        0       -\n",
      "  SEG changed w     ->      202    75.5\n",
      "\n",
      "  UNSEG too short   ->        0       -\n",
      "  UNSEG strand      ->        0       -\n",
      "  UNSEG too few V/J ->        1    76.0\n",
      "  UNSEG only V/5'   ->        6    75.0\n",
      "  UNSEG only J/3'   ->        0       -\n",
      "  UNSEG < delta_min ->        0       -\n",
      "  UNSEG ambiguous   ->        0       -\n",
      "  UNSEG too short w ->        0       -\n",
      "Sort windows by number of occurrences\n",
      "  ==> tmp/vidjil//result.windows.fa\n",
      "\n",
      "Diversity measures      IGH   IGK   IGL   TRA   TRB   TRD   TRG   all   \n",
      "  index_H_entropy       1.560 3.300 2.423 0.000 1.228 2.540 3.086 3.224 \n",
      "  index_E_equitability  0.750 0.961 0.390 0.000 0.410 0.962 0.855 0.502 \n",
      "  index_Ds_diversity    0.893 0.994 0.728 0.000 0.621 0.989 0.970 0.820 \n",
      "Considering labeled windows and windows with >= 1 reads and with a ratio >= 0.000 (0)\n",
      "  ==> keep 192 windows in 612 reads (98.869%)  \n",
      "  ==> 192 clones\n",
      "No clustering\n",
      "\n",
      "Detailed analysis of all clones\n",
      "  ==> suggested edges in tmp/vidjil//result.edges\n",
      "#### Clone #001 –     240 reads – 39.2%  \n",
      ">clone-001--IGL--0000240--39.2%--window \n",
      "GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGGGACC\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ3*02 IGLJ3*01 IGLJ2*01\n",
      ">clone-001--IGL--0000240--39.2%--ST-E00107:437:h2gtvbbxx:8:1202:9554:1171-[0,75]-#2 - 76 bp (100% of 75.5 bp) + VJ\t 1 43 47 76\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 5.21e-12 7.32e-14/5.14e-12\n",
      "GGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAGGGACCAAGGTGACC\n",
      "\n",
      "\n",
      "#### Clone #002 –      95 reads – 15.5%  \n",
      ">clone-002--IGL--0000095--15.5%--window \n",
      "CTGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAG\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ3*02 IGLJ3*01 IGLJ2*01\n",
      ">clone-002--IGL--0000095--15.5%--ST-E00107:437:h2gtvbbxx:8:1201:10597:1326-[0,75]-#6 - 76 bp (100% of 75.7 bp) + VJ\t 1 56 60 76\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 1.08e-05 1.09e-21/1.08e-05\n",
      "AGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAGG\n",
      "\n",
      "\n",
      "#### Clone #003 –      28 reads – 4.58%  \n",
      ">clone-003--IGL--0000028--4.58%--window \n",
      "TATTACTGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGG\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ3*02 IGLJ3*01 IGLJ2*01\n",
      ">clone-003--IGL--0000028--4.58%--ST-E00107:437:h2gtvbbxx:8:1128:8072:1290-[0,74]-#1 - 75 bp (100% of 75.0 bp) + VJ\t 1 58 62 75\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 6.81e-04 6.73e-23/6.81e-04\n",
      "CCAGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGG\n",
      "\n",
      "\n",
      "#### Clone #004 –      12 reads – 1.96%  \n",
      ">clone-004--TRB--0000012--1.96%--window \n",
      "TCTGCAGTGCTAGAGAGTCGACTAGCGATCCAAAAAATGAGCAGTTCTTC\n",
      "[warning] (W69) Several genes with equal probability: TRBV20/OR9-2*01 TRBV20-1*01 TRBV20/OR9-2*03\n",
      ">clone-004--TRB--0000012--1.96%--ST-E00107:437:h2gtvbbxx:8:1201:10955:1318-[0,75]-#12 - 76 bp (100% of 75.6 bp) + VDJ\t 1 33 37 44 53 76\tTRBV20/OR9-2*01 0/GTC/2 TRBD2*01 6/ATCCAAAA/7 TRBJ2-1*01  TRB SEG_+ 2.68e-10 2.43e-15/2.68e-10 {20(48)67 p CSARESTSDPKNEQFF}\n",
      "AGACAGCAGCTTCTACATCTGCAGTGCTAGAGA\n",
      "GTCGACTAGCGATCCAAAA\n",
      "AATGAGCAGTTCTTCGGGCCAGGG\n",
      "\n",
      "\n",
      "#### Clone #005 –       6 reads – 0.98%  \n",
      ">clone-005--IGL--0000006--0.98%--window \n",
      "ATGGTAGGCAGCCCTCCCTCTCCATCCCCTCACCCCACCCCTTAGCCACA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-005--IGL--0000006--0.98%--ST-E00107:437:h2gtvbbxx:8:1202:3799:1263-[2,74]-#3 - 73 bp (96% of 75.3 bp) \t ! Possibly IGL @34  @8  IGL UNSEG only J/3' 1.42e+01 1.40e+01/1.70e-01\n",
      "CCCAGACCATGGTAGGCAGCCCTCCCTCTCCATCCCCTCACCCCACCCCTTAGCCACAGTGAAGGGAATGGAA\n",
      "\n",
      "\n",
      "#### Clone #006 –       5 reads – 0.817%  \n",
      ">clone-006--TRA--0000005--0.817%--window \n",
      "TGCGAAGAAGCCACTTGCAGTGACACCAGTGAGAGCATTTTGGAAGAGGA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-006--TRA--0000005--0.817%--ST-E00107:437:h2gtvbbxx:8:1202:5319:1162-[0,73]-#1 - 74 bp (99% of 74.4 bp) \t ! Possibly TRA @55  @37  TRA UNSEG only J/3' 3.07e+00 3.03e+00/4.29e-02\n",
      "GGAGTTTTGAGGAGTATCAGCTGCGAAGAAGCCACTTGCAGTGACACCAGTGAGAGCATTTTGGAAGAGGAACC\n",
      "\n",
      "\n",
      "#### Clone #007 –       4 reads – 0.654%  \n",
      ">clone-007--TRG--0000004--0.654%--window \n",
      "TATCAGAAAGAAGACATTACTATTTATTACGGCAATTCAAGATATTTTCT\n",
      ">clone-007--TRG--0000004--0.654%--ST-E00107:437:h2gtvbbxx:8:1127:9645:1324-[0,75]-#2 - 76 bp (100% of 76.0 bp) + VJ\t 13 38 44 60\tTRGV10*02 18/GGCAA/14 TRGJP1*01  TRG SEG_+ 1.06e-01 1.05e-01/6.27e-04\n",
      "GGAATGGCTATCAGAAAGAAGACATTACTATTTATTAC\n",
      "GGCAA\n",
      "TTCAAGATATTTTCTGATTTCCGGAAGTCAGGG\n",
      "\n",
      "\n",
      "#### Clone #008 –       4 reads – 0.654%  \n",
      ">clone-008--TRB--0000004--0.654%--window \n",
      "CTTCTGTGCCATCAGTGAGCCGACAGGGATCAGAAGAGACCCAGTACTTC\n",
      "[warning] (W69) Several genes with equal probability: TRBV10-3*01 TRBV10-3*02\n",
      ">clone-008--TRB--0000004--0.654%--ST-E00107:437:h2gtvbbxx:8:1202:9770:1109-[0,75]-#1 - 76 bp (100% of 75.5 bp) + VDJ\t 1 24 27 33 39 76\tTRBV10-3*01 2/CC/2 TRBD1*01 3/ATCAG/3 TRBJ2-5*01  TRB SEG_+ 1.28e-09 1.28e-09/2.50e-19 {10(46)55 u CAISEPTGIR#ETQYF}\n",
      "GTGTACTTCTGTGCCATCAGTGAG\n",
      "CCGACAGGGATCAG\n",
      "AAGAGACCCAGTACTTCGGGCCAGGCACGCGGCTCCTG\n",
      "\n",
      "\n",
      "#### Clone #009 –       4 reads – 0.654%  \n",
      ">clone-009--TRG--0000004--0.654%--window \n",
      "CAGATTGTGACTCCTGGAGATTTGGATATATGTGAATATAATTTTAGAAG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-009--TRG--0000004--0.654%--ST-E00107:437:h2gtvbbxx:8:1202:10862:1259-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly TRG @30  @49  TRG UNSEG only J/3' 3.29e+00 2.38e+00/9.08e-01\n",
      "ATAGAGCAACAGATTGTGACTCCTGGAGATTTGGATATATGTGAATATAATTTTAGAAGCTTATCATTTACCTATG\n",
      "\n",
      "\n",
      "#### Clone #010 –       4 reads – 0.654%  \n",
      ">clone-010--TRG--0000004--0.654%--window \n",
      "AGAAAAGCAATAATTTTACAGGCATTTTGAGGTGTCTCTTTCTTTGGGTT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-010--TRG--0000004--0.654%--ST-E00107:437:h2gtvbbxx:8:1202:10719:1141-[0,74]-#1 - 75 bp (99% of 75.2 bp) \t ! Possibly TRG @43  @18  TRG UNSEG too few V/J 1.02e+01 6.64e+00/3.58e+00\n",
      "CTTAGAGAAAGAAAAGCAATAATTTTACAGGCATTTTGAGGTGTCTCTTTCTTTGGGTTCTTTCTGTTTGAAAGG\n",
      "\n",
      "\n",
      "#### Clone #011 –       4 reads – 0.654%  \n",
      ">clone-011--IGL--0000004--0.654%--window \n",
      "AAGCCCTCAGCACCACTTTGAACAGAATTCCATCATCCCTGGCCCTCCCT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-011--IGL--0000004--0.654%--ST-E00107:437:h2gtvbbxx:8:1202:8645:1182-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @34  @23  IGL UNSEG only V/5' 4.04e+00 4.03e-02/4.00e+00\n",
      "CAGGAACAGACTTGGGAACAAACACAAGCCCTCAGCACCACTTTGAACAGAATTCCATCATCCCTGGCCCTCCCTT\n",
      "\n",
      "\n",
      "#### Clone #012 –       3 reads – 0.49%  \n",
      ">clone-012--IGL--0000003--0.49%--window \n",
      "GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAAGGACC\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ2*01 IGLJ3*01 IGLJ3*02\n",
      ">clone-012--IGL--0000003--0.49%--ST-E00107:437:h2gtvbbxx:8:1202:8094:1143-[0,74]-#1 - 75 bp (100% of 75.0 bp) + VJ\t 1 51 55 75\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 1.33e-06 1.10e-18/1.33e-06\n",
      "GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAAGGACC\n",
      "\n",
      "\n",
      "#### Clone #013 –       3 reads – 0.49%  \n",
      ">clone-013--IGL--0000003--0.49%--window \n",
      "CCCGTGCTGAAGAGCAGCCTGGCCTTGGGCCTGGAGCTGCGGGCCGCAGC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-013--IGL--0000003--0.49%--ST-E00107:437:h2gtvbbxx:8:1201:1772:1317-[0,75]-#2 - 76 bp (100% of 75.3 bp) \t ! Possibly IGL @25  @29  IGL UNSEG too few V/J 9.15e+00 5.15e+00/4.00e+00\n",
      "GATGCCCGTGCTGAAGAGCAGCCTGGCCTTGGGCCTGGAGCTGCGGGCCGCAGCCGGGAGCCACTTTGATGCTGCG\n",
      "\n",
      "\n",
      "#### Clone #014 –       3 reads – 0.49%  \n",
      ">clone-014--IGL--0000003--0.49%--window \n",
      "ACCCTCAGTGCCAGCCAGCCTCTGCCAGACCTCTCTTTCCCTCTTCTCCC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-014--IGL--0000003--0.49%--ST-E00107:437:h2gtvbbxx:8:1201:8130:1079-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @35  @1  IGL UNSEG only V/5' 4.00e+00 9.83e-06/4.00e+00\n",
      "GCACAGCTCCAGCTCGGACAGCACCCTCAGTGCCAGCCAGCCTCTGCCAGACCTCTCTTTCCCTCTTCTCCCCAAC\n",
      "\n",
      "\n",
      "#### Clone #015 –       2 reads – 0.327%  \n",
      ">clone-015--IGK--0000002--0.327%--window \n",
      "TTCCCTGTGACCCAGCCTGTTCCATAGCCAAAGGGACCAAAGTGAACGAT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-015--IGK--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:10087:1165-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGK @68  @34  IGK UNSEG only J/3' 7.03e+00 7.02e+00/9.07e-03\n",
      "CTGGCAGTTCCCTGTGACCCAGCCTGTTCCATAGCCAAAGGGACCAAAGTGAACGATCAGAGTGCCCCGCAACTAT\n",
      "\n",
      "\n",
      "#### Clone #016 –       2 reads – 0.327%  \n",
      ">clone-016--IGK--0000002--0.327%--window \n",
      "TTACTTATCAAATGGATTATCTCCCCCTCTTTAAAAGTAAGTTCATCTTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-016--IGK--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:3099:1077-[0,75]-#1 - 76 bp (100% of 75.5 bp) \t ! Possibly IGK @51  @67  IGK UNSEG too few V/J 5.04e+00 1.75e+00/3.28e+00\n",
      "CACCAGCCAGCTTCTCCAGTCTCCTTACTTATCAAATGGATTATCTCCCCCTCTTTAAAAGTAAGTTCATCTTCAT\n",
      "\n",
      "\n",
      "#### Clone #017 –       2 reads – 0.327%  \n",
      ">clone-017--IGL--0000002--0.327%--window \n",
      "TGTGCCTGGACGGGAAGATGGCAAGGAGGCACCGGAGGAGGCACAGCTCG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-017--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:5015:1246-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @50  @59  IGL UNSEG only J/3' 1.46e+01 1.46e+01/5.52e-03\n",
      "CAGGCCCAGATCAAGGGGCCCGTGGATGTGCCTGGACGGGAAGATGGCAAGGAGGCACCGGAGGAGGCACAGCTCG\n",
      "\n",
      "\n",
      "#### Clone #018 –       2 reads – 0.327%  \n",
      ">clone-018--IGL--0000002--0.327%--window \n",
      "TGGGGGACACAGGCACGCTGCTTTCCCTCCCAAGCCAGTGGGCCCCCACC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-018--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:10264:1397-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @45  @4  IGL UNSEG too few V/J 5.28e+00 1.29e+00/4.00e+00\n",
      "CTCCTGGGGGACACAGGCACGCTGCTTTCCCTCCCAAGCCAGTGGGCCCCCACCCGACTCCCCACTCTGCAGCTTC\n",
      "\n",
      "\n",
      "#### Clone #019 –       2 reads – 0.327%  \n",
      ">clone-019--TRG--0000002--0.327%--window \n",
      "TGAAAATATCTATTGAAAATGAGCAACTTGTGATTGGATCATATAGTCAG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-019--TRG--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:4510:1071-[0,75]-#1 - 76 bp (102% of 74.0 bp) \t ! Possibly TRG @36  @44  TRG UNSEG only J/3' 2.38e+00 2.38e+00/1.25e-03\n",
      "AAAGTACAGACTTCTGAAAATATCTATTGAAAATGAGCAACTTGTGATTGGATCATATAGTCAGCCTTCAGATTCC\n",
      "\n",
      "\n",
      "#### Clone #020 –       2 reads – 0.327%  \n",
      ">clone-020--TRD--0000002--0.327%--window \n",
      "TCTCAGAAAGCCTTTCTGACTGATCTCAGCATTACCTCTTTGGAAAAGGA\n",
      ">clone-020--TRD--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:1495:1101-[0,74]-#1 - 75 bp (100% of 75.0 bp) + VJ\t 46 60 61 72\tTRAV14/DV4*04 0//18 TRDJ1*01  TRD SEG_+ 8.04e-01 8.04e-01/2.86e-04\n",
      "CCCAGAAATCCAAGCCAGTTTTCTCAGAAAGCCTTTCTGACTGATCTCAGCATTACCTCT\n",
      "\n",
      "TTGGAAAAGGAAGGT\n",
      "\n",
      "\n",
      "#### Clone #021 –       2 reads – 0.327%  \n",
      ">clone-021--IGL--0000002--0.327%--window \n",
      "TACTGCAGTTCATATAGAGGCACCGCCACTTTCGAGGTGGTGTTCGGCGG\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ3*02 IGLJ3*01 IGLJ2*01\n",
      ">clone-021--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1201:7769:1266-[0,75]-#1 - 76 bp (100% of 76.0 bp) + VJ\t 1 56 60 76\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 1.08e-05 3.49e-20/1.08e-05\n",
      "AGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCACCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAGG\n",
      "\n",
      "\n",
      "#### Clone #022 –       2 reads – 0.327%  \n",
      ">clone-022--IGL--0000002--0.327%--window \n",
      "GTTTTTCTGCCTCAGCCTCCTGAGTAGCTAAGGGAGGTGTCTTGAGATTA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-022--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:6128:1156-[0,75]-#2 - 76 bp (100% of 75.5 bp) \t ! Possibly IGL @31  @59  IGL UNSEG too few V/J 1.14e+00 6.44e-01/5.00e-01\n",
      "GCGTGGTTTTTCTGCCTCAGCCTCCTGAGTAGCTAAGGGAGGTGTCTTGAGATTATCATCGGCTGAGGGTGGAAGC\n",
      "\n",
      "\n",
      "#### Clone #023 –       2 reads – 0.327%  \n",
      ">clone-023--IGL--0000002--0.327%--window \n",
      "GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGGAACC\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ2*01 IGLJ3*01 IGLJ3*02\n",
      ">clone-023--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:3700:1232-[0,74]-#1 - 75 bp (100% of 75.0 bp) + VJ\t 1 51 55 75\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 1.33e-06 1.10e-18/1.33e-06\n",
      "GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAGGAACC\n",
      "\n",
      "\n",
      "#### Clone #024 –       2 reads – 0.327%  \n",
      ">clone-024--IGL--0000002--0.327%--window \n",
      "GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGAGACC\n",
      "[warning] (W53) Similar to clone #1 - IGLV2-28*01 0/GAG/1 IGLJ2*01\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-28*01 IGLV2-18*04 IGLV2-18*03 IGLV2-18*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ2*01 IGLJ3*01 IGLJ3*02\n",
      ">clone-024--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:4142:1107-[0,74]-#1 - 75 bp (100% of 75.0 bp) + VJ\t 1 51 55 75\tIGLV2-28*01 0/GAG/1 IGLJ2*01  IGL SEG_+ 1.33e-06 1.10e-18/1.33e-06\n",
      "GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAG\n",
      "GTGGTGTTCGGCGGAGAGACC\n",
      "\n",
      "\n",
      "#### Clone #025 –       2 reads – 0.327%  \n",
      ">clone-025--IGL--0000002--0.327%--window \n",
      "GATCTGCCTATTACTGCTGTGCCCCATGGGGGGCTCCTTCCCTGCCTGAC\n",
      "[warning] (W69) Several genes with equal probability: IGLV10-67*01 IGLV10-67*02\n",
      "[warning] (W69) Several genes with equal probability: IGLJ7*02 IGLJ7*01\n",
      ">clone-025--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:10458:1104-[0,75]-#1 - 76 bp (100% of 76.0 bp) + VJ\t 30 44 48 47\tIGLV10-67*01 28/GTG/46 IGLJ7*01  IGL SEG_+ 7.05e-01 4.55e-01/2.50e-01\n",
      "ATGTGCTCTGAGCTTCCCTGAACCAGGATCTGCCTATTACTGCT\n",
      "GTG\n",
      "CCCCATGGGGGGCTCCTTCCCTGCCTGAC\n",
      "\n",
      "\n",
      "#### Clone #026 –       2 reads – 0.327%  \n",
      ">clone-026--TRG--0000002--0.327%--window \n",
      "GAGCAGCAGGAGGGGGGAGGGAAGGGAGCCAAGGGATGGACATCTTGTCA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-026--TRG--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1201:5113:1293-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly TRG @42  @33  TRG UNSEG too few V/J 7.51e+00 2.38e+00/5.14e+00\n",
      "GAGGAGCAGCAGGAGGGGGGAGGGAAGGGAGCCAAGGGATGGACATCTTGTCATTTTTTTTCTTTGAATAAATGTC\n",
      "\n",
      "\n",
      "#### Clone #027 –       2 reads – 0.327%  \n",
      ">clone-027--IGL--0000002--0.327%--window \n",
      "CCTCTGCCACCAGCCTCTTCTCCAACAGACTCTCCTTCCCCTTTTCCAGG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-027--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:8201:1155-[0,75]-#1 - 76 bp (101% of 75.0 bp) \t ! Possibly IGL @40  @9  IGL UNSEG too few V/J 9.30e+00 3.64e+00/5.65e+00\n",
      "CCACCTCCTCTTCGTCCTCTTCCTCCTCTGCCACCAGCCTCTTCTCCAACAGACTCTCCTTCCCCTTTTCCAGGTC\n",
      "\n",
      "\n",
      "#### Clone #028 –       2 reads – 0.327%  \n",
      ">clone-028--IGH--0000002--0.327%--window \n",
      "CAGGAAGGGGCTGGACTGGCTGTCCTTTCCAGTGGCCTGGCTCCGCTGTG\n",
      "[warning] (W69) Several genes with equal probability: IGHJ1P*02 IGHJ1P*01\n",
      ">clone-028--IGH--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1202:9291:1085-[0,75]-#2 - 76 bp (101% of 75.0 bp) + VDJ\t 6 21 22 27 28 22\tIGHV(II)-46-1*01 8//15 IGHD3/OR15-3a*01 10//34 IGHJ1P*01  IGH SEG_+ 4.11e-01 7.08e-02/3.40e-01\n",
      "CCTCACTCCAGGAAGGGGCTG\n",
      "GACTGG\n",
      "CTGTCCTTTCCAGTGGCCTGGCTCCGCTGTGTGGATGGGGAGATCGGGG\n",
      "\n",
      "\n",
      "#### Clone #029 –       2 reads – 0.327%  \n",
      ">clone-029--IGK--0000002--0.327%--window \n",
      "CAAGCCCCGGCTGAGCCTCAAGGGAATACCCACTCTGGTGTACAGACACG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-029--IGK--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:6890:1291-[0,75]-#1 - 76 bp (101% of 75.0 bp) \t ! Possibly IGK @56  @52  IGK UNSEG only J/3' 1.00e+01 9.92e+00/1.03e-01\n",
      "TCCAGGGCAAGCCCCGGCTGAGCCTCAAGGGAATACCCACTCTGGTGTACAGACACGACTGGAACTATGAACATCT\n",
      "\n",
      "\n",
      "#### Clone #030 –       2 reads – 0.327%  \n",
      ">clone-030--IGH--0000002--0.327%--window \n",
      "ATTTGCTCCCCTGAGTGTGTAGGTCCCTGAGACCTGCTCACCAGCCCCTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-030--IGH--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1201:10903:1363-[0,75]-#1 - 76 bp (101% of 75.0 bp) \t ! Possibly IGH @61  @40  IGH UNSEG too few V/J 1.29e+01 9.07e+00/3.85e+00\n",
      "GCTCTGGGCCATCCCAGGAGGGGGAATTTGCTCCCCTGAGTGTGTAGGTCCCTGAGACCTGCTCACCAGCCCCTCC\n",
      "\n",
      "\n",
      "#### Clone #031 –       2 reads – 0.327%  \n",
      ">clone-031--IGL--0000002--0.327%--window \n",
      "AGGGCCTCCTGGACACCTACAGCGTGTGGAGGAGTGGTCTCCCCAACTTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-031--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:4932:1253-[0,75]-#1 - 76 bp (100% of 75.5 bp) \t ! Possibly IGL @57  @44  IGL UNSEG too few V/J 6.47e+00 3.64e+00/2.83e+00\n",
      "CTTTTCCCCGGGCCTTCCAGGGCCTCCTGGACACCTACAGCGTGTGGAGGAGTGGTCTCCCCAACTTCCTAGCAGT\n",
      "\n",
      "\n",
      "#### Clone #032 –       2 reads – 0.327%  \n",
      ">clone-032--IGL--0000002--0.327%--window \n",
      "ACCATCAGTGTCAGCATTAGGGTTTTGGTTTTTGTTTCTTTTGGGTCTTT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-032--IGL--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1128:7815:1217-[0,75]-#2 - 76 bp (100% of 75.5 bp) \t ! Possibly IGL @21  @36  IGL UNSEG only J/3' 5.40e+00 5.15e+00/2.50e-01\n",
      "ACTGTTCAGGGTACTGACCATCAGTGTCAGCATTAGGGTTTTGGTTTTTGTTTCTTTTGGGTCTTTCTTTTTTGGC\n",
      "\n",
      "\n",
      "#### Clone #033 –       2 reads – 0.327%  \n",
      ">clone-033--IGH--0000002--0.327%--window \n",
      "AATTAGATCCCAGCCTGGACAAGGGCATGACTATTGCTGTTTGGGGACAC\n",
      "[warning] (W69) Several genes with equal probability: IGHV1/OR16-4*02 IGHV1/OR16-4*01 IGHV1/OR16-3*01 IGHV1/OR16-22*01\n",
      ">clone-033--IGH--0000002--0.327%--ST-E00107:437:h2gtvbbxx:8:1201:4878:1117-[0,75]-#1 - 76 bp (100% of 76.0 bp) + VDJ\t 40 56 57 60 61 75\tIGHV1/OR16-22*01 13//3 IGHD4/OR15-4a*01 12//9 IGHJ3*01  IGH SEG_+ 1.30e-01 1.00e-01/3.01e-02\n",
      "GGAGGGAGTCAACGACCTATTGTATCAATTAGATCCCAGCCTGGACAAGGGCATGA\n",
      "CTAT\n",
      "TGCTGTTTGGGGACAC\n",
      "\n",
      "\n",
      "#### Clone #034 –       1 reads – 0.163%  \n",
      ">clone-034--IGL--0000001--0.163%--window \n",
      "TTTTTTTTTTTTTTCTTCCTTTTCCCTTTCTCGGTTCTCTTCTCTCCTCT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-034--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1202:9858:1155-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @72  @25  IGL UNSEG too few V/J 1.13e+01 7.29e+00/4.00e+00\n",
      "CAATACTAAGCTTTGCTTTTTTTTTTTTTTCTTCCTTTTCCCTTTCTCGGTTCTCTTCTCTCCTCTGAATGGCTGG\n",
      "\n",
      "\n",
      "#### Clone #035 –       1 reads – 0.163%  \n",
      ">clone-035--IGL--0000001--0.163%--window \n",
      "TTTTTTTTTTTTTGAAATGGAGTATTTCTCATTAGGTTTCACTGTCTTAT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-035--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1201:4711:1342-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @74  @62  IGL UNSEG only J/3' 1.10e+01 1.03e+01/7.07e-01\n",
      "CACCTTTTATATGAGTGGATTTTTTTTTTTTTTTTGAAATGGAGTATTTCTCATTAGGTTTCACTGTCTTATCCTT\n",
      "\n",
      "\n",
      "#### Clone #036 –       1 reads – 0.163%  \n",
      ">clone-036--IGL--0000001--0.163%--window \n",
      "TTTTTTTTTTTTTAGCTGAAGGATGGTTAGACAGCACAGCGAAGA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-036--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:2960:1395-[0,73]-#1 - 74 bp (100% of 74.0 bp) \t ! Possibly IGL @68  @7  IGL UNSEG only J/3' 1.05e+01 1.00e+01/4.86e-01\n",
      "TTTTTTTTTTTTTTTAGCTGAAGGATGGTTAGACAGCACAGCGAAGACTGCTAAATGCACTGACCCCCCCCATT\n",
      "\n",
      "\n",
      "#### Clone #037 –       1 reads – 0.163%  \n",
      ">clone-037--IGL--0000001--0.163%--window \n",
      "TTTTTTTTTTATTTTGCCTTTGTTTGTTTTTAACGTTTATCCTCTCCTCT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-037--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1201:6253:1105-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @11  @24  IGL UNSEG only J/3' 2.70e+00 2.58e+00/1.25e-01\n",
      "CTGATACTGATTTTTTTTTTTTTTTTTTTATTTTGCCTTTGTTTGTTTTTAACGTTTATCCTCTCCTCTTTGTGTG\n",
      "\n",
      "\n",
      "#### Clone #038 –       1 reads – 0.163%  \n",
      ">clone-038--IGL--0000001--0.163%--window \n",
      "TTTTTTTTTTATAATGAGTATGATCTATTTCTTTTCAAATAATCTTTGAG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-038--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:10969:1112-[0,74]-#1 - 75 bp (100% of 75.0 bp) \t ! Possibly IGL @37  @17  IGL UNSEG too few V/J 5.57e+00 3.60e+00/1.97e+00\n",
      "ATCTTGTTTTTTTTTTTTTTTTATAATGAGTATGATCTATTTCTTTTCAAATAATCTTTGAGATCCCAGGAAAAA\n",
      "\n",
      "\n",
      "#### Clone #039 –       1 reads – 0.163%  \n",
      ">clone-039--IGK--0000001--0.163%--window \n",
      "TTTCAATGGGATTCAACAGGGCCACCTTTTTCCACTGAAGTATTAGGAGG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-039--IGK--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1127:1340:1396-[0,74]-#1 - 75 bp (100% of 75.0 bp) \t ! Possibly IGK @45  @35  IGK UNSEG too few V/J 2.19e+01 1.96e+01/2.29e+00\n",
      "TCAAGTCTGGGTTTTCAATGGGATTCAACAGGGCCACCTTTTTCCACTGAAGTATTAGGAGGACGATGAAGACGA\n",
      "\n",
      "\n",
      "#### Clone #040 –       1 reads – 0.163%  \n",
      ">clone-040--TRD--0000001--0.163%--window \n",
      "TTTATTTTTTTTATTTTTTTATTTTTTATTTTTGTATATTTCTTTAAATT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-040--TRD--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1202:10947:1050-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly TRD @16  @66  TRD UNSEG only V/5' 6.73e+00 1.80e-02/6.71e+00\n",
      "ATTCTGTGCATGTGACCTTTTTTTTTTATTTTTTTTATTTTTTTATTTTTTATTTTTGTATATTTCTTTAAATTCC\n",
      "\n",
      "\n",
      "#### Clone #041 –       1 reads – 0.163%  \n",
      ">clone-041--IGL--0000001--0.163%--window \n",
      "TTGGGATTTTTTTTTTTTTTTTTGGCTTTTGTTTTTGTTTGTTTTTTTGT\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-041--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1202:5423:1130-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @17  @46  IGL UNSEG only J/3' 1.85e+00 1.82e+00/3.12e-02\n",
      "AGTGTATATGGCTTGTGTTTTGGGATTTTTTTTTTTTTTTTTGGCTTTTGTTTTTGTTTGTTTTTTTGTTTCGTTT\n",
      "\n",
      "\n",
      "#### Clone #042 –       1 reads – 0.163%  \n",
      ">clone-042--TRG--0000001--0.163%--window \n",
      "TTGAAAATGCAAAGCGAGTGAGAAAAGAAAGTGAAAAAATCAAGG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-042--TRG--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:7455:1164-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly TRG @26  @65  TRG UNSEG only J/3' 4.81e+00 4.76e+00/5.67e-02\n",
      "AATGAAGTCAACAAAAGCACTGTTTACATTATTGAAAATGCAAAGCGAGTGAGAAAAGAAAGTGAAAAAATCAAGG\n",
      "\n",
      "\n",
      "#### Clone #043 –       1 reads – 0.163%  \n",
      ">clone-043--TRD--0000001--0.163%--window \n",
      "TTCTAAAGAAAGAAGGAGCATAGTACTCATTTATTTACTCCCGCATTCAG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-043--TRD--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:3842:1346-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly TRD @20  @39  TRD UNSEG too few V/J 1.59e+01 9.21e+00/6.71e+00\n",
      "CTTCTAAAGAAAGAAGGAGCATAGTACTCATTTATTTACTCCCGCATTCAGTACTTATTCCAATCACTTCATTAGG\n",
      "\n",
      "\n",
      "#### Clone #044 –       1 reads – 0.163%  \n",
      ">clone-044--IGL--0000001--0.163%--window \n",
      "TTCATATAGAGGCAGCGCCACTTTCGAGGGGGTGTTCGGCGGAGGGACCA\n",
      "[warning] (W69) Several genes with equal probability: IGLV2-18*02 IGLV2-18*03 IGLV2-18*04 IGLV2-28*01\n",
      ">clone-044--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:2070:1410-[0,73]-#1 - 74 bp (100% of 74.0 bp) + VJ\t 1 31 36 73\tIGLV2-18*02 0/GAGG/2 IGLJ3*02  IGL SEG_+ 1.20e-06 1.20e-06/1.08e-16 {2(42)43 p CSSYRGSATFEGVF}\n",
      "CTGCAGTTCATATAGAGGCAGCGCCACTTTC\n",
      "GAGG\n",
      "GGGTGTTCGGCGGAGGGACCAAGGTGACCGTCCTACGTC\n",
      "\n",
      "\n",
      "#### Clone #045 –       1 reads – 0.163%  \n",
      ">clone-045--TRD--0000001--0.163%--window \n",
      "TTCAGCACCATCCCTGAGCTCATTAACTACCCTCAGCACAACTCTGCAGG\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-045--TRD--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1201:7800:1203-[0,74]-#1 - 75 bp (100% of 75.0 bp) \t ! Possibly TRD @62  @34  TRD UNSEG only J/3' 9.09e+00 9.09e+00/8.08e-04\n",
      "TTCAGCACCATCCCTGAGCTCATTAACTACCCTCAGCACAACTCTGCAGGACTCATATCCAGGCTCAAATATCCA\n",
      "\n",
      "\n",
      "#### Clone #046 –       1 reads – 0.163%  \n",
      ">clone-046--IGK--0000001--0.163%--window \n",
      "TTAGATTACACTCATACCATTGTATCATCTTGCTATAAAAAGTTACTTAA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-046--IGK--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:7574:1305-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGK @50  @11  IGK UNSEG too few V/J 3.40e+00 1.75e+00/1.64e+00\n",
      "AAGTTTAGATTACACTCATACCATTGTATCATCTTGCTATAAAAAGTTACTTAAAACTATTTCTTTTGCATATAAA\n",
      "\n",
      "\n",
      "#### Clone #047 –       1 reads – 0.163%  \n",
      ">clone-047--IGL--0000001--0.163%--window \n",
      "TGTTTCTGTTTTTTTTTTTTTTTTGTGCCTCTTTTTCCCCTTCCCTACTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-047--IGL--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1128:9796:1410-[0,75]-#1 - 76 bp (100% of 76.0 bp) \t ! Possibly IGL @76  @20  IGL UNSEG too few V/J 1.14e+00 6.44e-01/5.00e-01\n",
      "TGTCGACTGAAACTCACTTTTATGTTTCTGTTTTTTTTTTTTTTTTGTGCCTCTTTTTCCCCTTCCCTACTCATCT\n",
      "\n",
      "\n",
      "#### Clone #048 –       1 reads – 0.163%  \n",
      ">clone-048--IGK--0000001--0.163%--window \n",
      "TGTTATAGTCTGATTATTTCTGTGTTTTGAATATGATTTTCCTAATGCTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-048--IGK--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1202:4873:1086-[0,73]-#1 - 74 bp (100% of 74.0 bp) \t ! Possibly IGK @67  @1  IGK UNSEG only V/5' 6.47e+00 7.55e-02/6.40e+00\n",
      "TGTATACCACTATCTATGCAGCATTGTTATAGTCTGATTATTTCTGTGTTTTGAATATGATTTTCCTAATGCTC\n",
      "\n",
      "\n",
      "#### Clone #049 –       1 reads – 0.163%  \n",
      ">clone-049--TRD--0000001--0.163%--window \n",
      "TGTGGTGTCTGAAGACCAAGACTACACAGCTCATTTTGCTGATCCTGATA\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-049--TRD--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1202:3024:1049-[0,74]-#1 - 75 bp (100% of 75.0 bp) \t ! Possibly TRD @71  @62  TRD UNSEG only J/3' 1.32e+01 1.29e+01/2.93e-01\n",
      "TTGTGGTGTCTGAAGACCAAGACTACACAGCTCATTTTGCTGATCCTGATACATTAGTTAACTGGGACTTTGTGG\n",
      "\n",
      "\n",
      "#### Clone #050 –       1 reads – 0.163%  \n",
      ">clone-050--TRG--0000001--0.163%--window \n",
      "TGTAGCAAAACAAAAAGTTTTATAAATTTCAGATAAGCTTTCTGTTTTTC\n",
      "[warning] (W68) V(D)J designation failed, possibly complex or not recombined sequence\n",
      ">clone-050--TRG--0000001--0.163%--ST-E00107:437:h2gtvbbxx:8:1201:2533:1310-[0,74]-#1 - 75 bp (100% of 75.0 bp) \t ! Possibly TRG @59  @39  TRG UNSEG only J/3' 2.13e+02 2.12e+02/8.96e-01\n",
      "TATTCAAATACCTTCTGTTTTCTATGTAGCAAAACAAAAAGTTTTATAAATTTCAGATAAGCTTTCTGTTTTTCT\n",
      "\n",
      "\n",
      ".....\n",
      "\n",
      "#### Clones #510 to #192...\n",
      "#### end of clones\n",
      "\n",
      "Please review the 25 suggested edge(s) in tmp/vidjil//.edges\n",
      "Comparing clone consensus sequences 2 by 2\n",
      "    | 0001 0002 0003 0004 0005 0006 0007 0008 0009 0010 0011 0012 0013 0014 0015 \n",
      "001 | 0100 0070 0065 0054 0045 0053 0051 0053 0046 0046 0047 0079 0052 0043 0048    ST-E00107:437:h2gtvbbxx:8:1202:9554:1171-[0,75]-#2 - 76 bp (100% of 75.5 bp)\n",
      "002 |      0100 0093!0058 0046 0051 0052 0051 0048 0055 0043 0088 0050 0047 0047    ST-E00107:437:h2gtvbbxx:8:1201:10597:1326-[0,75]-#6 - 76 bp (100% of 75.7 bp)\n",
      "003 |           0100 0057 0048 0051 0052 0050 0047 0054 0044 0082 0053 0048 0048    ST-E00107:437:h2gtvbbxx:8:1128:8072:1290-[0,74]-#1 - 75 bp (100% of 75.0 bp)\n",
      "004 |                0100 0045 0054 0052 0057 0055 0049 0048 0058 0047 0051 0052    ST-E00107:437:h2gtvbbxx:8:1201:10955:1318-[0,75]-#12 - 76 bp (100% of 75.6 bp)\n",
      "005 |                     0100 0047 0043 0040 0041 0040 0044 0047 0052 0052 0048    ST-E00107:437:h2gtvbbxx:8:1202:3799:1263-[2,74]-#3 - 73 bp (96% of 75.3 bp)\n",
      "006 |                          0100 0052 0052 0047 0048 0048 0055 0053 0048 0050    ST-E00107:437:h2gtvbbxx:8:1202:5319:1162-[0,73]-#1 - 74 bp (99% of 74.4 bp)\n",
      "007 |                               0100 0048 0052 0055 0051 0053 0045 0048 0044    ST-E00107:437:h2gtvbbxx:8:1127:9645:1324-[0,75]-#2 - 76 bp (100% of 76.0 bp)\n",
      "008 |                                    0100 0048 0040 0051 0052 0053 0048 0056    ST-E00107:437:h2gtvbbxx:8:1202:9770:1109-[0,75]-#1 - 76 bp (100% of 75.5 bp)\n",
      "009 |                                         0100 0051 0049 0045 0044 0043 0050    ST-E00107:437:h2gtvbbxx:8:1202:10862:1259-[0,75]-#1 - 76 bp (100% of 76.0 bp)\n",
      "010 |                                              0100 0046 0054 0043 0043 0038    ST-E00107:437:h2gtvbbxx:8:1202:10719:1141-[0,74]-#1 - 75 bp (99% of 75.2 bp)\n",
      "011 |                                                   0100 0048 0041 0056 0051    ST-E00107:437:h2gtvbbxx:8:1202:8645:1182-[0,75]-#1 - 76 bp (100% of 76.0 bp)\n",
      "012 |                                                        0100 0048 0047 0046    ST-E00107:437:h2gtvbbxx:8:1202:8094:1143-[0,74]-#1 - 75 bp (100% of 75.0 bp)\n",
      "013 |                                                             0100 0052 0049    ST-E00107:437:h2gtvbbxx:8:1201:1772:1317-[0,75]-#2 - 76 bp (100% of 75.3 bp)\n",
      "014 |                                                                  0100 0051    ST-E00107:437:h2gtvbbxx:8:1201:8130:1079-[0,75]-#1 - 76 bp (100% of 76.0 bp)\n",
      "015 |                                                                       0100    ST-E00107:437:h2gtvbbxx:8:1128:10087:1165-[0,75]-#1 - 76 bp (100% of 76.0 bp)\n",
      "\n",
      "  ==> tmp/vidjil/result.tsv   \t(AIRR output)\n",
      "  ==> tmp/vidjil/result.vidjil\t(main output file, may be opened by the Vidjil web application)\n"
     ]
    }
   ],
   "source": [
    "!rm -rf tmp/vidjil/\n",
    "!{VidjilWrapper().run_cmd('assets/sample*.fastq.gz', 'tmp/vidjil')}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['locus', 'duplicate_count', 'v_call', 'd_call', 'j_call', 'sequence_id',\n",
      "       'sequence', 'productive', 'vj_in_frame', 'stop_codon', 'junction_aa',\n",
      "       'junction', 'cdr3_aa', 'warnings', 'v_sequence_start', 'v_sequence_end',\n",
      "       'd_sequence_start', 'd_sequence_end', 'j_sequence_start',\n",
      "       'j_sequence_end', 'cdr3_sequence_start', 'cdr3_sequence_end',\n",
      "       'v_support', 'j_support', 'rev_comp', 'sequence_alignment',\n",
      "       'germline_alignment', 'v_cigar', 'd_cigar', 'j_cigar'],\n",
      "      dtype='object')\n",
      "    locus  duplicate_count            v_call            d_call  \\\n",
      "120   IGL              240       IGLV2-28*01               NaN   \n",
      "75    IGL               95       IGLV2-28*01               NaN   \n",
      "140   IGL               28       IGLV2-28*01               NaN   \n",
      "156   TRB               12   TRBV20/OR9-2*01          TRBD2*01   \n",
      "80    TRB                4       TRBV10-3*01          TRBD1*01   \n",
      "138   TRG                4         TRGV10*02               NaN   \n",
      "116   IGL                3       IGLV2-28*01               NaN   \n",
      "153   TRD                2     TRAV14/DV4*04               NaN   \n",
      "89    IGL                2      IGLV10-67*01               NaN   \n",
      "118   IGL                2       IGLV2-28*01               NaN   \n",
      "117   IGL                2       IGLV2-28*01               NaN   \n",
      "135   IGL                2       IGLV2-28*01               NaN   \n",
      "7     IGH                2  IGHV1/OR16-22*01  IGHD4/OR15-4a*01   \n",
      "43    IGH                2  IGHV(II)-46-1*01  IGHD3/OR15-3a*01   \n",
      "129   IGL                1       IGLV2-28*01               NaN   \n",
      "134   IGL                1       IGLV2-28*01               NaN   \n",
      "136   IGL                1       IGLV2-28*01               NaN   \n",
      "137   IGL                1       IGLV3-32*01               NaN   \n",
      "139   TRG                1          TRGVA*01               NaN   \n",
      "133   IGK                1       IGKV2-10*01               NaN   \n",
      "123   IGL                1       IGLV2-28*01               NaN   \n",
      "128   IGL                1       IGLV2-28*01               NaN   \n",
      "114   TRB                1         TRBV15*01               NaN   \n",
      "127   IGL                1       IGLV2-18*02               NaN   \n",
      "119   IGL                1       IGLV2-28*01               NaN   \n",
      "121   IGL                1       IGLV2-28*01               NaN   \n",
      "122   IGL                1       IGLV2-28*01               NaN   \n",
      "142   IGL                1       IGLV2-28*01               NaN   \n",
      "124   IGL                1       IGLV2-28*01               NaN   \n",
      "125   IGL                1       IGLV2-28*01               NaN   \n",
      "126   IGL                1       IGLV2-28*01               NaN   \n",
      "141   IGL                1       IGLV2-28*01               NaN   \n",
      "143   IGL                1      IGLV10-54*01               NaN   \n",
      "180   IGL                1       IGLV2-18*02               NaN   \n",
      "145   IGK                1       IGKV1-32*01               NaN   \n",
      "146   IGL                1       IGLV2-28*01               NaN   \n",
      "160   TRD                1          TRDV3*01          TRDD3*01   \n",
      "162   TRG                1         TRGV11*01               NaN   \n",
      "32    IGL                1       IGLV2-28*01               NaN   \n",
      "45    IGK                1       IGKV2-23*01               NaN   \n",
      "46    IGL                1       IGLV2-18*02               NaN   \n",
      "47    TRG                1          TRGV9*01               NaN   \n",
      "15    TRG                1         TRGV10*02               NaN   \n",
      "20    IGL                1      IGLV10-54*01               NaN   \n",
      "98    IGL                1       IGLV2-18*02               NaN   \n",
      "67    IGL                1       IGLV2-28*01               NaN   \n",
      "59    IGL                1      IGLV10-67*01               NaN   \n",
      "70    IGL                1       IGLV2-28*01               NaN   \n",
      "71    IGL                1       IGLV2-28*01               NaN   \n",
      "72    IGL                1       IGLV2-28*01               NaN   \n",
      "73    IGL                1       IGLV2-28*01               NaN   \n",
      "74    IGL                1       IGLV2-28*01               NaN   \n",
      "76    IGL                1       IGLV2-28*01               NaN   \n",
      "\n",
      "             j_call                                        sequence_id  \\\n",
      "120        IGLJ2*01  GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGG...   \n",
      "75         IGLJ2*01  CTGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGC...   \n",
      "140        IGLJ2*01  TATTACTGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGT...   \n",
      "156      TRBJ2-1*01  TCTGCAGTGCTAGAGAGTCGACTAGCGATCCAAAAAATGAGCAGTT...   \n",
      "80       TRBJ2-5*01  CTTCTGTGCCATCAGTGAGCCGACAGGGATCAGAAGAGACCCAGTA...   \n",
      "138       TRGJP1*01  TATCAGAAAGAAGACATTACTATTTATTACGGCAATTCAAGATATT...   \n",
      "116        IGLJ2*01  GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAAG...   \n",
      "153        TRDJ1*01  TCTCAGAAAGCCTTTCTGACTGATCTCAGCATTACCTCTTTGGAAA...   \n",
      "89         IGLJ7*01  GATCTGCCTATTACTGCTGTGCCCCATGGGGGGCTCCTTCCCTGCC...   \n",
      "118        IGLJ2*01  GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGG...   \n",
      "117        IGLJ2*01  GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGA...   \n",
      "135        IGLJ2*01  TACTGCAGTTCATATAGAGGCACCGCCACTTTCGAGGTGGTGTTCG...   \n",
      "7          IGHJ3*01  AATTAGATCCCAGCCTGGACAAGGGCATGACTATTGCTGTTTGGGG...   \n",
      "43        IGHJ1P*01  CAGGAAGGGGCTGGACTGGCTGTCCTTTCCAGTGGCCTGGCTCCGC...   \n",
      "129  IGLJ-C/OR18*01  GTTCTCAGCAGGGCTCTCCGCAGGGAGGGGGCCGGCTCCTTCCCTG...   \n",
      "134        IGLJ2*01  TACTGCAGTTCATATAGAGCCAGCGCCACTTTCGAGGTGGTGTTCG...   \n",
      "136        IGLJ2*01  TACTGCAGTTCATATAGAGGCAGAGACACTGTCGAGGTGGTGTTCG...   \n",
      "137        IGLJ5*01  TAGAAAGCTACTCAACAGACATAATATTGTGTTTGGTGAGTACACA...   \n",
      "139        TRGJP*01  TATGAGGTGGTGTTGGGAGGGAGGCTGGCACATCAACTGTCTTGGG...   \n",
      "133        IGKJ2*03      TACACATTATCCATCTTACAAGGTTAACAATTTTACACTAAAACA   \n",
      "123        IGLJ2*01  GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGG...   \n",
      "128        IGLJ2*01  GTTCATATATAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGG...   \n",
      "114      TRBJ2-4*01  GTGGTAGCGATTTTCAGCAGAGACGTAGAAGGCGCCGGGACCCGGA...   \n",
      "127        IGLJ2*01  GTTCATATAGCGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGG...   \n",
      "119        IGLJ2*01  GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGG...   \n",
      "121        IGLJ2*01  GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGG...   \n",
      "122        IGLJ2*01  GTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGG...   \n",
      "142        IGLJ2*01  TATTCCTGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGT...   \n",
      "124        IGLJ2*01  GTTCATATAGAGGCAGCGCCACTTTCGATGTGGTGTTCGGCGGAGG...   \n",
      "125        IGLJ2*01  GTTCATATAGAGGCAGCGCCACTTTCGTGGTGGTGTTCGGCGGAGG...   \n",
      "126        IGLJ2*01  GTTCATATAGATGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGGGG...   \n",
      "141        IGLJ2*01  TATTACTGCCGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGT...   \n",
      "143        IGLJ7*01  TCACATAGGGCCAGGCAGCCTCTGTGTTTCTTTCCCTGGTCCTGAA...   \n",
      "180        IGLJ3*02  TTCATATAGAGGCAGCGCCACTTTCGAGGGGGTGTTCGGCGGAGGG...   \n",
      "145        IGKJ3*01      TCAGAGCCCCTGCCGGGACCAAAGTGAACCTCGCCGCCTCCGCAG   \n",
      "146        IGLJ3*02  TCATATAGAGGCAGCGCCACTTTCGAGGAGGTGTTCAGCGGAGGGA...   \n",
      "160        TRDJ3*01      TGCCACTTACTAGCTGGGACACCCACATTTCATCTCTCATCTTCA   \n",
      "162        TRGJ1*01  TGCCTTGTGCCCCTACTGTGCCTGCCAGTACTACTGTTTATTATAA...   \n",
      "32         IGLJ2*01  ATTATTACTGCAGTTCATATAGATCCAGCGCCACTTTCGAGGTGGT...   \n",
      "45         IGKJ2*02  CAGTGGAGATTTGCCAACCTGCCCAACAATGCCAAGCTGGAGATGG...   \n",
      "46         IGLJ2*01  CAGTTCATATAGAGGTAGCGCCACTTTCGAGGTGGTGTTCGGCGGA...   \n",
      "47        TRGJP2*01  CATAATCCCCATGTGTCTAGGGAGAGACCTGGTGGGAGGTGATTGG...   \n",
      "15        TRGJP2*01  ACTGTGCCTGGCCCCAACATCTCCACTTCTGATCTCATTTTTCTGT...   \n",
      "20   IGLJ-C/OR18*01  AGCAGCCTCAGCCTCCCTGTCGTGGCCTGGAGCCTGGTCTCCCCTG...   \n",
      "98         IGLJ2*01  GCTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGGAGG...   \n",
      "67         IGLJ2*01  CTGAAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGC...   \n",
      "59         IGLJ6*01  CCTGGGCAGCACCTCCGGGACTGACCTTCGGCAGTGGCTGGGGGAC...   \n",
      "70         IGLJ7*01  CTGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGA...   \n",
      "71         IGLJ2*01  CTGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGC...   \n",
      "72         IGLJ2*01  CTGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGC...   \n",
      "73         IGLJ2*01  CTGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGC...   \n",
      "74         IGLJ2*01  CTGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGC...   \n",
      "76         IGLJ2*01  CTGCAGTTCATATCGAGGCAGCGCCACTTTCGAGGTGGTGTTCGGC...   \n",
      "\n",
      "                                              sequence productive vj_in_frame  \\\n",
      "120  GGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTCGAG...        NaN         NaN   \n",
      "75   AGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAG...        NaN         NaN   \n",
      "140  CCAGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGC...        NaN         NaN   \n",
      "156  AGACAGCAGCTTCTACATCTGCAGTGCTAGAGAGTCGACTAGCGAT...          T           T   \n",
      "80   GTGTACTTCTGTGCCATCAGTGAGCCGACAGGGATCAGAAGAGACC...          F           F   \n",
      "138  GGAATGGCTATCAGAAAGAAGACATTACTATTTATTACGGCAATTC...        NaN         NaN   \n",
      "116  GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCA...        NaN         NaN   \n",
      "153  CCCAGAAATCCAAGCCAGTTTTCTCAGAAAGCCTTTCTGACTGATC...        NaN         NaN   \n",
      "89   ATGTGCTCTGAGCTTCCCTGAACCAGGATCTGCCTATTACTGCTGT...        NaN         NaN   \n",
      "118  GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCA...        NaN         NaN   \n",
      "117  GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCA...        NaN         NaN   \n",
      "135  AGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAC...        NaN         NaN   \n",
      "7    GGAGGGAGTCAACGACCTATTGTATCAATTAGATCCCAGCCTGGAC...        NaN         NaN   \n",
      "43   CCTCACTCCAGGAAGGGGCTGGACTGGCTGTCCTTTCCAGTGGCCT...        NaN         NaN   \n",
      "129  CAGCTGTTCTCAGCAGGGCTCTCCGCAGGGAGGGGGCCGGCTCCTT...        NaN         NaN   \n",
      "134  CAGGCTGAGGCCGAGGCTGATTATTACTGCAGTTCATATAGAGCCA...        NaN         NaN   \n",
      "136  ATTATTACTGCAGTTCATATAGAGGCAGAGACACTGTCGAGGTGGT...        NaN         NaN   \n",
      "137  GTTAGAAAGCTACTCAACAGACATAATATTGTGTTTGGTGAGTACA...        NaN         NaN   \n",
      "139  GTTATGAGGTGGTGTTGGGAGGGAGGCTGGCACATCAACTGTCTTG...        NaN         NaN   \n",
      "133  ATACACATTATCCATCTTACAAGGTTAACAATTTTACACTAAAACA...        NaN         NaN   \n",
      "123  GGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACT...        NaN         NaN   \n",
      "128  CGAGGCTGATTATTACTGCAGTTCATATATAGGCAGCGCCACTTTC...        NaN         NaN   \n",
      "114  GAACAACATGGCGGCGGAAAGTGGTAGCGATTTTCAGCAGAGACGT...        NaN         NaN   \n",
      "127  GCAGTTCATATAGCGGCAGCGCCACTTTCGAGGTGGTGTTCGGCGG...        NaN         NaN   \n",
      "119  GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCA...        NaN         NaN   \n",
      "121  GGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACT...        NaN         NaN   \n",
      "122  GAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCA...        NaN         NaN   \n",
      "142  CCAGGCTGAGGACGAGGCTGATTATTCCTGCAGTTCATATAGAGGC...        NaN         NaN   \n",
      "124  GGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTCGAT...        NaN         NaN   \n",
      "125  GAGGACTAGGCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCA...        NaN         NaN   \n",
      "126  GATTATTACTGCAGTTCATATAGATGCAGCGCCACTTTCGAGGTGG...        NaN         NaN   \n",
      "141  CCAGGCTGAGGACGAGGCTGATTATTACTGCCGTTCATATAGAGGC...        NaN         NaN   \n",
      "143  GTCGGCCCCCAAACGTTTCACATAGGGCCAGGCAGCCTCTGTGTTT...        NaN         NaN   \n",
      "180  CTGCAGTTCATATAGAGGCAGCGCCACTTTCGAGGGGGTGTTCGGC...          T           T   \n",
      "145  GGTCAGAGCCCCTGCCGGGACCAAAGTGAACCTCGCCGCCTCCGCA...        NaN         NaN   \n",
      "146  GCTGATTATTACTGCAGTTCATATAGAGGCAGCGCCACTTTCGAGG...        NaN         NaN   \n",
      "160  CTGCCACTTACTAGCTGGGACACCCACATTTCATCTCTCATCTTCA...        NaN         NaN   \n",
      "162  CAAACTTGTCTTGAATAAGACAGATGCCTTGTGCCCCTACTGTGCC...        NaN         NaN   \n",
      "32   CCAGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGATCC...        NaN         NaN   \n",
      "45   GTGCTCGACCTTTCTCTCCAGTGGAGATTTGCCAACCTGCCCAACA...        NaN         NaN   \n",
      "46   CTGCAGTTCATATAGAGGTAGCGCCACTTTCGAGGTGGTGTTCGGC...          T           T   \n",
      "47   ATTTCATCTCGAATTGTAATCCCCATAATCCCCATGTGTCTAGGGA...        NaN         NaN   \n",
      "15   GTGATGGGATTACAGGCCTGGGCCACTGTGCCTGGCCCCAACATCT...        NaN         NaN   \n",
      "20   CTTCTCTTTCACAGCAGCCTCAGCCTCCCTGTCGTGGCCTGGAGCC...        NaN         NaN   \n",
      "98   GCTGATTATTACTGCAGCTCATATAGAGGCAGCGCCACTTTCGAGG...          T           T   \n",
      "67   CAGGCTTAGGACGAGGCTGATTATTACTGAAGTTCATATAGAGGCA...        NaN         NaN   \n",
      "59   GGAGCCTGGGCAGCACCTCCGGGACTGACCTTCGGCAGTGGCTGGG...        NaN         NaN   \n",
      "70   AGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCAG...        NaN         NaN   \n",
      "71   CAGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCA...        NaN         NaN   \n",
      "72   CAGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCA...        NaN         NaN   \n",
      "73   CAGGCTGAGCACGAGGCTGCTTATTACTGCAGTTCATATAGAGGCA...        NaN         NaN   \n",
      "74   CAGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATAGAGGCA...        NaN         NaN   \n",
      "76   CAGGCTGAGGACGAGGCTGATTATTACTGCAGTTCATATCGAGGCA...        NaN         NaN   \n",
      "\n",
      "    stop_codon  ... cdr3_sequence_start  cdr3_sequence_end     v_support  \\\n",
      "120        NaN  ...                 NaN                NaN  2.260000e-12   \n",
      "75         NaN  ...                 NaN                NaN  4.970000e-32   \n",
      "140        NaN  ...                 NaN                NaN  3.340000e-35   \n",
      "156          F  ...                23.0               64.0  4.960000e-50   \n",
      "80         NaN  ...                13.0               52.0  6.570000e-27   \n",
      "138        NaN  ...                 NaN                NaN  1.610000e-01   \n",
      "116        NaN  ...                 NaN                NaN  3.070000e-24   \n",
      "153        NaN  ...                 NaN                NaN  2.480000e-01   \n",
      "89         NaN  ...                 NaN                NaN  8.550000e-03   \n",
      "118        NaN  ...                 NaN                NaN  3.070000e-24   \n",
      "117        NaN  ...                 NaN                NaN  3.070000e-24   \n",
      "135        NaN  ...                 NaN                NaN  3.770000e-30   \n",
      "7          NaN  ...                 NaN                NaN  8.180000e-02   \n",
      "43         NaN  ...                 NaN                NaN  1.150000e-11   \n",
      "129        NaN  ...                 NaN                NaN  8.920000e-02   \n",
      "134        NaN  ...                 NaN                NaN  3.320000e-16   \n",
      "136        NaN  ...                 NaN                NaN  4.340000e-10   \n",
      "137        NaN  ...                 NaN                NaN  1.090000e-02   \n",
      "139        NaN  ...                 NaN                NaN  5.060000e-01   \n",
      "133        NaN  ...                 NaN                NaN  6.340000e-01   \n",
      "123        NaN  ...                 NaN                NaN  3.430000e-21   \n",
      "128        NaN  ...                 NaN                NaN  1.030000e-16   \n",
      "114        NaN  ...                 NaN                NaN  9.060000e-01   \n",
      "127        NaN  ...                 NaN                NaN  3.550000e-01   \n",
      "119        NaN  ...                 NaN                NaN  3.070000e-24   \n",
      "121        NaN  ...                 NaN                NaN  3.430000e-21   \n",
      "122        NaN  ...                 NaN                NaN  3.070000e-24   \n",
      "142        NaN  ...                 NaN                NaN  1.590000e-22   \n",
      "124        NaN  ...                 NaN                NaN  2.260000e-12   \n",
      "125        NaN  ...                 NaN                NaN  2.510000e-17   \n",
      "126        NaN  ...                 NaN                NaN  3.580000e-13   \n",
      "141        NaN  ...                 NaN                NaN  7.530000e-21   \n",
      "143        NaN  ...                 NaN                NaN  1.430000e-01   \n",
      "180          F  ...                 5.0               40.0  3.530000e-02   \n",
      "145        NaN  ...                 NaN                NaN  1.770000e-02   \n",
      "146        NaN  ...                 NaN                NaN  7.630000e-12   \n",
      "160        NaN  ...                 NaN                NaN  4.430000e-02   \n",
      "162        NaN  ...                 NaN                NaN  5.940000e-01   \n",
      "32         NaN  ...                 NaN                NaN  2.720000e-33   \n",
      "45         NaN  ...                 NaN                NaN  2.910000e-02   \n",
      "46           F  ...                 5.0               40.0  5.050000e-01   \n",
      "47         NaN  ...                 NaN                NaN  5.940000e-01   \n",
      "15         NaN  ...                 NaN                NaN  5.940000e-01   \n",
      "20         NaN  ...                 NaN                NaN  3.090000e-05   \n",
      "98           F  ...                16.0               51.0  1.460000e-12   \n",
      "67         NaN  ...                 NaN                NaN  1.490000e-07   \n",
      "59         NaN  ...                 NaN                NaN  1.090000e-02   \n",
      "70         NaN  ...                 NaN                NaN  4.970000e-32   \n",
      "71         NaN  ...                 NaN                NaN  1.300000e-33   \n",
      "72         NaN  ...                 NaN                NaN  1.300000e-33   \n",
      "73         NaN  ...                 NaN                NaN  6.920000e-09   \n",
      "74         NaN  ...                 NaN                NaN  1.300000e-33   \n",
      "76         NaN  ...                 NaN                NaN  1.020000e-31   \n",
      "\n",
      "        j_support  rev_comp  sequence_alignment  germline_alignment  v_cigar  \\\n",
      "120  5.010000e-43       NaN                 NaN                 NaN      NaN   \n",
      "75   6.830000e-14       NaN                 NaN                 NaN      NaN   \n",
      "140  2.930000e-05       NaN                 NaN                 NaN      NaN   \n",
      "156  2.080000e-38       NaN                 NaN                 NaN      NaN   \n",
      "80   4.820000e-89       NaN                 NaN                 NaN      NaN   \n",
      "138  1.880000e-01       NaN                 NaN                 NaN      NaN   \n",
      "116  4.500000e-10       NaN                 NaN                 NaN      NaN   \n",
      "153  1.500000e-09       NaN                 NaN                 NaN      NaN   \n",
      "89   9.340000e-02       NaN                 NaN                 NaN      NaN   \n",
      "118  4.920000e-13       NaN                 NaN                 NaN      NaN   \n",
      "117  4.500000e-10       NaN                 NaN                 NaN      NaN   \n",
      "135  1.170000e-11       NaN                 NaN                 NaN      NaN   \n",
      "7    8.520000e-01       NaN                 NaN                 NaN      NaN   \n",
      "43   2.710000e-03       NaN                 NaN                 NaN      NaN   \n",
      "129  4.170000e-01       NaN                 NaN                 NaN      NaN   \n",
      "134  4.760000e-09       NaN                 NaN                 NaN      NaN   \n",
      "136  5.220000e-32       NaN                 NaN                 NaN      NaN   \n",
      "137  3.270000e-01       NaN                 NaN                 NaN      NaN   \n",
      "139  2.910000e-01       NaN                 NaN                 NaN      NaN   \n",
      "133  3.290000e-01       NaN                 NaN                 NaN      NaN   \n",
      "123  6.910000e-16       NaN                 NaN                 NaN      NaN   \n",
      "128  2.580000e-37       NaN                 NaN                 NaN      NaN   \n",
      "114  1.120000e-06       NaN                 NaN                 NaN      NaN   \n",
      "127  2.840000e-58       NaN                 NaN                 NaN      NaN   \n",
      "119  1.580000e-22       NaN                 NaN                 NaN      NaN   \n",
      "121  3.150000e-22       NaN                 NaN                 NaN      NaN   \n",
      "122  1.580000e-22       NaN                 NaN                 NaN      NaN   \n",
      "142  2.930000e-05       NaN                 NaN                 NaN      NaN   \n",
      "124  5.010000e-43       NaN                 NaN                 NaN      NaN   \n",
      "125  4.920000e-13       NaN                 NaN                 NaN      NaN   \n",
      "126  3.370000e-14       NaN                 NaN                 NaN      NaN   \n",
      "141  2.930000e-05       NaN                 NaN                 NaN      NaN   \n",
      "143  1.230000e-01       NaN                 NaN                 NaN      NaN   \n",
      "180  9.520000e-60       NaN                 NaN                 NaN      NaN   \n",
      "145  3.290000e-01       NaN                 NaN                 NaN      NaN   \n",
      "146  3.030000e-20       NaN                 NaN                 NaN      NaN   \n",
      "160  2.800000e-01       NaN                 NaN                 NaN      NaN   \n",
      "162  5.080000e-05       NaN                 NaN                 NaN      NaN   \n",
      "32   1.120000e-01       NaN                 NaN                 NaN      NaN   \n",
      "45   3.160000e-04       NaN                 NaN                 NaN      NaN   \n",
      "46   2.210000e-55       NaN                 NaN                 NaN      NaN   \n",
      "47   8.320000e-03       NaN                 NaN                 NaN      NaN   \n",
      "15   6.470000e-03       NaN                 NaN                 NaN      NaN   \n",
      "20   2.630000e-01       NaN                 NaN                 NaN      NaN   \n",
      "98   4.350000e-46       NaN                 NaN                 NaN      NaN   \n",
      "67   5.770000e-11       NaN                 NaN                 NaN      NaN   \n",
      "59   2.950000e-01       NaN                 NaN                 NaN      NaN   \n",
      "70   9.370000e-11       NaN                 NaN                 NaN      NaN   \n",
      "71   2.950000e-02       NaN                 NaN                 NaN      NaN   \n",
      "72   2.950000e-02       NaN                 NaN                 NaN      NaN   \n",
      "73   2.950000e-02       NaN                 NaN                 NaN      NaN   \n",
      "74   6.600000e-08       NaN                 NaN                 NaN      NaN   \n",
      "76   5.770000e-11       NaN                 NaN                 NaN      NaN   \n",
      "\n",
      "     d_cigar  j_cigar  \n",
      "120      NaN      NaN  \n",
      "75       NaN      NaN  \n",
      "140      NaN      NaN  \n",
      "156      NaN      NaN  \n",
      "80       NaN      NaN  \n",
      "138      NaN      NaN  \n",
      "116      NaN      NaN  \n",
      "153      NaN      NaN  \n",
      "89       NaN      NaN  \n",
      "118      NaN      NaN  \n",
      "117      NaN      NaN  \n",
      "135      NaN      NaN  \n",
      "7        NaN      NaN  \n",
      "43       NaN      NaN  \n",
      "129      NaN      NaN  \n",
      "134      NaN      NaN  \n",
      "136      NaN      NaN  \n",
      "137      NaN      NaN  \n",
      "139      NaN      NaN  \n",
      "133      NaN      NaN  \n",
      "123      NaN      NaN  \n",
      "128      NaN      NaN  \n",
      "114      NaN      NaN  \n",
      "127      NaN      NaN  \n",
      "119      NaN      NaN  \n",
      "121      NaN      NaN  \n",
      "122      NaN      NaN  \n",
      "142      NaN      NaN  \n",
      "124      NaN      NaN  \n",
      "125      NaN      NaN  \n",
      "126      NaN      NaN  \n",
      "141      NaN      NaN  \n",
      "143      NaN      NaN  \n",
      "180      NaN      NaN  \n",
      "145      NaN      NaN  \n",
      "146      NaN      NaN  \n",
      "160      NaN      NaN  \n",
      "162      NaN      NaN  \n",
      "32       NaN      NaN  \n",
      "45       NaN      NaN  \n",
      "46       NaN      NaN  \n",
      "47       NaN      NaN  \n",
      "15       NaN      NaN  \n",
      "20       NaN      NaN  \n",
      "98       NaN      NaN  \n",
      "67       NaN      NaN  \n",
      "59       NaN      NaN  \n",
      "70       NaN      NaN  \n",
      "71       NaN      NaN  \n",
      "72       NaN      NaN  \n",
      "73       NaN      NaN  \n",
      "74       NaN      NaN  \n",
      "76       NaN      NaN  \n",
      "\n",
      "[53 rows x 30 columns]\n"
     ]
    }
   ],
   "source": [
    "df = pd.read_csv('tmp/vidjil/result.tsv',\n",
    "                 sep='\\t',\n",
    "                 low_memory=False, \n",
    "                 usecols=lambda c: not c.startswith('Unnamed:'))\n",
    "print(df.columns)\n",
    "df.sort_values('duplicate_count', inplace=True, ascending=False)\n",
    "df.dropna(subset=['v_call', 'j_call'], inplace=True)\n",
    "print(df)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
